--- alsa-utils-1.0.23.orig/configure.in	Sun Sep 19 13:44:56 2010
+++ alsa-utils-1.0.23/configure.in	Sun Sep 19 22:43:45 2010
@@ -261,6 +261,23 @@
 dnl Enable largefile support
 AC_SYS_LARGEFILE
 
+AC_CHECK_HEADERS([alloca.h endian.h byteswap.h sys/byteorder.h search.h err.h])
+AC_CHECK_HEADERS([sys/sockio.h features.h sys/ioccom.h])
+
+dnl Check for -lnsl
+AC_CHECK_LIB([nsl], [gethostent], [HAVE_LIBNSL="yes"])
+if test "$HAVE_LIBNSL" = "yes"; then
+  LIBS="$LIBS -lnsl"
+  AC_DEFINE([HAVE_LIBNSL], 1, [Have libnsl])
+fi
+
+dnl Check for -lsocket
+AC_CHECK_LIB([socket], [setsockopt], [HAVE_LIBSOCKET="yes"])
+if test "$HAVE_LIBSOCKET" = "yes"; then
+  LIBS="$LIBS -lsocket"
+  AC_DEFINE([HAVE_LIBSOCKET], 1, [Have libsocket])
+fi
+
 SAVE_UTIL_VERSION
 
 AC_SUBST(LIBRT)
