--- alsa-lib-1.0.23.orig/include/Makefile.am	Sun Sep 19 11:59:31 2010
+++ alsa-lib-1.0.23/include/Makefile.am	Sun Sep 19 12:09:43 2010
@@ -5,7 +5,8 @@
 
 alsainclude_HEADERS = asoundlib.h asoundef.h \
 		      version.h global.h input.h output.h error.h \
-		      conf.h control.h iatomic.h
+		      conf.h control.h iatomic.h byteswap-compat.h \
+		      endian-compat.h
 
 if BUILD_CTL_PLUGIN_EXT
 alsainclude_HEADERS += control_external.h
