diff --git a/plugins/js-debugger/debugger-server.c b/plugins/js-debugger/debugger-server.c
index 54d7aff..95c6a80 100644
--- a/plugins/js-debugger/debugger-server.c
+++ b/plugins/js-debugger/debugger-server.c
@@ -20,6 +20,9 @@
 #include <sys/socket.h>
 #include <sys/types.h>
 #include <sys/ioctl.h>
+#ifdef __sun
+#include <sys/filio.h>
+#endif
 #include <netinet/in.h>
 #include <netinet/tcp.h>
 #include <unistd.h>
