--- AssaultCube_v1.0.2/assaultcube.sh.orig	2009-09-08 13:09:58.689875510 +0100
+++ AssaultCube_v1.0.2/assaultcube.sh	2009-09-08 13:10:53.865875388 +0100
@@ -2,7 +2,7 @@
 # CUBE_DIR should refer to the directory in which Cube is placed.
 #CUBE_DIR=~/cube
 #CUBE_DIR=/usr/local/cube
-CUBE_DIR=./
+CUBE_DIR=/usr/share/assaultcube
 
 # CUBE_OPTIONS contains any command line options you would like to start Cube with.
 #CUBE_OPTIONS="-f"
@@ -49,6 +49,7 @@
 
 if [ -x ${CUBE_DIR}/bin_unix/${SYSTEM_NAME}${MACHINE_NAME}client ]
 then
+  unset LANG # Solaris bug
   cd ${CUBE_DIR}
   exec ${CUBE_DIR}/bin_unix/${SYSTEM_NAME}${MACHINE_NAME}client ${CUBE_OPTIONS} "$@"
 else
