--- 1.1.0.4/assaultcube.sh	Sun Nov 14 19:48:42 2010
+++ 1.1.0.4/assaultcube.sh	Sun Oct 23 19:59:30 2011
@@ -3,7 +3,8 @@
 #CUBE_DIR=~/cube
 #CUBE_DIR=/usr/local/cube
 #CUBE_DIR=./
-CUBE_DIR=$(dirname $(readlink -f "${0}"))
+#CUBE_DIR=$(dirname $(readlink -f "${0}"))
+CUBE_DIR=/usr/share/assaultcube
 
 # CUBE_OPTIONS contains any command line options you would like to start Cube with.
 #CUBE_OPTIONS="-f"
