--- audacity-src-1.3.9/m4/audacity_checklib_libsoundtouch.m4-orig	2009-09-08 22:53:50.626057506 -0500
+++ audacity-src-1.3.9/m4/audacity_checklib_libsoundtouch.m4	2009-09-08 22:53:58.026847623 -0500
@@ -17,7 +17,7 @@ AC_DEFUN([AUDACITY_CHECKLIB_LIBSOUNDTOUC
 
    dnl see if soundtouch is installed on the system
 
-   PKG_CHECK_MODULES(SOUNDTOUCH, soundtouch-1.0 >= 1.3.0,
+   PKG_CHECK_MODULES(SOUNDTOUCH, soundtouch-1.4 >= 1.3.0,
                      soundtouch_available_system="yes",
                      soundtouch_available_system="no")
 
