--- blender-2.49b/release/Makefile.orig	2011-06-15 12:42:11.797465500 -0400
+++ blender-2.49b/release/Makefile	2011-06-15 12:43:06.961373500 -0400
@@ -196,8 +196,8 @@
 	@echo "----> Compressing distribution to $(BLENDNAME)$(EXT1)$(EXT2)"
 	@$(COMPRESS) $(COMPRESSFLAGS) $(NAN_OBJDIR)/$(VERSION)/$(BLENDNAME)$(EXT1)
     endif
-	@#echo "****> Clean up temporary distribution directory"
-	@rm -fr $(DISTDIR)
+	#@#echo "****> Clean up temporary distribution directory"
+	#@rm -fr $(DISTDIR)
 	@echo "****> $(NAN_OBJDIR)/$(VERSION)/$(BLENDNAME)$(EXT1)$(EXT2) is ready"
 
 version: FORCE
