--- boost_1_45_0/tools/build/v2/tools/sun.jam.orig	Sun Jan 23 21:22:59 2011
+++ boost_1_45_0/tools/build/v2/tools/sun.jam	Sun Jan 23 21:22:14 2011
@@ -23,6 +23,11 @@
     : <cxxflags>-library=stlport4 <linkflags>-library=stlport4
     ;
 
+feature.extend stdlib : sun-stdcxx ;
+feature.compose <stdlib>sun-stdcxx
+    : <cxxflags>-library=stdcxx4 <linkflags>-library=stdcxx4
+    ;
+
 rule init ( version ? : command * : options * ) 
 {
     local condition = [ 
