--- boost_1_43_0/tools/build/v2/tools/sun.jam.bjam-stdcxx	2010-08-07 15:05:33.278522136 +0200
+++ boost_1_43_0/tools/build/v2/tools/sun.jam	2010-08-07 15:05:01.704665651 +0200
@@ -20,7 +20,7 @@
 
 feature.extend stdlib : sun-stlport ;
 feature.compose <stdlib>sun-stlport
-    : <cxxflags>-library=stlport4 <linkflags>-library=stlport4
+    : <cxxflags>-library=%none <linkflags>-library=%none
     ;
 
 rule init ( version ? : command * : options * ) 
