--- cherokee-0.99.15/cherokee/spawner.c.orig	2009-06-02 17:56:42.883392390 +0800
+++ cherokee-0.99.15/cherokee/spawner.c	2009-06-02 17:55:56.849313888 +0800
@@ -225,7 +225,7 @@
 	 */
 	ret = cherokee_sem_post (&cherokee_spawn_sem);	
 	if (unlikely (ret != ret_ok)) {
-		PRINT_ERROR ("WARNING: Couldn't unlock spawning semaphore..\n");
+		PRINT_ERROR_S ("WARNING: Couldn't unlock spawning semaphore..\n");
 	}
 	
 	/* Wait for the PID
