--- clutter-0.9.2/tests/conform/test-conform-common.h-orig	2009-04-03 11:39:29.232810000 -0500
+++ clutter-0.9.2/tests/conform/test-conform-common.h	2009-04-03 11:39:44.315481000 -0500
@@ -15,6 +15,7 @@ typedef struct _TestConformSharedState
  * the test is finished. */
 typedef struct _TestConformSimpleFixture
 {
+  gboolean dummy;
   /**/
 } TestConformSimpleFixture;
 
--- clutter-0.9.2/build/autotools/dolt.m4-orig	2009-04-03 23:23:02.676608000 -0500
+++ clutter-0.9.2/build/autotools/dolt.m4	2009-04-03 23:28:27.200552000 -0500
@@ -18,9 +18,6 @@ dolt_supported=yes
 if test x$DOLT_BASH = x; then
     dolt_supported=no
 fi
-if test x$GCC != xyes; then
-    dolt_supported=no
-fi
 case $host in
 i?86-*-linux*|x86_64-*-linux*|powerpc-*-linux* \
 |amd64-*-freebsd*|i?86-*-freebsd*|ia64-*-freebsd*)
@@ -29,6 +26,9 @@ i?86-*-linux*|x86_64-*-linux*|powerpc-*-
 i?86-apple-darwin*)
     pic_options='-fno-common'
     ;;
+*solaris*)
+    pic_options='-KPIC'
+    ;;
 *)
     dolt_supported=no
     ;;
