diff --git a/tools/ck-dynamic.c b/tools/ck-dynamic.c
index d9279e2..5a660f7 100644
--- a/tools/ck-dynamic.c
+++ b/tools/ck-dynamic.c
@@ -137,6 +137,7 @@ main (int argc, char *argv[])
         gboolean         list = FALSE;
         gboolean         show_version = FALSE;
         char            *type = NULL;
+        char            *tty = NULL;
         int              num = -1;
         gboolean         res;
 
@@ -146,6 +147,7 @@ main (int argc, char *argv[])
                 { "list", 'l', 0, G_OPTION_ARG_NONE, &list, N_("List all attached displays"), NULL},
                 { "num", 'n', 0, G_OPTION_ARG_INT, &num, N_("Display number"), NULL},
                 { "type", 't', 0, G_OPTION_ARG_STRING, &type, N_("Display type"), NULL},
+                { "tty", 'T', 0, G_OPTION_ARG_STRING, &tty, N_("Tty device"), NULL},
                 { "version", 'V', 0, G_OPTION_ARG_NONE, &show_version, N_("Version of this application"), NULL },
                 { NULL }
         };   
@@ -213,7 +215,7 @@ main (int argc, char *argv[])
 
 
         if (add) {
-                create_dynamic_session (connection, proxy, num, type, NULL);
+                create_dynamic_session (connection, proxy, num, type, tty);
         } else if (delete) {
                 remove_dynamic_session (connection, proxy, num);
         } else if (list) {
