--- cppcheck-1.54/cli/cmdlineparser.cpp	2012-04-14 22:25:08.000000000 -0700
+++ cppcheck-1.54-patched/cli/cmdlineparser.cpp	2012-05-19 01:21:07.495742393 -0700
@@ -184,8 +184,8 @@
                 std::string message("cppcheck: Couldn't open the file: \"");
                 message += std::string(filename);
                 message += "\".";
-                if (count(filename.begin(), filename.end(), ',') > 0 ||
-                    count(filename.begin(), filename.end(), '.') > 1) {
+                if (std::count(filename.begin(), filename.end(), ',') > 0 ||
+                    std::count(filename.begin(), filename.end(), '.') > 1) {
                     // If user tried to pass multiple files (we can only guess that)
                     // e.g. like this: --suppressions-list=a.txt,b.txt
                     // print more detailed error message to tell user how he can solve the problem
