--- ctrlproxy-3.0.8.orig/configure.ac	2009-10-07 11:58:39.842762000 -0400
+++ ctrlproxy-3.0.8/configure.ac	2009-10-07 11:59:04.269585297 -0400
@@ -31,9 +31,15 @@
 # Checks for programs.
 AC_PROG_CC
 AC_PROG_INSTALL
-AC_PATH_PROG(LD,ld)
 AC_PATH_PROG(IRCDTORTURE,ircdtorture)
 
+if test "$GCC" = "yes"; then
+	CFLAGS="$CFLAGS -ansi -Wall"
+	DYNAMIC="-rdynamic"
+fi
+AC_SUBST(GCC)
+AC_SUBST(DYNAMIC)
+
 AC_ARG_ENABLE(gcov,
 [ --enable-gcov		Enable GCOV support ],
 [ WITH_GCOV=1 ], [ WITH_GCOV=0 ])
--- ctrlproxy-3.0.8.orig/Makefile	2008-12-06 12:36:33.000000000 -0500
+++ ctrlproxy-3.0.8/Makefile	2009-10-07 11:57:11.853721092 -0400
@@ -18,7 +18,7 @@
 CFLAGS += $(GNUTLS_CFLAGS)
 
 CFLAGS+=-DHAVE_CONFIG_H -DDEFAULT_CONFIG_DIR=\"$(DEFAULT_CONFIG_DIR)\" -DHELPFILE=\"$(HELPFILE)\"
-CFLAGS+=-ansi -Wall -DMODULESDIR=\"$(modulesdir)\" -DSTRICT_MEMORY_ALLOCS=
+CFLAGS+=-DMODULESDIR=\"$(modulesdir)\" -DSTRICT_MEMORY_ALLOCS=
 
 LIBIRC_STATIC = libirc.a
 LIBIRC = $(LIBIRC_STATIC)
@@ -103,11 +103,11 @@
 
 linestack-cmd$(EXEEXT): src/linestack-cmd.o $(objs) $(LIBIRC)
 	@echo Linking $@
-	@$(LD) $(LIBS) -lreadline -rdynamic -o $@ $^
+	@$(CC) $(LIBS) -lreadline $(DYNAMIC) -o $@ $^
 
 ctrlproxy$(EXEEXT): src/main.o $(objs) $(LIBIRC)
 	@echo Linking $@
-	@$(LD) $(LDFLAGS) -rdynamic -o $@ $^ $(LIBS)
+	@$(CC) $(LDFLAGS) $(DYNAMIC) -o $@ $^ $(LIBS)
 
 src/settings.o: CFLAGS+=-DSYSCONFDIR=\"${sysconfdir}\"
 
@@ -119,11 +119,11 @@
 
 ctrlproxyd$(EXEEXT): $(daemon_objs) $(objs) $(LIBIRC)
 	@echo Linking $@
-	@$(LD) $(LDFLAGS) -rdynamic -o $@ $^ $(LIBS)
+	@$(CC) $(LDFLAGS) $(DYNAMIC) -o $@ $^ $(LIBS)
 
 ctrlproxy-admin$(EXEEXT): src/admin-cmd.o
 	@echo Linking $@
-	@$(LD) $(LDFLAGS) -rdynamic -o $@ $^ $(LIBS)
+	@$(CC) $(LDFLAGS) $(DYNAMIC) -o $@ $^ $(LIBS)
 
 %.o: %.c
 	@echo Compiling $<
@@ -215,7 +215,7 @@
 	@ar -rcs $@ $^
 
 $(LIBIRC_SHARED): $(lib_objs)
-	$(LD) -shared $(LDFLAGS) -Wl,-soname,$(LIBIRC_SONAME) -o $@ $^
+	$(CC) -shared $(LDFLAGS) -Wl,-soname,$(LIBIRC_SONAME) -o $@ $^
 
 cscope.out::
 	cscope -b -R
@@ -286,4 +286,6 @@
 check-gdb: 
 	$(MAKE) check-nofork DEBUGGER="gdb --args"
 
+ifeq ($(GCC),yes)
 -include $(dep_files)
+endif
--- ctrlproxy-3.0.8.orig/src/log_subst.c	2008-12-06 12:36:34.000000000 -0500
+++ ctrlproxy-3.0.8/src/log_subst.c	2009-10-07 11:03:11.593731248 -0400
@@ -30,7 +30,7 @@
 struct log_mapping {
 	char *command;
 	char subst;
-	unsigned int index;
+	size_t index;
 	/* If index is -1 */
 	char *(*callback) (struct subst_context *subst_ctx, 
 					   const struct irc_line *line,
@@ -161,20 +161,20 @@
 }
 
 static struct log_mapping mappings[] = {
-	{NULL, '@', -1, get_identifier },
-	{NULL, 'h', -1, get_hours },
-	{NULL, 'M', -1, get_minutes },
-	{NULL, 's', -1, get_seconds },
-	{NULL, 'd', -1, get_day },
-	{NULL, 'B', -1, get_month },
-	{NULL, 'Y', -1, get_year },
-	{NULL, 'e', -1, get_seconds_since_1970 },
-	{NULL, 'b', -1, get_monthname },
-	{NULL, 'n', -1, get_nick },
-	{NULL, 'u', -1, get_user },
-	{NULL, 'N', -1, get_network },
-	{NULL, 'S', -1, get_server },
-	{NULL, '%', -1, get_percent },
+	{NULL, '@', (size_t)-1, get_identifier },
+	{NULL, 'h', (size_t)-1, get_hours },
+	{NULL, 'M', (size_t)-1, get_minutes },
+	{NULL, 's', (size_t)-1, get_seconds },
+	{NULL, 'd', (size_t)-1, get_day },
+	{NULL, 'B', (size_t)-1, get_month },
+	{NULL, 'Y', (size_t)-1, get_year },
+	{NULL, 'e', (size_t)-1, get_seconds_since_1970 },
+	{NULL, 'b', (size_t)-1, get_monthname },
+	{NULL, 'n', (size_t)-1, get_nick },
+	{NULL, 'u', (size_t)-1, get_user },
+	{NULL, 'N', (size_t)-1, get_network },
+	{NULL, 'S', (size_t)-1, get_server },
+	{NULL, '%', (size_t)-1, get_percent },
 	{"JOIN", 'c', 1, NULL },
 	{"PART", 'c', 1, NULL },
 	{"PART", 'm', 2, NULL },
@@ -192,7 +192,7 @@
 	{"TOPIC", 't', 2, NULL },
 	{"MODE", 't', 1, NULL },
 	{"MODE", 'p', 2, NULL },
-	{"MODE", 'c', -1, get_modechanges },
+	{"MODE", 'c', (size_t)-1, get_modechanges },
 	{"NICK", 'r', 1, NULL },
 	{NULL, '0', 0, NULL },
 	{NULL, '1', 1, NULL },
--- ctrlproxy-3.0.8.orig/src/settings.c	2008-12-06 12:36:34.000000000 -0500
+++ ctrlproxy-3.0.8/src/settings.c	2009-10-07 10:54:42.998143329 -0400
@@ -1009,6 +1009,9 @@
 	g_strfreev(groups);
 }
 
+#ifndef __STRING
+#define __STRING(x) #x
+#endif
 #define FETCH_SETTING(data, kf, section, prefix, name) (data)->name = g_key_file_get_string((kf), (section), prefix __STRING(name), NULL)
 #define STORE_SETTING(data, kf, section, prefix, name) g_key_file_set_string((kf), (section), prefix __STRING(name), (data)->name)
 
