--- dillo-2.2.1/dw/fltkui.cc.orig	2011-07-15 14:06:50.000000000 +0100
+++ dillo-2.2.1/dw/fltkui.cc	2011-08-30 22:23:21.322287767 +0100
@@ -989,9 +989,11 @@ template <class I> void FltkSelectionRes
          itemWidget->set_selected();
          if (setSelectedItems ()) {
             // Handle multiple item selection.
-            int *pos = new int[widgetStack->stack->size ()];
-            int i;
-            Iterator <TypedPointer < ::fltk::Menu> > it;
+            // int *pos = new int[widgetStack->stack->size ()];
+	 int *pos = NULL;            
+	 int i;
+	 pos = new int[widgetStack->stack->size ()];           
+	    Iterator <TypedPointer < ::fltk::Menu> > it;
             for (it = widgetStack->stack->iterator (),
                     i = widgetStack->stack->size () - 1;
                  it.hasNext ();
