--- doxygen-1.7.1/configure-orig	2010-08-04 15:21:59.255379308 -0500
+++ doxygen-1.7.1/configure	2010-08-04 15:22:13.424058454 -0500
@@ -214,9 +214,9 @@ if test -z "$f_platform"; then
       f_platform=sunos-g++
       ;;
     SunOS:5*)
-      f_platform=solaris-g++
+      f_platform=solaris-cc
       if test "$f_insttool" = NO; then
-        f_insttool=/usr/bin/install
+        f_insttool=install
       fi
       ;;
     ULTRIX:*)
--- doxygen-1.7.1/Makefile.in-orig	2010-08-04 15:22:25.812675587 -0500
+++ doxygen-1.7.1/Makefile.in	2010-08-04 15:24:44.274479289 -0500
@@ -44,13 +44,13 @@ distclean: clean
 
 DATE=$(shell date "+%B %Y")
 
-MAN1DIR = man/man1
+MAN1DIR = share/man/man1
 
 install: doxywizard_install 
-	$(INSTTOOL) -d $(DESTDIR)/$(INSTALL)/bin
+	mkdir -p $(DESTDIR)/$(INSTALL)/bin
 	$(INSTTOOL) -m 755 bin/doxygen    $(DESTDIR)/$(INSTALL)/bin
 	$(INSTTOOL) -m 755 bin/doxytag    $(DESTDIR)/$(INSTALL)/bin
-	$(INSTTOOL) -d $(DESTDIR)/$(INSTALL)/$(MAN1DIR)
+	mkdir -p $(DESTDIR)/$(INSTALL)/$(MAN1DIR)
 	cat doc/doxygen.1    | sed -e "s/DATE/$(DATE)/g" -e "s/VERSION/$(VERSION)/g" > doxygen.1 
 	$(INSTTOOL) -m 644 doxygen.1 $(DESTDIR)/$(INSTALL)/$(MAN1DIR)/doxygen.1
 	rm doxygen.1
