--- epiphany.orig/configure	Fri Feb 12 08:42:47 2010
+++ epiphany/configure	Fri Feb 12 08:43:19 2010
@@ -17486,8 +17486,8 @@
 if test "$have_iso_codes" = "yes"; then
 	{ $as_echo "$as_me:$LINENO: checking whether iso-codes has iso-639 and iso-3166 domains" >&5
 $as_echo_n "checking whether iso-codes has iso-639 and iso-3166 domains... " >&6; }
-	if $PKG_CONFIG --variable=domains iso-codes | grep -q 639 && \
-	   $PKG_CONFIG --variable=domains iso-codes | grep -q 3166 ; then
+	if $PKG_CONFIG --variable=domains iso-codes | grep 639 && \
+	   $PKG_CONFIG --variable=domains iso-codes | grep 3166 ; then
 		result=yes
 	else
 		result=no
--- epiphany.orig/configure.ac	Fri Feb 12 08:43:56 2010
+++ epiphany/configure.ac	Fri Feb 12 08:44:39 2010
@@ -384,8 +384,8 @@
 
 if test "$have_iso_codes" = "yes"; then
 	AC_MSG_CHECKING([whether iso-codes has iso-639 and iso-3166 domains])
-	if $PKG_CONFIG --variable=domains iso-codes | grep -q 639 && \
-	   $PKG_CONFIG --variable=domains iso-codes | grep -q 3166 ; then
+	if $PKG_CONFIG --variable=domains iso-codes | grep 639 && \
+	   $PKG_CONFIG --variable=domains iso-codes | grep 3166 ; then
 		result=yes
 	else
 		result=no
