--- ffmpeg-0.11/doc/Makefile	Sat May 26 20:40:33 2012
+++ ffmpeg-0.11/doc/Makefile	Sat May 26 20:51:17 2012
@@ -41,7 +41,7 @@
 doc/%.html: TAG = HTML
 doc/%.html: doc/%.texi $(SRC_PATH)/doc/t2h.init $(GENTEXI)
 	$(Q)$(TEXIDEP)
-	$(M)texi2html -I doc -monolithic --init-file $(SRC_PATH)/doc/t2h.init --output $@ $<
+	$(M)texi2html -Idoc -monolithic $<
 
 doc/%.pod: TAG = POD
 doc/%.pod: doc/%.texi $(GENTEXI)
