
Solaris names the include file sys/videodev2.h

--- ffmpeg-0.8.2/configure.orig	2011-08-13 17:32:19.002653025 +0200
+++ ffmpeg-0.8.2/configure	2011-08-13 17:47:31.193139160 +0200
@@ -1474,7 +1474,7 @@
 sndio_indev_deps="sndio_h"
 sndio_outdev_deps="sndio_h"
 v4l_indev_deps="linux_videodev_h"
-v4l2_indev_deps_any="linux_videodev2_h sys_videoio_h"
+v4l2_indev_deps_any="linux_videodev2_h sys_videoio_h sys_videodev2_h"
 vfwcap_indev_deps="capCreateCaptureWindow vfwcap_defines"
 vfwcap_indev_extralibs="-lavicap32"
 x11_grab_device_indev_deps="x11grab XShmCreateImage"
