--- ffmpeg-0.5/configure	Mon Mar 16 21:05:16 2009
+++ ffmpeg-0.5-patch/configure	Mon Mar 16 22:00:40 2009
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/ksh
 #
 # FFmpeg configure script
 #
@@ -1227,7 +1227,7 @@
     thing=$1
     pattern=$2
     file=$source_path/$3
-    sed -n "s/^[^#]*$pattern.*([^,]*, *\([^,]*\)\(,.*\)*).*/\1_$thing/p" "$file"
+    sed -n "s/^[^#]*$pattern.*([^,]*, *\([^,)]*\)[,)].*/\1_$thing/p" "$file"
 }
 
 ENCODER_LIST=$(find_things  encoder  ENC      libavcodec/allcodecs.c)
@@ -2026,7 +2026,7 @@
     echo
 fi
 
-if enabled vhook; then
+if enabled vhook && test "$target_os" != sunos; then
     check_ldflags -rdynamic
     check_ldflags -export-dynamic
 fi
@@ -2063,7 +2063,8 @@
     restore_flags
 fi
 
-texi2html -version > /dev/null 2>&1 && enable texi2html || disable texi2html
+(texi2html -version > /dev/null 2>&1 || texi2html /dev/null > /dev/null 2>&1) && enable texi2html || disable texi2html
+pod2man /dev/null >/dev/null 2>&1 && enable pod2man || disable pod2man
 
 ##########################################
 # Network check
@@ -2407,7 +2408,7 @@
     echo "SDL_LIBS=`"${SDL_CONFIG}" --libs`" >> config.mak
     echo "SDL_CFLAGS=`"${SDL_CONFIG}" --cflags`" >> config.mak
 fi
-if enabled texi2html; then
+if enabled texi2html && enabled pod2man; then
     echo "BUILD_DOC=yes" >> config.mak
 fi
 
