--- fltk-2.0.x-r7513/configure.in	Fri Jul 24 01:08:12 2009
+++ fltk-2.0.x-r7513-patch/configure.in	Sat Jun 12 20:56:38 2010
@@ -327,7 +327,7 @@
 AC_CHECK_HEADER(sys/select.h,AC_DEFINE(HAVE_SYS_SELECT_H))
 AC_CHECK_HEADER(sys/stdtypes.h,AC_DEFINE(HAVE_SYS_SELECT_H))
 AC_CHECK_FUNC(scandir,[
-    if test "$uname" = SunOS -o "$uname" = QNX; then
+    if test "$uname" = QNX; then
         AC_MSG_WARN(Not using $uname scandir emulation function.)
     else
         AC_DEFINE(HAVE_SCANDIR)
--- fltk-2.0.x-r7513/src/filename_list.cxx	Fri Jul 24 01:08:12 2009
+++ fltk-2.0.x-r7513-patch/src/filename_list.cxx	Sat Jun 12 20:59:22 2010
@@ -67,6 +67,9 @@
   int n = scandir(d, list, 0, (int(*)(const dirent **, const dirent **))sort);
 #elif defined(__linux) || defined (__FreeBSD__) || defined (__NetBSD__)
   int n = scandir(d, list, 0, (int(*)(const void*,const void*))sort);
+#elif defined(sun)
+  // Solaris
+  int n = scandir(d, list, 0, (int(*)(const struct dirent **, const struct dirent **))sort);
 #elif defined(__osf__) || defined(__sgi)
   // OSF, DU 4.0x
   int n = scandir(d, list, 0, (int(*)(dirent **, dirent **))sort);
