--- fluidsynth-1.1.5/src/bindings/fluid_rtkit.c	Sun Sep  4 09:38:58 2011
+++ fluidsynth-1.1.5/src/bindings/fluid_rtkit.c	Tue Feb  7 19:30:25 2012
@@ -34,7 +34,7 @@
 #include "fluid_rtkit.h"
 
 
-#if defined(__linux__) || defined(__APPLE__)
+#if defined(__linux__) || defined(__APPLE__) || defined(__sun)
 
 #ifndef _GNU_SOURCE
 #define _GNU_SOURCE
@@ -49,7 +49,7 @@
 
 
 static pid_t _gettid(void) {
-        return (pid_t) syscall(SYS_gettid);
+        return (pid_t) syscall(SYS_lwp_self);
 }
 
 static int translate_error(const char *name) {
