--- freeciv-2.2.3/common/map.h	2010-09-12 22:15:39.000000000 +0200
+++ freeciv-2.2.3-patch/common/map.h	2010-09-23 17:10:15.762963900 +0200
@@ -49,7 +49,7 @@
 
   union {
     struct {
-      /* Nothing yet. */
+      int i; /* Nothing yet. */
     } client;
 
     struct {
--- freeciv-2.2.3/common/game.h	2010-09-12 22:15:39.000000000 +0200
+++ freeciv-2.2.3-patch/common/game.h	2010-09-23 17:11:14.412110700 +0200
@@ -64,7 +64,7 @@
 
   union {
     struct {
-      /* Nothing yet. */
+      int i; /* Nothing yet. */
     } client;
 
     struct {
