--- fuse-ext2-0.0.7/fuse-ext2/Makefile.am.orig	2011-11-09 22:54:30.548044951 +0000
+++ fuse-ext2-0.0.7/fuse-ext2/Makefile.am	2011-11-09 22:55:26.091722514 +0000
@@ -17,7 +17,6 @@
 
 if DARWIN
 fuse_ext2_probe_CFLAGS = \
-	-Wall \
 	-DHAVE_CONFIG_H \
 	-I$(top_srcdir)/e2fsprogs-1.41.9 \
 	-I/usr/local/include
@@ -30,7 +29,6 @@
 endif
 if LINUX
 fuse_ext2_probe_CFLAGS = \
-	-Wall \
 	-DHAVE_CONFIG_H \
 	-I/usr/local/include
 
@@ -116,7 +114,6 @@
 	op_rename.c
 
 umfuseext2_la_CFLAGS = \
-	-Wall \
 	-DHAVE_CONFIG_H \
 	-D_GNU_SOURCE \
 	$(DEVELFLAGS) \
@@ -131,7 +128,6 @@
 endif
 
 fuse_ext2_CFLAGS = \
-	-Wall \
 	-DHAVE_CONFIG_H \
 	-I$(top_srcdir)/e2fsprogs-1.41.9 \
 	-I/usr/local/include
@@ -146,7 +142,6 @@
 	fuse-ext2.wait.c
 
 fuse_ext2_wait_CFLAGS = \
-	-Wall \
 	-D__FreeBSD__=10
 
 fuse_ext2_wait_LDFLAGS = \
