--- gdm-2.26.0/daemon/gdm-session-solaris-auditor.c-orig	2009-04-08 02:32:14.739082000 -0500
+++ gdm-2.26.0/daemon/gdm-session-solaris-auditor.c	2009-04-08 02:32:27.413671000 -0500
@@ -177,17 +177,17 @@ gdm_session_solaris_auditor_report_login
 
                 /* If display is on console or VT */
                 if (hostname != NULL && hostname[0] != '\0') {
-                        /* login from the local host */
-                        if (adt_load_ttyname (display_device, &tid) != 0) {
-                                syslog (LOG_AUTH | LOG_ALERT,
-                                        "adt_loadhostname (localhost): %m");
-                        }
-                } else {
                         /* Login from a remote host */
                         if (adt_load_hostname (hostname, &tid) != 0) {
                                 syslog (LOG_AUTH | LOG_ALERT,
                                         "adt_loadhostname (%s): %m", hostname);
                         }
+                } else {
+                        /* login from the local host */
+                        if (adt_load_ttyname (display_device, &tid) != 0) {
+                                syslog (LOG_AUTH | LOG_ALERT,
+                                        "adt_loadhostname (localhost): %m");
+                        }
                 }
 
                 if (adt_set_user (ah,
--- gdm-2.26.0/daemon/gdm-session-worker.c-orig	2009-04-08 03:05:36.539234000 -0500
+++ gdm-2.26.0/daemon/gdm-session-worker.c	2009-04-08 03:05:41.106313000 -0500
@@ -1122,6 +1122,12 @@ gdm_session_worker_uninitialize_pam (Gdm
 #endif  /* HAVE_LOGINDEVPERM */
 
         } else {
+                void *p;
+
+                if ((pam_get_item (worker->priv->pam_handle, PAM_USER, &p)) == PAM_SUCCESS) {
+                        gdm_session_auditor_set_username (worker->priv->auditor, (const char *)p);
+                }
+
                 gdm_session_auditor_report_login_failure (worker->priv->auditor,
                                                           status,
                                                           pam_strerror (worker->priv->pam_handle, status));
@@ -1144,7 +1150,11 @@ static char *
 _get_tty_for_pam (const char *x11_display_name,
                   const char *display_device)
 {
+#ifdef __sun
+        return g_strdup (display_device);
+#else
         return g_strdup (x11_display_name);
+#endif
 }
 
 #ifdef PAM_XAUTHDATA
