diff --git a/daemon/gdm-welcome-session.c b/daemon/gdm-welcome-session.c
index 4f5b4e1..a356cd4 100644
--- a/daemon/gdm-welcome-session.c
+++ b/daemon/gdm-welcome-session.c
@@ -280,6 +280,8 @@ get_welcome_environment (GdmWelcomeSession *welcome_session)
 
         g_hash_table_insert (hash, g_strdup ("XAUTHORITY"), g_strdup (welcome_session->priv->x11_authority_file));
         g_hash_table_insert (hash, g_strdup ("DISPLAY"), g_strdup (welcome_session->priv->x11_display_name));
+	/* Due to HOME dirctory of gdm user, use tmp dir instead */
+        g_hash_table_insert (hash, g_strdup ("ICEAUTHORITY"), g_strdup ("/tmp/.ICEauthority"));
 
 #if 0
         /* hackish ain't it */
diff --git a/gui/simple-greeter/gdm-option-widget.c b/gui/simple-greeter/gdm-option-widget.c
index 15b7340..9afb5e2 100644
--- a/gui/simple-greeter/gdm-option-widget.c
+++ b/gui/simple-greeter/gdm-option-widget.c
@@ -282,9 +282,10 @@ gdm_option_widget_set_default_item (GdmOptionWidget *widget,
                                     const char      *item)
 {
         g_return_if_fail (GDM_IS_OPTION_WIDGET (widget));
-        g_return_if_fail (item == NULL ||
-                          gdm_option_widget_lookup_item (widget, item,
-                                                         NULL, NULL, NULL));
+        if ((item != NULL &&
+                          ! gdm_option_widget_lookup_item (widget, item,
+                                                         NULL, NULL, NULL)))
+	return;
 
         if (widget->priv->default_item_id == NULL ||
             item == NULL ||
