diff --git a/acconfig.h b/acconfig.h
index 4307fd3..83a4a59 100644
--- a/acconfig.h
+++ b/acconfig.h
@@ -30,7 +30,6 @@
 #undef HAVE_SETENV
 #undef HAVE_SETRESUID
 #undef HAVE_SHADOW
-#undef HAVE_SOLARIS_XINERAMA
 #undef HAVE_STPCPY
 #undef HAVE_SYS_SOCKIO_H
 #undef HAVE_TCPWRAPPERS
diff --git a/configure.ac b/configure.ac
index 8a09f68..04d2951 100644
--- a/configure.ac
+++ b/configure.ac
@@ -799,27 +799,6 @@ fi
 ALL_X_LIBS="$X_LIBS $X_PRE_LIBS -lX11 $X_EXTRA_LIBS"
 
 if test ! x$with_xinerama = xno ; then
-  if test x$os_solaris = xyes ; then
-	# Check for solaris
-	use_solaris_xinerama=yes
-	AC_CHECK_LIB(Xext, XineramaGetInfo,
-                     use_solaris_xinerama=yes, use_solaris_xinerama=no,
-                     $ALL_X_LIBS)
-	if test "x$use_solaris_xinerama" = "xyes"; then
-            AC_CHECK_HEADER(X11/extensions/xinerama.h,
-                            if test -z "`echo $ALL_X_LIBS | grep "\-lXext" 2> /dev/null`"; then
- 				X_EXTRA_LIBS="-lXext $X_EXTRA_LIBS"
-                            fi
-                            AC_DEFINE(HAVE_SOLARIS_XINERAMA, 1, [Define if have Solaris xinerama])
-                            AC_DEFINE(HAVE_XINERAMA, 1, [Define if have xinerama])
-		     	    XINERAMA_LIBS=""
-		     	    XINERAMA_SUPPORT=yes,
-                            use_solaris_xinerama=no,
-                            [#include <X11/Xlib.h>])
-	fi
-        AC_MSG_CHECKING(for Xinerama support on Solaris)
-	AC_MSG_RESULT($use_solaris_xinerama);
-  else
       	# Check for XFree
 	use_xfree_xinerama=yes
 	AC_CHECK_LIB(Xinerama, XineramaQueryExtension,
@@ -836,7 +815,6 @@ if test ! x$with_xinerama = xno ; then
             use_xfree_xinerama=no, -lXext $ALL_X_LIBS)
 	AC_MSG_CHECKING(for Xinerama support on XFree86)
 	AC_MSG_RESULT($use_xfree_xinerama);
-  fi
 fi
 
 AC_SUBST(XINERAMA_LIBS)
