--- gir-repository-0.6.3/gir/Makefile.am-orig	2009-08-05 14:39:07.353502000 -0500
+++ gir-repository-0.6.3/gir/Makefile.am	2009-08-05 14:39:30.145429000 -0500
@@ -520,6 +520,7 @@ Gst-0.10.gir: $(INTROSPECTION_SCANNER)
              --include=GModule-2.0 \
              --include=libxml2-2.0 \
 	     --library=gstbase-0.10 \
+             `pkg-config --libs gstreamer-base-0.10` \
 	     --libtool="$(LIBTOOL)" \
              --pkg gobject-2.0 \
              --pkg gstreamer-0.10 \
