--- gmp-4.3.2/configure.in.orig	2010-01-07 21:09:02.000000000 +0100
+++ gmp-4.3.2/configure.in	2010-06-06 01:41:33.120664488 +0200
@@ -1429,7 +1429,7 @@
       geode)                path="x86/k6/k62mmx x86/k6/mmx x86/k6 x86" ;;
       # we don't have any specific 32-bit code for athlon64/opteron, the
       # athlon code should be reasonable
-      athlon | athlon64)    path="x86/k7/mmx x86/k7 x86" ;;
+      athlon | athlon64 | i386)    path="x86/k7/mmx x86/k7 x86" ;;
       i786 | pentium4)      path="x86/pentium4/sse2 x86/pentium4/mmx x86/pentium4 x86" ;;
       # VIA/Centaur processors, sold as CyrixIII and C3.
       viac32)               path="x86/p6/p3mmx x86/p6/mmx x86/p6 x86";;
@@ -1459,7 +1459,7 @@
     fi
 
     case $host_cpu in
-      athlon64 | atom | core2 | pentium4 | x86_64)
+      athlon64 | atom | core2 | pentium4 | x86_64 | i386)
         cclist_64="gcc"
         gcc_64_cflags="$gcc_64_cflags -m64"
         gcc_64_cflags_optlist="cpu"
@@ -1470,13 +1470,17 @@
         case $host in
           *-*-solaris*)
             # Sun cc.
-            cclist_64="$cclist_64 cc"
+            cclist="cc gcc"
+            cclist_64="cc $cclist_64"
             cc_64_cflags="-xO3 -m64"
+            abilist="64 32"
+            path="x86/p6/p3mmx x86/p6/mmx x86/p6 x86"
+            path_64="x86_64"
             ;;
         esac
 
 	case $host_cpu in
-	  athlon64 | x86_64)
+	  athlon64 | x86_64 | i386)
 	    abilist="64 32"
 	    path_64="x86_64"
 	    ;;
