--- gmp-4.3.2/configure.in.orig	2010-01-07 21:09:02.000000000 +0100
+++ gmp-4.3.2/configure.in	2010-06-06 00:51:06.952256588 +0200
@@ -1470,8 +1470,12 @@
         case $host in
           *-*-solaris*)
             # Sun cc.
-            cclist_64="$cclist_64 cc"
+            cclist="cc gcc"
+            cclist_64="cc $cclist_64"
             cc_64_cflags="-xO3 -m64"
+            abilist="64 32"
+            path="x86/p6/p3mmx x86/p6/mmx x86/p6 x86"
+            path_64="x86_64"
             ;;
         esac
 
