--- gnome-shell-3.2.1/js/ui/appDisplay.js-orig	2011-10-21 20:50:09.217339209 -0500
+++ gnome-shell-3.2.1/js/ui/appDisplay.js	2011-10-21 20:50:32.421905047 -0500
@@ -589,17 +589,23 @@ AppWellIcon.prototype = {
 
     _onActivate: function (event) {
         this.emit('launching');
+/*
         let modifiers = Shell.get_event_state(event);
+*/
 
         if (this._onActivateOverride) {
             this._onActivateOverride(event);
         } else {
+/*
             if (modifiers & Clutter.ModifierType.CONTROL_MASK
                 && this.app.state == Shell.AppState.RUNNING) {
                 this.app.open_new_window(-1);
             } else {
-                this.app.activate();
+*/
+        this.app.activate();
+/*
             }
+*/
         }
         Main.overview.hide();
     },
--- gnome-shell-3.2.1/js/ui/viewSelector.js-orig	2011-10-21 20:58:16.038727454 -0500
+++ gnome-shell-3.2.1/js/ui/viewSelector.js	2011-10-21 20:58:48.505813356 -0500
@@ -548,12 +548,15 @@ ViewSelector.prototype = {
     },
 
     _onStageKeyPress: function(actor, event) {
+/*
         let modifiers = Shell.get_event_state(event);
+*/
         let symbol = event.get_key_symbol();
 
         if (symbol == Clutter.Escape) {
             Main.overview.hide();
             return true;
+/*
         } else if (modifiers & Clutter.ModifierType.CONTROL_MASK) {
             if (symbol == Clutter.Page_Up) {
                 if (!this._searchTab.active)
@@ -564,6 +567,7 @@ ViewSelector.prototype = {
                     this._nextTab();
                 return true;
             }
+*/
         } else if (Clutter.keysym_to_unicode(symbol)) {
             this._searchTab.startSearch(event);
         }
