diff --git a/data/Makefile.am b/data/Makefile.am
index 139d7d0..aae022d 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -2,8 +2,11 @@ imagedir = $(pkgdatadir)/images
 
 dist_image_DATA =		\
 	add-workspace.svg	\
-	info.svg			\
-	remove-workspace.svg
+	close.svg		\
+	info.svg		\
+	remove-workspace.svg	\
+	view-more-activated.svg	\
+	view-more.svg
 
 schemadir  = @GCONF_SCHEMA_FILE_DIR@
 schema_DATA = gnome-shell.schemas
