--- a/configure.in	2011-10-01 00:51:47.304372464 -0400
+++ b/configure.in	2011-10-01 00:52:47.686781646 -0400
@@ -49,7 +49,9 @@
 # not 64 bit clean in cross-compile
 AC_CHECK_SIZEOF(void *, 4)
 
+if [ "x$GCC" = xyes ]; then
 CFLAGS='-g -Wall -Wunused -Wmissing-prototypes -Wmissing-declarations -Wstrict-prototypes  -Wmissing-prototypes -Wnested-externs  -Wshadow -Wpointer-arith -Wno-cast-qual -Wcast-align -Wwrite-strings'
+fi
 
 if test "x$enable_maintainer_mode" = "xyes"; then
 DEBUG_FLAGS='-debug'
