--- gnome-shell-2.29.0/src/st/st-table.c-orig	2010-03-10 16:24:29.102886997 -0600
+++ gnome-shell-2.29.0/src/st/st-table.c	2010-03-10 16:26:42.744561326 -0600
@@ -162,7 +162,7 @@ st_container_lower (ClutterContainer *co
                     ClutterActor     *sibling)
 {
   /* XXX: not yet implemented */
-  g_warning ("%s() not yet implemented", __FUNCTION__);
+  g_warning ("%s() not yet implemented", "st_container_lower");
 }
 
 static void
@@ -171,14 +171,14 @@ st_container_raise (ClutterContainer *co
                     ClutterActor     *sibling)
 {
   /* XXX: not yet implemented */
-  g_warning ("%s() not yet implemented", __FUNCTION__);
+  g_warning ("%s() not yet implemented", "st_container_raise");
 }
 
 static void
 st_container_sort_depth_order (ClutterContainer *container)
 {
   /* XXX: not yet implemented */
-  g_warning ("%s() not yet implemented", __FUNCTION__);
+  g_warning ("%s() not yet implemented", "st_container_sort_depth_order");
 }
 
 static void
--- gnome-shell-2.29.0/src/st/st-overflow-box.c-orig	2010-03-10 16:24:35.439180301 -0600
+++ gnome-shell-2.29.0/src/st/st-overflow-box.c	2010-03-10 16:26:18.284184721 -0600
@@ -218,7 +218,7 @@ static void
 st_overflow_box_sort_depth_order (ClutterContainer *container)
 {
   /* XXX: not yet implemented */
-  g_warning ("%s() not yet implemented", __FUNCTION__);
+  g_warning ("%s() not yet implemented", "st_overflow_box_sort_depth_order");
 }
 
 static void
--- gnome-shell-2.29.0/src/st/st-box-layout.c-orig	2010-03-10 16:24:25.227597048 -0600
+++ gnome-shell-2.29.0/src/st/st-box-layout.c	2010-03-10 16:25:34.336030015 -0600
@@ -400,7 +400,7 @@ static void
 st_box_container_sort_depth_order (ClutterContainer *container)
 {
   /* XXX: not yet implemented */
-  g_warning ("%s() not yet implemented", __FUNCTION__);
+  g_warning ("%s() not yet implemented", "st_box_container_sort_depth_order");
 }
 
 static void
