--- gnome-zeitgeist/Makefile-orig	2009-05-12 19:33:39.644793000 -0500
+++ gnome-zeitgeist/Makefile	2009-05-12 19:53:02.426842000 -0500
@@ -1,5 +1,5 @@
-PREFIX = /usr/local
-MO_PREFIX = /usr
+PREFIX = $(DESTDIR)/usr
+MO_PREFIX = $(DESTDIR)/usr
 
 all:
 	@echo "Makefile: Available actions: install, uninstall, clean, tarball, generate-pot, build-translations, install-translations, update-po."
@@ -26,9 +26,9 @@ install:
 	rm -f $(PREFIX)/bin/zeitgeist-daemon \
 		$(PREFIX)/bin/zeitgeist-journal
 	
-	ln -s $(PREFIX)/share/gnome-zeitgeist/src/zeitgeist-daemon.py \
+	ln -s ../share/gnome-zeitgeist/src/zeitgeist-daemon.py \
 		$(PREFIX)/bin/zeitgeist-daemon
-	ln -s $(PREFIX)/share/gnome-zeitgeist/src/zeitgeist_gui/zeitgeist-journal.py \
+	ln -s ../share/gnome-zeitgeist/src/zeitgeist_gui/zeitgeist-journal.py \
 		$(PREFIX)/bin/zeitgeist-journal
 	
 	@echo "Makefile: Installed GNOME Zeitgeist."
