--- gnome-zeitgeist/extra/test-daemon.py-orig	2009-05-12 20:17:36.200981000 -0500
+++ gnome-zeitgeist/extra/test-daemon.py	2009-05-12 20:17:41.154058000 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2.6
 # -.- coding: utf-8 -.-
 
 import sys
--- gnome-zeitgeist/src/zeitgeist_experimental/zeitgeist-timeline.py-orig	2009-05-12 20:15:27.724852000 -0500
+++ gnome-zeitgeist/src/zeitgeist_experimental/zeitgeist-timeline.py	2009-05-12 20:15:34.317458000 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2.6
 # -.- encoding: utf-8 -.-
 
 import sys
--- gnome-zeitgeist/src/zeitgeist_experimental/zeitgeist-projectviewer.py-orig	2009-05-12 20:15:48.872799000 -0500
+++ gnome-zeitgeist/src/zeitgeist_experimental/zeitgeist-projectviewer.py	2009-05-12 20:15:55.424975000 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2.6
 # -.- encoding: utf-8 -.-
 
 import sys
--- gnome-zeitgeist/src/zeitgeist_gui/zeitgeist-journal.py-orig	2009-05-12 20:16:22.091390000 -0500
+++ gnome-zeitgeist/src/zeitgeist_gui/zeitgeist-journal.py	2009-05-12 20:16:33.792044000 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2.6
 # -.- encoding: utf-8 -.-
 
 import sys
--- gnome-zeitgeist/src/zeitgeist_gui/zeitgeist-trayicon.py-orig	2009-05-12 20:16:40.455555000 -0500
+++ gnome-zeitgeist/src/zeitgeist_gui/zeitgeist-trayicon.py	2009-05-12 20:16:45.175481000 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2.6
 # -.- encoding: utf-8 -.-
 
 import sys
--- gnome-zeitgeist/src/zeitgeist-daemon.py-orig	2009-05-12 20:16:52.718170000 -0500
+++ gnome-zeitgeist/src/zeitgeist-daemon.py	2009-05-12 20:16:58.615180000 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2.6
 # -.- encoding: utf-8 -.-
 
 import sys
