--- gnome-shell-2.28.0/src/st/st-table.c-orig	2009-10-11 09:36:17.529894548 -0500
+++ gnome-shell-2.28.0/src/st/st-table.c	2009-10-11 09:36:38.182954612 -0500
@@ -162,7 +162,7 @@ st_container_lower (ClutterContainer *co
                     ClutterActor     *sibling)
 {
   /* XXX: not yet implemented */
-  g_warning ("%s() not yet implemented", __FUNCTION__);
+  g_warning ("%s() not yet implemented", "st_container_lower");
 }
 
 static void
@@ -171,14 +171,14 @@ st_container_raise (ClutterContainer *co
                     ClutterActor     *sibling)
 {
   /* XXX: not yet implemented */
-  g_warning ("%s() not yet implemented", __FUNCTION__);
+  g_warning ("%s() not yet implemented", "st_container_raise");
 }
 
 static void
 st_container_sort_depth_order (ClutterContainer *container)
 {
   /* XXX: not yet implemented */
-  g_warning ("%s() not yet implemented", __FUNCTION__);
+  g_warning ("%s() not yet implemented", "st_container_sort_depth_order");
 }
 
 static void
--- gnome-shell-2.28.0/src/st/st-box-layout.c-orig	2009-10-11 09:34:00.777110008 -0500
+++ gnome-shell-2.28.0/src/st/st-box-layout.c	2009-10-11 09:36:01.756520519 -0500
@@ -295,7 +295,7 @@ st_box_container_lower (ClutterContainer
                         ClutterActor     *sibling)
 {
   /* XXX: not yet implemented */
-  g_warning ("%s() not yet implemented", __FUNCTION__);
+  g_warning ("%s() not yet implemented", "st_box_container_lower");
 }
 
 static void
@@ -304,14 +304,14 @@ st_box_container_raise (ClutterContainer
                         ClutterActor     *sibling)
 {
   /* XXX: not yet implemented */
-  g_warning ("%s() not yet implemented", __FUNCTION__);
+  g_warning ("%s() not yet implemented", "st_box_container_raise");
 }
 
 static void
 st_box_container_sort_depth_order (ClutterContainer *container)
 {
   /* XXX: not yet implemented */
-  g_warning ("%s() not yet implemented", __FUNCTION__);
+  g_warning ("%s() not yet implemented", "st_box_container_sort_depth_order");
 }
 
 static void
