--- gnome-shell/src/gnome-shell.in-orig	2009-04-06 18:20:13.886166000 -0500
+++ gnome-shell/src/gnome-shell.in	2009-04-06 18:20:41.398354000 -0500
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2.6
 
 import atexit
 import optparse
@@ -25,8 +25,7 @@ def find_cmd (cmd_list):
     raise SystemExit("None of the commands %s exist" % cmd_list)
     
 def pidof(command):
-    pidof_cmd = find_cmd(["/sbin/pidof", "/bin/pidof", "/usr/bin/pidof"])
-    pidof = subprocess.Popen([pidof_cmd, command], stdout=subprocess.PIPE)
+    pidof = subprocess.Popen(["/bin/sh", "-c", "/usr/bin/ps -u $USER -o pid,comm | /usr/bin/grep %s | /usr/bin/awk '{print $1}'" % command], stdout=subprocess.PIPE)
     pids = pidof.communicate()[0].split()
     pidof.wait()
 
@@ -183,7 +182,7 @@ def start_shell():
     else:
         args = []
         
-    plugin = os.path.join(plugin_dir, "libgnome-shell.la")
+    plugin = os.path.join(plugin_dir, "libgnome-shell.so")
     args.extend(['metacity', '--mutter-plugins=' + plugin, '--replace'])
     if options.sync:
         args.append('--sync')
