--- gnumeric-1.10.4/plugins/fn-numtheory/numtheory.c	Sat Apr 24 04:31:21 2010
+++ gnumeric-1.10.4-patch/plugins/fn-numtheory/numtheory.c	Wed May 26 11:09:38 2010
@@ -41,1 +41,2 @@
-static const double bit_max = MIN (1 / GNM_EPSILON, (gnm_float)G_MAXUINT64);
+//static const double bit_max = MIN (1 / GNM_EPSILON, (gnm_float)G_MAXUINT64);
+static const double bit_max = G_MAXUINT64;
--- gnumeric-1.10.4/src/graph.c	Wed May 26 12:39:49 2010
+++ gnumeric-1.10.4-patch/src/graph.c	Wed May 26 12:40:00 2010
@@ -56,7 +56,7 @@
 static void
 set_pending_str (const GOData *data, const char *str)
 {
-	return g_object_set_data_full (G_OBJECT (data),
+	g_object_set_data_full (G_OBJECT (data),
 				       "unserialize", g_strdup (str),
 				       g_free);
 }
