--- gobject-introspection-0.6.3/giscanner/dumper.py-orig	2009-08-05 14:35:10.458388000 -0500
+++ gobject-introspection-0.6.3/giscanner/dumper.py	2009-08-05 14:36:04.498009000 -0500
@@ -79,7 +79,7 @@ class DumpCompiler(object):
         self._tmpdir = tempfile.mkdtemp('', 'tmp-introspect')
 
         self._compiler_cmd = os.environ.get('CC', 'gcc')
-        self._linker_cmd = os.environ.get('LD', self._compiler_cmd)
+        self._linker_cmd = os.environ.get('CC', self._compiler_cmd)
         self._pkgconfig_cmd = os.environ.get('PKG_CONFIG', 'pkg-config')
 
         self._uninst_srcdir = os.environ.get(
@@ -207,12 +207,11 @@ class DumpCompiler(object):
         # hack for building GIRepository.gir, skip -lgirepository-1.0 since
         # libgirepository-1.0.la is not in current directory and we refer to it
         # explicitly below anyway
-        if (not uninst_builddir or
-            self._options.libraries[0] != 'girepository-1.0'):
-            # We only use the first library; assume others are "custom"
-            # libraries like from gir-repository.  Right now those don't define
-            # new GTypes, so we don't need to introspect them.
-            args.append('-l' + self._options.libraries[0])
+        for library in self._options.libraries:
+            if (uninst_builddir and
+                self._options.libraries[0] == 'girepository-1.0'):
+                continue
+            args.append('-l' + library)
 
         # hack for building gobject-introspection itself
         if uninst_builddir:
--- gobject-introspection/giscanner/sourcescanner.py-orig	2009-08-05 14:27:41.086228000 -0500
+++ gobject-introspection/giscanner/sourcescanner.py	2009-08-05 14:27:45.935995000 -0500
@@ -250,7 +250,7 @@ class SourceScanner(object):
 
         defines = ['__GI_SCANNER__']
         undefs = []
-        cpp_args = ['gcc', '-E', '-C', '-I.', '-']
+        cpp_args = ['cc', '-E', '-C', '-I.', '-']
 
         cpp_args += self._cpp_options
         proc = subprocess.Popen(cpp_args,
