--- gst-plugins-ugly-0.10.13/ext/x264/gstx264enc.c	Mon Oct 12 12:25:41 2009
+++ gst-plugins-ugly-0.10.13-patch/ext/x264/gstx264enc.c	Sun Apr 18 17:49:25 2010
@@ -582,7 +582,12 @@
   encoder->x264param.analyse.i_noise_reduction = encoder->noise_reduction;
   encoder->x264param.i_frame_reference = encoder->ref;
   encoder->x264param.i_bframe = encoder->bframes;
+#if X264_BUILD < 78
   encoder->x264param.b_bframe_pyramid = encoder->b_pyramid;
+#else
+  encoder->x264param.i_bframe_pyramid =
+      encoder->b_pyramid ? X264_B_PYRAMID_NORMAL : X264_B_PYRAMID_NONE;
+#endif
 #if X264_BUILD < 63
   encoder->x264param.b_bframe_adaptive = encoder->b_adapt;
 #else
