--- gst-plugins-bad-0.10.13.orig/configure.ac	Wed Jun 17 16:10:42 2009
+++ gst-plugins-bad-0.10.13/configure.ac	Wed Sep  2 19:45:12 2009
@@ -523,13 +523,10 @@
 #include <string.h>
 #include <openssl/rand.h>
 ],[
-int main(void)
-{
         char a[2048];
         memset(a, 0, sizeof(a));
         RAND_add(a, sizeof(a), sizeof(a));
         return(RAND_status() <= 0);
-}
                           ],
                           [
                                   found_crypto=1
