--- gst-plugins-bad-0.10.10/gst/deinterlace2/tvtime/greedy.c-orig	2009-01-20 13:19:02.097368000 -0600
+++ gst-plugins-bad-0.10.10/gst/deinterlace2/tvtime/greedy.c	2009-01-20 13:20:44.280125000 -0600
@@ -121,6 +121,8 @@ deinterlace_greedy_packed422_scanline_c 
   }
 }
 
+#undef BUILD_X86_ASM
+
 #ifdef BUILD_X86_ASM
 #include "mmx.h"
 static void
--- gst-plugins-bad-0.10.10/gst/deinterlace2/tvtime/greedyh.c-orig	2009-01-20 13:22:33.820928000 -0600
+++ gst-plugins-bad-0.10.10/gst/deinterlace2/tvtime/greedyh.c	2009-01-20 13:22:41.370980000 -0600
@@ -205,6 +205,7 @@ greedyDScaler_C (GstDeinterlaceMethodGre
   }
 }
 
+#undef BUILD_X86_ASM
 #ifdef BUILD_X86_ASM
 
 #define IS_MMXEXT
--- gst-plugins-bad-0.10.10/gst/deinterlace2/tvtime/vfir.c-orig	2009-01-20 13:22:52.503546000 -0600
+++ gst-plugins-bad-0.10.10/gst/deinterlace2/tvtime/vfir.c	2009-01-20 13:22:58.965446000 -0600
@@ -89,6 +89,7 @@ deinterlace_line_c (GstDeinterlaceMethod
   }
 }
 
+#undef BUILD_X86_ASM
 #ifdef BUILD_X86_ASM
 #include "mmx.h"
 static void
--- gst-plugins-bad-0.10.10/gst/deinterlace2/tvtime/linear.c-orig	2009-01-20 13:23:24.408896000 -0600
+++ gst-plugins-bad-0.10.10/gst/deinterlace2/tvtime/linear.c	2009-01-20 13:23:30.402309000 -0600
@@ -52,6 +52,7 @@ deinterlace_scanline_linear_c (GstDeinte
     out[i] = (scanlines->t0[i] + scanlines->b0[i]) / 2;
 }
 
+#undef BUILD_X86_ASM
 #ifdef BUILD_X86_ASM
 #include "mmx.h"
 static void
--- gst-plugins-bad-0.10.10/gst/deinterlace2/tvtime/linearblend.c-orig	2009-01-20 13:23:39.878640000 -0600
+++ gst-plugins-bad-0.10.10/gst/deinterlace2/tvtime/linearblend.c	2009-01-20 13:23:45.877989000 -0600
@@ -76,6 +76,7 @@ deinterlace_scanline_linear_blend2_c (Gs
   }
 }
 
+#undef BUILD_X86_ASM
 #ifdef BUILD_X86_ASM
 #include "mmx.h"
 static inline void
--- gst-plugins-bad-0.10.10/gst/deinterlace2/tvtime/tomsmocomp.c-orig	2009-01-20 13:23:53.864627000 -0600
+++ gst-plugins-bad-0.10.10/gst/deinterlace2/tvtime/tomsmocomp.c	2009-01-20 13:24:00.062670000 -0600
@@ -79,6 +79,7 @@ Fieldcopy (void *dest, const void *src, 
 #undef  SIMD_TYPE
 #undef  FUNCT_NAME
 
+#undef BUILD_X86_ASM
 #ifdef BUILD_X86_ASM
 
 #include "tomsmocomp/tomsmocompmacros.h"
