--- gst-plugins-bad-0.10.10/gst/dccp/gstdccp.c-orig	2009-01-22 14:47:02.910338000 -0600
+++ gst-plugins-bad-0.10.10/gst/dccp/gstdccp.c	2009-01-22 14:48:49.089313000 -0600
@@ -141,8 +141,6 @@ gst_dccp_read_buffer (GstElement * this,
   iov.iov_len = readsize;
   mh.msg_iov = &iov;
   mh.msg_iovlen = 1;
-  mh.msg_control = NULL;
-  mh.msg_controllen = 0;
 
   bytes_read = recvmsg (socket, &mh, 0);
 
@@ -324,8 +322,6 @@ gst_dccp_socket_write (GstElement * elem
       iov.iov_len = MIN (packet_size, size - bytes_written);
       mh.msg_iov = &iov;
       mh.msg_iovlen = 1;
-      mh.msg_control = NULL;
-      mh.msg_controllen = 0;
 
       wrote = sendmsg (socket, &mh, 0);
     } while (wrote == -1 && errno == EAGAIN);
