--- gst-plugins-bad-0.10.10/configure.ac-orig	2009-01-22 14:36:34.948180000 -0600
+++ gst-plugins-bad-0.10.10/configure.ac	2009-01-22 14:44:13.910072000 -0600
@@ -155,6 +155,10 @@ AM_CONDITIONAL(HAVE_GCC_ASM, test "x$HAV
 
 dnl *** checks for library functions ***
 
+dnl *** checks for socket and nsl libraries ***
+AC_CHECK_FUNC(socket,,[AC_CHECK_LIB(socket,socket)])
+AC_CHECK_FUNC(gethostbyname,,[AC_CHECK_LIB(nsl,gethostbyname)])
+
 dnl *** checks for dependency libraries ***
 
 dnl GLib is required
--- gst-plugins-bad-0.10.10/gst/modplug/Makefile.am-orig	2009-01-22 14:44:35.031703000 -0600
+++ gst-plugins-bad-0.10.10/gst/modplug/Makefile.am	2009-01-22 14:44:47.445448000 -0600
@@ -5,7 +5,7 @@ plugin_LTLIBRARIES = libgstmodplug.la
 libgstmodplug_la_SOURCES = gstmodplug.cc
 libgstmodplug_la_CXXFLAGS = $(GST_PLUGINS_BASE_CFLAGS) $(GST_CXXFLAGS)
 libgstmodplug_la_LIBADD = $(top_builddir)/gst/modplug/libmodplug/libmodplug.la $(GST_PLUGINS_BASE_LIBS) -lstdc++
-libgstmodplug_la_LDFLAGS = $(GST_PLUGIN_LDFLAGS)
+libgstmodplug_la_LDFLAGS = $(GST_PLUGIN_LDFLAGS) $(LIBM)
 libgstmodplug_la_LIBTOOLFLAGS = --tag=disable-static
 
 noinst_HEADERS = gstmodplug.h
