Index: src/main/java/org/apache/hadoop/hbase/mapreduce/TableInputFormatBase.java
===================================================================
--- src/main/java/org/apache/hadoop/hbase/mapreduce/TableInputFormatBase.java   (revision 7444)
+++ src/main/java/org/apache/hadoop/hbase/mapreduce/TableInputFormatBase.java   (working copy)
@@ -124,6 +124,11 @@
     sc.setStopRow(tSplit.getEndRow());
     trr.setScan(sc);
     trr.setHTable(table);
+    try {
+        trr.initialize(tSplit, context);
+    }catch(InterruptedException e){
+        throw new IOException(e);
+    }
     return trr;
   }
