--- iodine-0.6.0-rc1.orig/tests/Makefile	2009-02-10 14:48:42.000000000 -0500
+++ iodine-0.6.0-rc1/tests/Makefile	2011-03-29 01:41:09.175805912 -0400
@@ -1,4 +1,3 @@
-CC = gcc
 TEST = test
 OBJS = test.o base32.o base64.o read.o dns.o encoding.o login.o user.o fw_query.o
 SRCOBJS = ../src/base32.o  ../src/base64.o ../src/read.o ../src/dns.o ../src/encoding.o ../src/login.o ../src/md5.o ../src/user.o ../src/fw_query.o
@@ -7,7 +6,7 @@
 
 CHECK_PATH = /usr/local
 LDFLAGS = -L$(CHECK_PATH)/lib -lcheck `../src/osflags link`
-CFLAGS = -g -Wall -D$(OS) -I../src -I$(CHECK_PATH)/include -pedantic `../src/osflags cflags`
+CFLAGS += -g -D$(OS) -I../src -I$(CHECK_PATH)/include `../src/osflags cflags`
 
 all: $(TEST)
 	@LD_LIBRARY_PATH=${CHECK_PATH}/lib ./$(TEST)
--- iodine-0.6.0-rc1.orig/src/Makefile	2009-12-29 15:25:33.000000000 -0500
+++ iodine-0.6.0-rc1/src/Makefile	2011-03-29 01:40:23.842854773 -0400
@@ -9,7 +9,7 @@
 
 LIBPATH = -L.
 LDFLAGS +=  -lz `sh osflags $(TARGETOS) link` $(LIBPATH)
-CFLAGS += -c -g -Wall -D$(OS) -pedantic `sh osflags $(TARGETOS) cflags`
+CFLAGS += -c -g -D$(OS) `sh osflags $(TARGETOS) cflags`
 
 all: stateos $(CLIENT) $(SERVER)
 
