--- json-glib-0.6.2.orig/configure.ac	2008-05-20 04:55:28.000000000 -0400
+++ json-glib-0.6.2/configure.ac	2009-06-26 13:59:18.871164803 -0400
@@ -98,10 +98,12 @@
                              [Use strict compiler flags @<:@default=maintainer_flags_default@:>@]),,
               enable_maintainer_flags=maintainer_flags_default)
 
-if test "x$enable_maintainer_flags" = "xyes"; then
-  CPPFLAGS="$CPPFLAGS -g -Wall -Wshadow -Wcast-align -Wno-uninitialized -Werror"
-else
-  CPPFLAGS="$CPPFLAGS -g -Wall"
+if test "$xGCC" = "xyes"; then
+  if test "x$enable_maintainer_flags" = "xyes"; then
+    CPPFLAGS="$CPPFLAGS -g -Wall -Wshadow -Wcast-align -Wno-uninitialized -Werror"
+  else
+    CPPFLAGS="$CPPFLAGS -g -Wall"
+  fi
 fi
 
 GTK_DOC_CHECK([1.8])
