diff --git a/champlain-gtk/Makefile.am b/champlain-gtk/Makefile.am
index 343c5f1..f382624 100644
--- a/champlain-gtk/Makefile.am
+++ b/champlain-gtk/Makefile.am
@@ -46,7 +46,7 @@ GtkChamplain-0.3.gir: $(INTROSPECTION_SCANNER) libchamplain-gtk-0.3.la
 	$(INCLUDES) \
 	--add-include-path=$(srcdir) --add-include=path=. \
 	--c-include="$(libchamplain_include_HEADERS)" \
-	--include=Clutter-0.8 \
+	--include=Clutter-1.0 \
 	--include=Gtk-2.0 \
 	--library=champlain-gtk-0.3 \
 	--output $@ \
diff --git a/champlain/Makefile.am b/champlain/Makefile.am
index 330807c..ab414f7 100644
--- a/champlain/Makefile.am
+++ b/champlain/Makefile.am
@@ -167,7 +167,7 @@ Champlain-0.3.gir: $(INTROSPECTION_SCANNER) libchamplain-0.3.la
 	$(INCLUDES) \
 	--add-include-path=$(srcdir) --add-include=path=. \
 	--c-include="$(libchamplain_include_HEADERS)" \
-	--include=Clutter-0.8 \
+	--include=Clutter-1.0 \
 	--include=Gtk-2.0 \
 	--library=champlain-0.3 \
 	--output $@ \
