--- libdvbpsi-0.2.2/configure	Thu Nov  3 16:55:01 2011
+++ libdvbpsi-0.2.2/configure	Sat Aug 18 22:27:54 2012
@@ -11588,2 +11588,1 @@
-CFLAGS="${CFLAGS} -Wall -DDVBPSI_DIST --std=gnu99"
-CFLAGS="${CFLAGS} -Wpointer-arith -Wcast-align -Wcast-qual -Wstrict-prototypes -Wshadow -Waggregate-return -Wmissing-prototypes -Wnested-externs -Wsign-compare"
+CFLAGS="${CFLAGS} -DDVBPSI_DIST"
@@ -11658,1 +11657,1 @@
-  CFLAGS="${CFLAGS} -O6"
+  CFLAGS="${CFLAGS} -O3"
