--- libebml-1.2.0/src/EbmlBinary.cpp.orig	Thu Aug 19 08:01:28 2010
+++ libebml-1.2.0/src/EbmlBinary.cpp	Sat Feb  5 15:52:23 2011
@@ -35,6 +35,7 @@
 	\author Julien Coloos	<suiryc @ users.sf.net>
 */
 #include <cassert>
+#include <string>
 
 #include "ebml/EbmlBinary.h"
 #include "ebml/StdIOCallback.h"
@@ -93,7 +94,7 @@
 
 	Data = (binary *)malloc(GetSize() * sizeof(binary));
     if (Data == NULL)
-        throw CRTError("Error allocating data");
+        throw CRTError(std::string("Error allocating data"));
 	SetValueIsSet();
 	return input.read(Data, GetSize());
 }
