--- libfm-0.1.14/configure.ac-orig	2010-08-27 00:22:58.277160931 -0500
+++ libfm-0.1.14/configure.ac	2010-08-27 00:23:21.588493826 -0500
@@ -3,7 +3,7 @@
 
 AC_PREREQ([2.63])
 AC_INIT([libfm], [0.1.17], [http://pcmanfm.sorceforge.net/])
-AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+AM_INIT_AUTOMAKE()
 AC_CONFIG_MACRO_DIR(m4)
 AC_CONFIG_HEADERS([config.h])
 
--- libfm-0.1.12/src/Makefile.am-orig	2010-08-27 00:38:47.380412379 -0500
+++ libfm-0.1.12/src/Makefile.am	2010-08-27 00:38:59.923612592 -0500
@@ -185,7 +185,6 @@ libfm_la_CFLAGS = \
 	$(GIO_CFLAGS) \
 	$(MENU_CACHE_CFLAGS) \
 	$(DBUS_CFLAGS) \
-	-Werror-implicit-function-declaration \
 	$(NULL)
 
 libfm_la_LIBADD = \
@@ -211,8 +210,7 @@ libfm_gtk_la_CFLAGS = \
 	$(MENU_CACHE_CFLAGS) \
 	-I$(srcdir) \
 	-I$(srcdir)/base \
-	-I$(srcdir)/job \
-	-Werror-implicit-function-declaration
+	-I$(srcdir)/job
 
 libfm_gtk_la_LIBADD = \
 	$(GTK_LIBS) \
@@ -237,7 +235,6 @@ libfm_pref_apps_CFLAGS = \
 	$(GTK_CFLAGS) \
 	$(GIO_CFLAGS) \
 	$(MENU_CACHE_CFLAGS) \
-	-Werror-implicit-function-declaration \
 	$(NULL)
 
 libfm_pref_apps_LDADD = \
@@ -258,7 +255,6 @@ libfm_demo_SOURCES = \
 libfm_demo_CFLAGS = \
 	$(GTK_CFLAGS) \
 	$(MENU_CACHE_CFLAGS) \
-	-Werror-implicit-function-declaration \
 	$(NULL)
 
 libfm_demo_LDADD = \
@@ -285,7 +281,6 @@ libgiofm_la_CFLAGS = \
 	$(GIO_CFLAGS) \
 	-DGIO_MODULE_DIR=\"$(giomodulesdir)\"	\
 	-DGVFS_LOCALEDIR=\""$(localedir)"\"	\
-	-DG_DISABLE_DEPRECATED \
 	$(NULL)
 
 libgiofm_la_LDFLAGS = 	\
--- libfm-0.1.14/src/tests/Makefile.am-orig	2011-06-08 11:21:16.760136885 -0500
+++ libfm-0.1.14/src/tests/Makefile.am	2011-06-08 11:21:24.260181393 -0500
@@ -10,7 +10,6 @@ AM_CPPFLAGS = \
 	$(GIO_CFLAGS) \
 	$(MENU_CACHE_CFLAGS) \
 	$(DBUS_CFLAGS) \
-	-Werror-implicit-function-declaration \
 	$(NULL)
 
 noinst_PROGRAMS = $(TEST_PROGS)
