diff --git a/libgda/gda-sql-builder.c b/libgda/gda-sql-builder.c
index 8f4a278..a7a4c3f 100644
--- a/libgda/gda-sql-builder.c
+++ b/libgda/gda-sql-builder.c
@@ -62,6 +62,7 @@ static GObjectClass *parent_class = NULL;
 
 /* signals */
 enum {
+	DUMMY,
 	LAST_SIGNAL
 };
 
diff --git a/libgda/gda-tree-manager.c b/libgda/gda-tree-manager.c
index 98edc44..f1da76f 100644
--- a/libgda/gda-tree-manager.c
+++ b/libgda/gda-tree-manager.c
@@ -43,6 +43,7 @@ static void gda_tree_manager_get_property (GObject *object,
 					   GParamSpec *pspec);
 
 enum {
+	DUMMY,
 	LAST_SIGNAL
 };
 
diff --git a/libgda/gda-tree-node.c b/libgda/gda-tree-node.c
index 6eaf64a..c70dc9c 100644
--- a/libgda/gda-tree-node.c
+++ b/libgda/gda-tree-node.c
@@ -65,6 +65,7 @@ static gchar *gda_tree_node_dump_header (GdaTreeNode *node);
 static void gda_tree_node_dump_children (GdaTreeNode *node, const gchar *prefix, GString *in_string);
 
 enum {
+	DUMMY,
 	LAST_SIGNAL
 };
 
diff --git a/libgda/gda-tree.c b/libgda/gda-tree.c
index cc5a72d..a4d2f25 100644
--- a/libgda/gda-tree.c
+++ b/libgda/gda-tree.c
@@ -51,6 +51,7 @@ static void gda_tree_get_property (GObject *object,
 				   GParamSpec *pspec);
 
 enum {
+	DUMMY,
 	LAST_SIGNAL
 };
 
diff --git a/libgda/thread-wrapper/gda-thread-wrapper.c b/libgda/thread-wrapper/gda-thread-wrapper.c
index c6f7dd7..d10e87b 100644
--- a/libgda/thread-wrapper/gda-thread-wrapper.c
+++ b/libgda/thread-wrapper/gda-thread-wrapper.c
@@ -273,6 +273,7 @@ static void gda_thread_wrapper_get_property (GObject *object,
 					     GParamSpec *pspec);
 
 enum {
+	DUMMY,
 	LAST_SIGNAL
 };
 
