Index: trunk/libgda/sqlite/gda-sqlite-provider.c
===================================================================
--- trunk/libgda/sqlite/gda-sqlite-provider.c	(revision 3279)
+++ trunk/libgda/sqlite/gda-sqlite-provider.c	(revision 3286)
@@ -1108,7 +1108,7 @@
 
 			if (g_unlink (filename)) {
 				g_set_error (error, GDA_SERVER_PROVIDER_ERROR, GDA_SERVER_PROVIDER_OPERATION_ERROR,
-					      "%s", sys_errlist [errno]);
+					      "%s", g_strerror (errno));
 				retval = FALSE;
 			}
 			g_free (filename);
