--- libmapi-0.8.2-ROMULUS/libmapi/socket/netif.c.orig	2009-08-03 15:04:09.741276000 +0800
+++ libmapi-0.8.2-ROMULUS/libmapi/socket/netif.c	2009-08-03 15:04:25.573721000 +0800
@@ -29,6 +29,7 @@
 
 */
 
+#include <sys/sockio.h>
 #include <libmapi/libmapi.h>
 
 #ifdef __COMPAR_FN_T
--- libmapi-0.8.2-ROMULUS/libmapi/x500.c.orig	2009-08-03 15:04:50.606470000 +0800
+++ libmapi-0.8.2-ROMULUS/libmapi/x500.c	2009-08-03 15:05:27.597100000 +0800
@@ -19,6 +19,20 @@
 
 #include <libmapi/libmapi.h>
 
+char *strcasestr(const char *haystack, const char *needle)
+{
+       const char *s;
+       size_t nlen = strlen(needle);
+
+       for (s=haystack;*s;s++) {
+               if (toupper(*needle) == toupper(*s) &&
+                               strncasecmp(s, needle, nlen) == 0) {
+                       return (char *)((uintptr_t)s);
+               }
+       }
+       return NULL;
+}
+
 
 /**
    \details Extract a DN element from a given DN
--- libmapi-0.8.2-ROMULUS/utils/mapiprofile.c.orig	2009-08-03 15:16:09.418864000 +0800
+++ libmapi-0.8.2-ROMULUS/utils/mapiprofile.c	2009-08-03 15:16:39.034405000 +0800
@@ -114,6 +114,9 @@
 	/* catch CTRL-C */
 	g_profname = profname;
 
+        typedef void (*sighandler_t)(int signum);
+
+
 #if defined (__FreeBSD__)
 	(void) signal(SIGINT, (sig_t) signal_delete_profile);
 #else
--- libmapi-0.8.2-ROMULUS/utils/openchangeclient.c.orig	2009-08-03 15:16:57.532055000 +0800
+++ libmapi-0.8.2-ROMULUS/utils/openchangeclient.c	2009-08-03 15:17:43.359540000 +0800
@@ -216,7 +216,7 @@
 		oclient->attach[oclient->attach_num].filename = talloc_strdup(mem_ctx, filename);
 		oclient->attach[oclient->attach_num].bin.lpb = talloc_size(mem_ctx, sb.st_size);
 		oclient->attach[oclient->attach_num].bin.cb = sb.st_size;
-		if ((oclient->attach[oclient->attach_num].bin.lpb = mmap(NULL, sb.st_size, PROT_READ, MAP_FILE|MAP_SHARED, fd, 0)) == (void *) -1) {
+		if ((oclient->attach[oclient->attach_num].bin.lpb = mmap(NULL, sb.st_size, PROT_READ, MAP_SHARED, fd, 0)) == (void *) -1) {
 			perror("mmap");
 			close(fd);
 			return false;
--- libmapi-0.8.2-ROMULUS/mapiproxy/dcesrv_mapiproxy_unused.c.orig	2009-08-03 15:09:21.459081000 +0800
+++ libmapi-0.8.2-ROMULUS/mapiproxy/dcesrv_mapiproxy_unused.c	2009-08-03 15:09:28.306773000 +0800
@@ -44,7 +44,6 @@
 #include "mapiproxy/dcesrv_mapiproxy_proto.h"
 
 #include <sys/types.h>
-#include <sys/cdefs.h>
 
 /*
    endpoint server for the exchange_store_admin3 pipe
--- libmapi-0.8.2-ROMULUS/configure.ac.orig	2009-08-03 15:03:33.068283000 +0800
+++ libmapi-0.8.2-ROMULUS/configure.ac	2009-08-03 15:03:48.108308000 +0800
@@ -218,6 +218,7 @@
 dnl Samba4 modules
 dnl ---------------------------------------------------------------------------
 PKG_CHECK_MODULES(TALLOC, talloc)
+PKG_CHECK_MODULES(TEVENT, tevent)
 PKG_CHECK_MODULES(SAMBA, dcerpc ndr samba-hostconfig)
 PKG_CHECK_MODULES(LDB, ldb)
 
