--- libmapi-0.8.2-ROMULUS/torture/exchange_createuser.c.old1	2009-12-02 16:25:33.979454000 +0800
+++ libmapi-0.8.2-ROMULUS/torture/exchange_createuser.c	2009-12-02 16:25:49.977041000 +0800
@@ -105,7 +105,7 @@
 	/* open LDAP connection */
 	remote_ldb_url = talloc_asprintf(mem_ctx, "ldap://%s", profile->server);
 	remote_ldb = ldb_wrap_connect(mem_ctx, ev, global_mapi_ctx->lp_ctx, remote_ldb_url, 
-								  NULL, cmdline_credentials, 0, NULL);
+								  NULL, cmdline_credentials, 0);
 	if (!remote_ldb) return NT_STATUS_UNSUCCESSFUL;
 
 	/* search the user's record using the user dom_sid */
--- libmapi-0.8.2-ROMULUS/libmapiadmin/mapiadmin_user.c.old1	2009-12-02 16:24:56.977710000 +0800
+++ libmapi-0.8.2-ROMULUS/libmapiadmin/mapiadmin_user.c	2009-12-02 16:25:23.896512000 +0800
@@ -228,7 +228,7 @@
 	remote_ldb_url = talloc_asprintf(mem_ctx, "ldap://%s", profile->server);
 	MAPI_RETVAL_IF(!remote_ldb_url, MAPI_E_CORRUPT_DATA, mem_ctx);
 	remote_ldb = ldb_wrap_connect(mem_ctx, ev, global_mapi_ctx->lp_ctx, remote_ldb_url, 
-				      NULL, mapiadmin_ctx->session->profile->credentials, 0, NULL);
+				      NULL, mapiadmin_ctx->session->profile->credentials, 0);
 	MAPI_RETVAL_IF(!remote_ldb, MAPI_E_NETWORK_ERROR, mem_ctx);
 
 	/* Search the user_dn */
--- libmapi-0.8.2-ROMULUS/mapiproxy/dcesrv_mapiproxy.c.old1	2009-12-02 16:26:12.349182000 +0800
+++ libmapi-0.8.2-ROMULUS/mapiproxy/dcesrv_mapiproxy.c	2009-12-02 16:27:24.095882000 +0800
@@ -142,7 +142,7 @@
 		if (!NT_STATUS_IS_OK(status)) {
 			return status;
 		}
-		dce_call->context->assoc_group_id = private->c_pipe->assoc_group_id;
+		// dce_call->context->assoc_group_id = private->c_pipe->assoc_group_id;
  		
 	} else {
 		status = dcerpc_pipe_connect(dce_call->context,
@@ -154,7 +154,7 @@
 		if (!NT_STATUS_IS_OK(status)) {
 			return status;
 		}
-		dce_call->context->assoc_group_id = private->c_pipe->assoc_group_id;
+		//dce_call->context->assoc_group_id = private->c_pipe->assoc_group_id;
 	}
 
 	DEBUG(5, ("dcerpc_mapiproxy: RPC proxy: CONNECTED\n"));
