--- libmapi-0.8.2-ROMULUS/ndr_mapi.c.old	2009-10-30 16:25:24.078770000 +0800
+++ libmapi-0.8.2-ROMULUS/ndr_mapi.c	2009-10-30 16:25:47.981192000 +0800
@@ -501,7 +501,8 @@
 
 void ndr_print_mapi_SPropValue_wrap(struct ndr_print *ndr, const char *name, const struct mapi_SPropValue_wrap *r)
 {
-	return ndr_print_mapi_SPropValue(ndr, name, (const struct mapi_SPropValue *)r);
+	ndr_print_mapi_SPropValue(ndr, name, (const struct mapi_SPropValue *)r);
+        return;
 }
 
 
@@ -522,7 +523,8 @@
 
 void ndr_print_mapi_SPropValue_array_wrap(struct ndr_print *ndr, const char *name, const struct mapi_SPropValue_array_wrap *r)
 {
-	return ndr_print_mapi_SPropValue_array(ndr, name, (const struct mapi_SPropValue_array *)r);
+	ndr_print_mapi_SPropValue_array(ndr, name, (const struct mapi_SPropValue_array *)r);
+        return;
 }
 
 enum ndr_err_code ndr_push_RestrictionVariable(struct ndr_push *ndr, int ndr_flags, const union RestrictionVariable *r)
