--- libmapi-0.8.2-ROMULUS/config.mk.in.old	2009-12-02 16:20:25.568642000 +0800
+++ libmapi-0.8.2-ROMULUS/config.mk.in	2009-12-02 16:21:30.783148000 +0800
@@ -29,8 +29,7 @@
 sambaprefix=@sambaprefix@
 
 DSOOPT=-shared -fPIC
-CFLAGS=@CFLAGS@ -I. -O3 -Wall -Wmissing-prototypes -Wstrict-prototypes -g3 -Wp,-D_FORTIFY_SOURCE=2	\
-	-fstrict-aliasing -Wstrict-aliasing=3	   		       	   				\
+CFLAGS=@CFLAGS@ -I. -g -Wp,-D_FORTIFY_SOURCE=2	\
 	   -DDEFAULT_LDIF=\"$(datadir)/setup/profiles\"							\
 	   -DMAPISTORE_BACKEND_INSTALLDIR=\"$(libdir)/mapistore_backends\"				\
 	   -DMAPISTORE_MAPPING_PATH=\"$(prefix)/private/mapistore\"	
@@ -40,7 +39,7 @@
 PACKAGE_VERSION=@PACKAGE_VERSION@
 
 # Portability hack...
-CFLAGS+=-Duint_t="unsigned int" 
+#CFLAGS+=-Duint_t="unsigned int" 
 
 CFLAGS+=@SAMBA_CFLAGS@ @LDB_CFLAGS@ @TALLOC_CFLAGS@
 LIBS+=@SAMBA_LIBS@ @LDB_LIBS@ @TALLOC_LIBS@
