--- libquicktime-1.1.3/configure.ac	Mon Jul 13 22:32:03 2009
+++ libquicktime-1.1.3-patch/configure.ac	Fri Aug 21 15:27:49 2009
@@ -831,10 +831,6 @@
 dnl
 
 COMMON_CFLAGS=""
-lqt_test_cflags="-finline-functions -Wall -Winline -Wmissing-declarations -Wdeclaration-after-statement"
-for i in $lqt_test_cflags; do
-  LQT_TRY_CFLAGS($i, COMMON_CFLAGS="$COMMON_CFLAGS $i") ;
-done
 
 dnl
 dnl GCC Visibility
--- libquicktime-1.2.3/configure	Sun Nov 20 14:24:06 2011
+++ libquicktime-1.2.3-patch/configure	Sun Nov 20 14:24:26 2011
@@ -17747,41 +17747,8 @@
 
 
 COMMON_CFLAGS=""
-lqt_test_cflags="-finline-functions -Wall -Winline -Wmissing-declarations -Wdeclaration-after-statement"
-for i in $lqt_test_cflags; do
-  { $as_echo "$as_me:${as_lineno-$LINENO}: checking if $CC supports $i flags" >&5
-$as_echo_n "checking if $CC supports $i flags... " >&6; }
-    SAVE_CFLAGS="$CFLAGS"
-    CFLAGS="$i"
-    cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
 
-int
-main ()
-{
 
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_compile "$LINENO"; then :
-  lqt_try_cflags_ok=yes
-else
-  lqt_try_cflags_ok=no
-fi
-rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
-    CFLAGS="$SAVE_CFLAGS"
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: $lqt_try_cflags_ok" >&5
-$as_echo "$lqt_try_cflags_ok" >&6; }
-    if test x"$lqt_try_cflags_ok" = x"yes"; then
-        COMMON_CFLAGS="$COMMON_CFLAGS $i"
-    else
-        :
-    fi ;
-done
-
-
-
 
 
 # Check whether --with-visibility was given.
