--- rest-0.6/rest/Makefile.am-orig	2009-10-11 12:30:57.211009503 -0500
+++ rest-0.6/rest/Makefile.am	2009-10-11 12:31:02.105573201 -0500
@@ -3,7 +3,7 @@ lib_LTLIBRARIES = librest.la
 librest_la_CFLAGS = $(GLIB_CFLAGS) $(GTHREAD_CFLAGS) \
 		    $(SOUP_CFLAGS) $(SOUP_GNOME_CFLAGS) \
 		    $(XML_CFLAGS) \
-		    -I$(top_srcdir) -Wall -DG_LOG_DOMAIN=\"Rest\"
+		    -I$(top_srcdir) -DG_LOG_DOMAIN=\"Rest\"
 librest_la_LIBADD = $(GLIB_LIBS) $(GTHREAD_LIBS) \
                     $(SOUP_LIBS) $(SOUP_GNOME_LIBS) $(XML_LIBS)
 librest_la_SOURCES = rest-proxy.c 	\
