--- rest-0.7.10/rest/Makefile.in	Thu Mar 31 23:12:11 2011
+++ rest-0.7.10/rest/Makefile.in	Sat Oct 29 16:48:04 2011
@@ -351,1 +351,1 @@
-		    -I$(top_srcdir) -Wall -DG_LOG_DOMAIN=\"Rest\"
+		    -I$(top_srcdir) -DG_LOG_DOMAIN=\"Rest\"
--- rest-0.7.10/rest-extras/Makefile.in	Sat Oct 29 16:50:56 2011
+++ rest-0.7.10/rest-extras/Makefile.in	Sat Oct 29 16:51:01 2011
@@ -322,1 +322,1 @@
-		    -I$(top_srcdir) -Wall -DG_LOG_DOMAIN=\"Rest\"
+		    -I$(top_srcdir) -DG_LOG_DOMAIN=\"Rest\"
