--- x264-snapshot-20101029-2245/configure.orig	Fri Oct 29 16:45:07 2010
+++ x264-snapshot-20101029-2245/configure	Mon Nov  1 15:54:31 2010
@@ -761,7 +761,7 @@
     elif [ "$SYS" = "SunOS" ]; then
         echo "SOSUFFIX=so" >> config.mak
         echo "SONAME=libx264.so.$API" >> config.mak
-        echo 'SOFLAGS=-Wl,-h,$(SONAME)' >> config.mak
+        echo 'SOFLAGS=-Wl,-h$(SONAME)' >> config.mak
     else
         echo "SOSUFFIX=so" >> config.mak
         echo "SONAME=libx264.so.$API" >> config.mak
