--- x264-snapshot-20101029-2245/Makefile.orig	Fri Oct 29 16:45:07 2010
+++ x264-snapshot-20101029-2245/Makefile	Fri Oct 29 22:46:48 2010
@@ -133,7 +133,7 @@
 	$(RANLIB) libx264.a
 
 $(SONAME): .depend $(OBJS) $(OBJASM) $(OBJSO)
-	$(CC) -shared -o $@ $(OBJS) $(OBJASM) $(OBJSO) $(SOFLAGS) $(LDFLAGS)
+	ld -G -B dynamic -o $@ $(OBJS) $(OBJASM) $(OBJSO) $(SOFLAGS) $(LDFLAGS)
 
 x264$(EXE): $(OBJCLI) libx264.a
 	$(CC) -o $@ $+ $(LDFLAGSCLI) $(LDFLAGS)
