--- x264-snapshot-20110827-2245/version.sh	2011-09-01 20:19:22.418374638 +0200
+++ x264-snapshot-20110827-2245/version.sh	2011-09-01 20:18:52.652811257 +0200
@@ -1,20 +1,5 @@
 #!/bin/bash
-git rev-list HEAD | sort > config.git-hash
-LOCALVER=`wc -l config.git-hash | awk '{print $1}'`
-if [ $LOCALVER \> 1 ] ; then
-    VER=`git rev-list origin/master | sort | join config.git-hash - | wc -l | awk '{print $1}'`
-    if [ $VER != $LOCALVER ] ; then
-        VER="$VER+$(($LOCALVER-$VER))"
-    fi
-    if git status | grep -q "modified:" ; then
-        VER="${VER}M"
-    fi
-    VER="$VER $(git rev-list HEAD -n 1 | cut -c 1-7)"
-    echo "#define X264_VERSION \" r$VER\""
-else
+VER="git-2245M"
     echo "#define X264_VERSION \"\""
-    VER="x"
-fi
-rm -f config.git-hash
 API=`grep '#define X264_BUILD' < x264.h | sed -e 's/.* \([1-9][0-9]*\).*/\1/'`
 echo "#define X264_POINTVER \"0.$API.$VER\""
