--- x264-snapshot-20110427-2245/configure.orig	2011-04-28 21:25:06.796185861 +0100
+++ x264-snapshot-20110427-2245/configure	2011-04-28 21:30:39.842496844 +0100
@@ -782,7 +782,7 @@
 
 if [ "$gpac" = "auto" ] ; then
     gpac="no"
-    cc_check "" -lz && GPAC_LIBS="-lgpac_static -lz" || GPAC_LIBS="-lgpac_static"
+    cc_check "" -lz && GPAC_LIBS="-lgpac -lz" || GPAC_LIBS="-lgpac"
     if [ "$SYS" = "WINDOWS" ] ; then
         GPAC_LIBS="$GPAC_LIBS -lwinmm"
     fi
