--- x264-snapshot-20090208-2245/Makefile	ne úno  8 22:45:10 2009
+++ x264-snapshot-20090208-2245-patch/Makefile	so úno 14 17:52:59 2009
@@ -78,7 +78,7 @@
 	$(RANLIB) libx264.a
 
 $(SONAME): .depend $(OBJS) $(OBJASM)
-	$(CC) -shared -o $@ $(OBJS) $(OBJASM) $(SOFLAGS) $(LDFLAGS)
+	ld -G -B dynamic -o $@ $(OBJS) $(OBJASM) $(SOFLAGS) $(LDFLAGS)
 
 x264$(EXE): $(OBJCLI) libx264.a 
 	$(CC) -o $@ $+ $(LDFLAGS)
