--- lightdm-0.3.4/liblightdm-gobject/language.c-orig	2011-05-20 08:33:55.971918119 -0500
+++ lightdm-0.3.4/liblightdm-gobject/language.c	2011-05-20 08:34:49.135575430 -0500
@@ -76,7 +76,8 @@ ldm_language_get_name (LdmLanguage *lang
     {
         char *current = setlocale(LC_ALL, NULL);
         setlocale(LC_ALL, language->priv->code);
-        language->priv->name = g_strdup (nl_langinfo (_NL_IDENTIFICATION_LANGUAGE));
+        //language->priv->name = g_strdup (nl_langinfo (_NL_IDENTIFICATION_LANGUAGE));
+        language->priv->name = g_strdup_printf ("%s", language->priv->code);
         setlocale(LC_ALL, current);
     }
 
@@ -100,7 +101,8 @@ ldm_language_get_territory (LdmLanguage 
     {
         char *current = setlocale(LC_ALL, NULL);
         setlocale(LC_ALL, language->priv->code);
-        language->priv->territory = g_strdup (nl_langinfo (_NL_IDENTIFICATION_TERRITORY));
+        //language->priv->territory = g_strdup (nl_langinfo (_NL_IDENTIFICATION_TERRITORY));
+        language->priv->territory = g_strdup_printf ("%s", language->priv->code);
         setlocale(LC_ALL, current);
     }
 
--- lightdm-0.3.4/greeters/Makefile.am-orig	2011-05-20 08:38:34.654758738 -0500
+++ lightdm-0.3.4/greeters/Makefile.am	2011-05-20 08:38:39.762124030 -0500
@@ -4,9 +4,9 @@ if HAVE_GTK_GREETER
 SUBDIRS += gtk
 endif
 
-if HAVE_VALA_GTK_GREETER
-SUBDIRS += vala-gtk
-endif
+#if HAVE_VALA_GTK_GREETER
+#SUBDIRS += vala-gtk
+#endif
 
 if HAVE_QT_GREETER
 SUBDIRS += qt
