--- lightdm-0.3.4/src/display-manager.c-orig	2011-05-20 08:39:17.181716022 -0500
+++ lightdm-0.3.4/src/display-manager.c	2011-05-20 08:45:53.818122932 -0500
@@ -9,6 +9,14 @@
  * license.
  */
 
+#if _sun
+#define LIGHTDM_USE_CONSIO_VT
+#define LIGHTDM_CONSOLEDEVICE "/dev/vt/0"
+#else
+#define LIGHTDM_USE_SYS_VT
+#define LIGHTDM_CONSOLEDEVICE "/dev/console"
+#endif
+
 #include <config.h>
 
 #include <stdlib.h>
@@ -19,7 +27,12 @@
 #include <fcntl.h>
 #include <glib/gstdio.h>
 #include <sys/ioctl.h>
-#include <linux/vt.h>
+
+#if defined (LIGHTDM_USE_SYS_VT)
+#include <sys/vt.h>
+#elif defined (LIGHDM_USE_CONSIO_VT)
+#include <sys/consio.h>
+#endif
 
 #include "display-manager.h"
 #include "xdmcp-server.h"
@@ -250,6 +263,14 @@ string_to_xdm_auth_key (const gchar *key
     }
 }
 
+#if defined (LIGHTDM_USE_SYS_VT)
+static gint
+get_vt (DisplayManager *manager, gchar *config_section)
+{
+}
+
+#elif defined (LIGHTDM_USE_CONSIO_VT)
+
 static gint
 get_vt (DisplayManager *manager, gchar *config_section)
 {
@@ -264,7 +285,7 @@ get_vt (DisplayManager *manager, gchar *
     if (tty)
         return atoi (tty);
 
-    console_fd = g_open ("/dev/console", O_RDONLY | O_NOCTTY);
+    console_fd = g_open (LIGHTDM_CONSOLEDEVICE, O_RDONLY | O_NOCTTY);
     if (console_fd < 0)
         return -1;
 
@@ -273,6 +294,7 @@ get_vt (DisplayManager *manager, gchar *
 
     return number;  
 }
+#endif
 
 static XServer *
 make_xserver (DisplayManager *manager, gchar *config_section)
