diff -Naur lives-1.1.2/src/keyboard.c lives-1.1.2-patch/src/keyboard.c
--- lives-1.1.2/src/keyboard.c	2009-09-05 14:36:30.000000000 +0200
+++ lives-1.1.2-patch/src/keyboard.c	2009-09-17 16:38:38.047154107 +0200
@@ -85,7 +85,9 @@
 
 #ifdef OMC_MIDI_IMPL
   midi_check_rate=prefs->midi_check_rate;
+#ifdef ALSA_MIDI
   if (prefs->use_alsa_midi) midi_check_rate=1; // because we loop for events in midi_mangle()
+#endif
 
   if (mainw->ext_cntl[EXT_CNTL_MIDI]) {
     do {
@@ -96,7 +98,9 @@
 	  omc_process_string(OMC_MIDI,string,FALSE,NULL);
 	  g_free(string);
 	  string=NULL;
+#ifdef ALSA_MIDI
 	  if (prefs->use_alsa_midi) gotone=TRUE;
+#endif
 	}
       }
     } while (gotone);
diff -Naur lives-1.1.2/src/preferences.c lives-1.1.2-patch/src/preferences.c
--- lives-1.1.2/src/preferences.c	2009-09-05 14:36:30.000000000 +0200
+++ lives-1.1.2-patch/src/preferences.c	2009-09-17 16:36:30.977414402 +0200
@@ -1259,8 +1259,8 @@
   GtkWidget *eventbox;
 
 #ifdef ENABLE_OSC
-#ifdef OMC_JS_IMPL
   GtkWidget *buttond;
+#ifdef OMC_JS_IMPL
   GtkWidget *vbox2;
 #else
 #ifdef OMC_MIDI_IMPL
@@ -3052,8 +3052,9 @@
    gtk_box_pack_start (GTK_BOX (hbox), raw_midi_button, TRUE, TRUE, 20);
    gtk_tooltips_set_tip (mainw->tooltips, raw_midi_button, (_("Read directly from the MIDI device")), NULL);
 
+#ifdef ALSA_MIDI
    gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (raw_midi_button),!prefs->use_alsa_midi);
-
+#endif
 
    hbox = gtk_hbox_new (FALSE, 0);
    gtk_box_pack_start (GTK_BOX (vbox), hbox, TRUE, TRUE, 0);
