--- lives-1.1.8/src/main.c	Sat Jan 16 16:23:43 2010
+++ lives-1.1.8-patch/src/main.c	Sat Jan 16 16:22:58 2010
@@ -1085,12 +1085,17 @@
 	    gchar jackd_loc[512];
 	    get_location("jackd",jackd_loc,512);
 	    if (strlen(jackd_loc)) {
-#ifndef IS_DARWIN
-	      com=g_strdup_printf("echo \"%s -Z -d alsa -s\">%s",jackd_loc,prefs->jack_aserver);
-#else
+#ifdef IS_DARWIN
 	      // use coreaudio on Darwin
 	      com=g_strdup_printf("echo \"%s -Z -d coreaudio -s\">%s",jackd_loc,prefs->jack_aserver);
+#else
+#ifdef IS_SOLARIS
+	      // use oss on Solaris
+	      com=g_strdup_printf("echo \"%s -Z -d oss -s\">%s",jackd_loc,prefs->jack_aserver);
+#else
+	      com=g_strdup_printf("echo \"%s -Z -d alsa -s\">%s",jackd_loc,prefs->jack_aserver);
 #endif
+#endif
 	      dummyvar=system(com);
 	      g_free(com);
 	      com=g_strdup_printf("/bin/chmod o+x %s",prefs->jack_aserver);
