--- lives-1.0.0/src/effects-weed.c	Fri Aug 21 13:06:27 2009
+++ lives-1.0.0-patch/src/effects-weed.c	Fri Aug 21 13:06:22 2009
@@ -28,7 +28,13 @@
 #define OIL_MEMCPY_MAX_BYTES 1024 // this can be tuned to provide optimal performance
 
 #ifdef ENABLE_OIL
-inline void *w_memcpy  (void *dest, const void *src, size_t n) {n<=OIL_MEMCPY_MAX_BYTES?oil_memcpy(dest,src,n):memcpy(dest,src,n); return dest;}
+inline void *w_memcpy  (void *dest, const void *src, size_t n) {
+	if (n<=OIL_MEMCPY_MAX_BYTES)
+		oil_memcpy(dest,src,n);
+	else 
+		memcpy(dest,src,n);
+	return dest;
+}
 #else
 inline void *w_memcpy  (void *dest, const void *src, size_t n) {return memcpy(dest,src,n);}
 #endif
