--- lives-1.6.1/src/utils.c	Fri Feb  3 16:41:54 2012
+++ lives-1.6.1-patch/src/utils.c	Sun Apr  1 14:57:22 2012
@@ -2926,9 +2926,20 @@
 
 gboolean check_dev_busy(gchar *devstr) {
   int ret;
+#ifdef IS_SOLARIS
+  struct flock lock;
+  lock.l_start = 0;
+  lock.l_whence = SEEK_SET;
+  lock.l_len = 0;
+  lock.l_type = F_WRLCK;
+#endif
   int fd=open(devstr,O_RDONLY|O_NONBLOCK);
   if (fd==-1) return FALSE;
+#ifdef IS_SOLARIS
+  ret=fcntl(fd, F_SETLK, &lock);
+#else
   ret=flock(fd,LOCK_EX|LOCK_NB);
+#endif
   close(fd);
   if (ret==-1) return FALSE;
   return TRUE;
--- lives-1.6.1/lives-plugins/plugins/decoders/asf_decoder.c	Fri Jan  6 01:21:41 2012
+++ lives-1.6.1-patch/lives-plugins/plugins/decoders/asf_decoder.c	Sun Apr  1 14:52:38 2012
@@ -30,7 +30,9 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <stdlib.h>
+#ifndef IS_SOLARIS
 #include <endian.h>
+#endif
 #include <sys/stat.h>
 
 const char *plugin_version="LiVES asf/wmv decoder version 1.0";
