--- llvm-2.5.orig/docs/CommandGuide/Makefile	2008-06-09 09:14:31.000000000 +0200
+++ llvm-2.5/docs/CommandGuide/Makefile	2009-05-08 15:25:40.763500732 +0200
@@ -52,11 +52,11 @@
 .SUFFIXES: .html .pod .1 .ps
 
 $(DST_HTML_DIR)%.html: %.pod $(DST_HTML_DIR)/.dir
-	pod2html --css=manpage.css --htmlroot=. \
+	${POD2HTML} --css=manpage.css --htmlroot=. \
 	  --podpath=. --noindex --infile=$< --outfile=$@ --title=$*
 
 $(DST_MAN_DIR)%.1: %.pod $(DST_MAN_DIR)/.dir
-	pod2man --release=CVS --center="LLVM Command Guide" $< $@
+	${POD2MAN} --release=CVS --center="LLVM Command Guide" $< $@
 
 $(DST_PS_DIR)%.ps: $(DST_MAN_DIR)%.1 $(DST_PS_DIR)/.dir
 	groff -Tps -man $< > $@
