--- lxsession-0.3.2/man/Makefile.in.orig	2009-03-16 11:54:02.174305978 +0800
+++ lxsession-0.3.2/man/Makefile.in	2009-03-16 11:55:20.944617695 +0800
@@ -404,9 +404,13 @@
 
 
 lxsession.1: lxsession.sgml
-	docbook-to-man $< > $@
+	(test -f "`which docbook-to-man`" && docbook-to-man $< > $@) || \
+	(test -f "`which db2man`" && db2man $< > $@) || \
+	(test -f "`which docbook2man`" && docbook2man $< > $@)
 lxsession-logout.1: lxsession-logout.sgml
-	docbook-to-man $< > $@
+	(test -f "`which docbook-to-man`" && docbook-to-man $< > $@) || \
+	(test -f "`which db2man`" && db2man $< > $@) || \
+	(test -f "`which docbook2man`" && docbook2man $< > $@)
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
--- lxsession-0.3.2/man/Makefile.am.orig	2009-03-16 12:23:52.486268686 +0800
+++ lxsession-0.3.2/man/Makefile.am	2009-03-16 12:24:58.029847706 +0800
@@ -4,7 +4,11 @@
 EXTRA_DIST = lxsession.sgml lxsession-logout.sgml
 
 lxsession.1: lxsession.sgml
-	docbook-to-man $< > $@
+	(test -f "`which docbook-to-man`" && docbook-to-man $< > $@) || \
+	(test -f "`which db2man`" && db2man $< > $@) || \
+	(test -f "`which docbook2man`" && docbook2man $< > $@)
 lxsession-logout.1: lxsession-logout.sgml
-	docbook-to-man $< > $@
-        
\ No newline at end of file
+	(test -f "`which docbook-to-man`" && docbook-to-man $< > $@) || \
+	(test -f "`which db2man`" && db2man $< > $@) || \
+	(test -f "`which docbook2man`" && docbook2man $< > $@)
+        
