--- lxsession-0.4.4/lxsession-logout/lxsession-logout.c-orig	2010-09-17 19:02:40.148284621 -0500
+++ lxsession-0.4.4/lxsession-logout/lxsession-logout.c	2010-09-17 19:03:29.216620337 -0500
@@ -370,6 +370,7 @@ int main(int argc, char * argv[])
         handler_context.reboot_ConsoleKit = TRUE;
     }
 
+#if 0
     /* Initialize capabilities of the UPower mechanism. */
     if (dbus_UPower_CanSuspend())
     {
@@ -403,6 +404,7 @@ int main(int argc, char * argv[])
         handler_context.hibernate_available = TRUE;
         handler_context.hibernate_HAL = TRUE;
     }
+#endif
 
     /* If we are under GDM, its "Switch User" is available. */
     if (verify_running("gdm", "gdmflexiserver"))
