--- menu-cache-0.2.5/menu-cache-daemon/Makefile.am-orig	2009-08-04 18:03:48.841591000 -0500
+++ menu-cache-0.2.5/menu-cache-daemon/Makefile.am	2009-08-04 18:03:52.220917000 -0500
@@ -7,7 +7,6 @@ INCLUDES =				\
 	$(DEBUG_CFLAGS)			\
 	$(WARN_CFLAGS)			\
 	-DMENUCACHE_LIBEXECDIR="\"$(libexecdir)\"" \
-	-Werror-implicit-function-declaration \
 	$(NULL)
 
 libexec_PROGRAMS = menu-cached
--- menu-cache-0.2.5/libmenu-cache/Makefile.am-orig	2009-08-04 18:04:04.994837000 -0500
+++ menu-cache-0.2.5/libmenu-cache/Makefile.am	2009-08-04 18:04:09.240302000 -0500
@@ -3,7 +3,6 @@ NULL =
 INCLUDES =				\
 	$(GLIB_CFLAGS)			\
 	$(MONITOR_CFLAGS)		\
-	-Werror-implicit-function-declaration \
 	-DMENUCACHE_LIBEXECDIR="\"$(libexecdir)\"" \
 	$(NULL)
 
--- menu-cache-0.2.5/menu-cache-gen/Makefile.am-orig	2009-08-04 18:06:14.744168000 -0500
+++ menu-cache-0.2.5/menu-cache-gen/Makefile.am	2009-08-04 18:06:17.988363000 -0500
@@ -7,7 +7,6 @@ INCLUDES =				\
 	$(MONITOR_CFLAGS)		\
 	$(DEBUG_CFLAGS)			\
 	$(WARN_CFLAGS)			\
-	-Werror-implicit-function-declaration \
 	$(NULL)
 
 libexec_PROGRAMS = menu-cache-gen
