--- menu-cache-0.3.3/libmenu-cache/Makefile.in	Sun Aug  5 15:00:28 2012
+++ menu-cache-0.3.3/libmenu-cache/Makefile.in	Sun Aug  5 15:00:58 2012
@@ -256,1 +256,0 @@
-	-Werror-implicit-function-declaration \
--- menu-cache-0.3.3/menu-cache-gen/Makefile.in	Sun May 20 16:53:53 2012
+++ menu-cache-0.3.3/menu-cache-gen/Makefile.in	Sun Aug  5 15:02:28 2012
@@ -231,1 +231,0 @@
-	-Werror-implicit-function-declaration \
--- menu-cache-0.3.3/menu-cache-daemon/Makefile.in	Sun May 20 16:53:53 2012
+++ menu-cache-0.3.3/menu-cache-daemon/Makefile.in	Sun Aug  5 15:04:29 2012
@@ -224,1 +224,0 @@
-	-Werror-implicit-function-declaration \
