--- metacity-clutter/src/compositor/mutter/compositor-mutter.c-orig	2009-04-03 21:43:44.430714000 -0500
+++ metacity-clutter/src/compositor/mutter/compositor-mutter.c	2009-04-03 21:43:51.223403000 -0500
@@ -1,7 +1,7 @@
 /* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
 
 #define _GNU_SOURCE
-#define _XOPEN_SOURCE 500 /* for usleep() */
+#define _XOPEN_SOURCE 600 /* for usleep() */
 
 #include <config.h>
 
--- metacity-clutter/src/compositor/compositor-xrender.c-orig	2009-04-03 21:23:10.810292000 -0500
+++ metacity-clutter/src/compositor/compositor-xrender.c	2009-04-03 21:23:19.335527000 -0500
@@ -22,7 +22,7 @@
  */
 
 #define _GNU_SOURCE
-#define _XOPEN_SOURCE 500 /* for usleep() */
+#define _XOPEN_SOURCE 600 /* for usleep() */
 
 #include <config.h>
 
