--- miniupnpd-1.6.20120207/upnpredirect.c	2011-06-22 13:34:50.000000000 -0700
+++ miniupnpd-1.6.20120207-patched/upnpredirect.c	2012-02-28 22:11:31.790656061 -0800
@@ -19,6 +19,7 @@
 #include <unistd.h>
 
 #include "config.h"
+#include "miniupnpdtypes.h"
 #include "upnpredirect.h"
 #include "upnpglobalvars.h"
 #include "upnpevents.h"
@@ -311,8 +312,7 @@
 {
 	/*syslog(LOG_INFO, "redirecting port %hu to %s:%hu protocol %s for: %s",
 		eport, iaddr, iport, protocol, desc);			*/
-	if(add_redirect_rule2(ext_if_name, rhost, eport, iaddr, iport, proto,
-	                      desc, timestamp) < 0) {
+	if(add_redirect_rule2(ext_if_name, eport, iaddr, iport, proto, desc) < 0) {
 		return -1;
 	}
 
@@ -321,7 +321,7 @@
 #endif
 /*	syslog(LOG_INFO, "creating pass rule to %s:%hu protocol %s for: %s",
 		iaddr, iport, protocol, desc);*/
-	if(add_filter_rule2(ext_if_name, rhost, iaddr, eport, iport, proto, desc) < 0) {
+	if(add_filter_rule2(ext_if_name, iaddr, eport, iport, proto, desc) < 0) {
 		/* clean up the redirect rule */
 #if !defined(__linux__)
 		delete_redirect_rule(ext_if_name, eport, proto);
