diff -Nar -U1 mjpegtools-1.9.0/lavtools/y4mstabilizer.c mjpegtools-1.9.0-patch/lavtools/y4mstabilizer.c
--- mjpegtools-1.9.0/lavtools/y4mstabilizer.c	2009-09-09 12:30:22.756035743 +0200
+++ mjpegtools-1.9.0-patch/lavtools/y4mstabilizer.c	2009-09-09 12:32:45.316513341 +0200
@@ -60,3 +60,7 @@
 
+#ifdef HAVE___PROGNAME
 extern  char    *__progname;
+#else
+char    *__progname="y4mstabilizer";
+#endif
 
diff -Nar -U1 mjpegtools-1.9.0/y4munsharp/y4munsharp.c mjpegtools-1.9.0-patch/y4munsharp/y4munsharp.c
--- mjpegtools-1.9.0/y4munsharp/y4munsharp.c	2009-09-09 12:30:22.551732672 +0200
+++ mjpegtools-1.9.0-patch/y4munsharp/y4munsharp.c	2009-09-09 12:36:33.040539836 +0200
@@ -48,3 +48,7 @@
 
+#ifdef HAVE___PROGNAME
 extern char *__progname;
+#else
+char    *__progname="y4munsharp";
+#endif
 
diff -Nar -U1 mjpegtools-1.9.0/y4mutils/yuv4mpeg.c mjpegtools-1.9.0-patch/y4mutils/yuv4mpeg.c
--- mjpegtools-1.9.0/y4mutils/yuv4mpeg.c	2009-09-09 12:30:22.848960405 +0200
+++ mjpegtools-1.9.0-patch/y4mutils/yuv4mpeg.c	2009-09-09 12:43:30.526280051 +0200
@@ -58,3 +58,7 @@
 
+#ifdef HAVE___PROGNAME
 extern	char	*__progname;
+#else
+char    *__progname="yuv4mpeg";
+#endif
 
diff -Nar -U1 mjpegtools-1.9.0/y4mutils/yuyvtoy4m.c mjpegtools-1.9.0-patch/y4mutils/yuyvtoy4m.c
--- mjpegtools-1.9.0/y4mutils/yuyvtoy4m.c	2009-09-09 12:30:22.840808078 +0200
+++ mjpegtools-1.9.0-patch/y4mutils/yuyvtoy4m.c	2009-09-09 12:37:28.137731562 +0200
@@ -40,3 +40,7 @@
 
+#ifdef HAVE___PROGNAME
 extern	char	*__progname;
+#else
+char    *__progname="yuyvtoy4m";
+#endif
 
diff -Nar -U1 mjpegtools-1.9.0/y4mutils/y4mblack.c mjpegtools-1.9.0-patch/y4mutils/y4mblack.c
--- mjpegtools-1.9.0/y4mutils/y4mblack.c	2009-09-09 12:30:22.847414530 +0200
+++ mjpegtools-1.9.0-patch/y4mutils/y4mblack.c	2009-09-09 12:41:38.603770141 +0200
@@ -17,3 +17,7 @@
 
+#ifdef HAVE___PROGNAME
 extern	char	*__progname;
+#else
+char   *__progname="y4mblack";
+#endif
 
diff -Nar -U1 mjpegtools-1.9.0/y4mutils/y4mshift.c mjpegtools-1.9.0-patch/y4mutils/y4mshift.c
--- mjpegtools-1.9.0/y4mutils/y4mshift.c	2009-09-09 12:30:22.841383810 +0200
+++ mjpegtools-1.9.0-patch/y4mutils/y4mshift.c	2009-09-09 12:38:22.767292691 +0200
@@ -24,3 +24,7 @@
 
+#ifdef HAVE___PROGNAME
 extern  char    *__progname;
+#else
+char    *__progname="y4mshift";
+#endif
 
diff -Nar -U1 mjpegtools-1.9.0/y4mutils/y4mspatialfilter.c mjpegtools-1.9.0-patch/y4mutils/y4mspatialfilter.c
--- mjpegtools-1.9.0/y4mutils/y4mspatialfilter.c	2009-09-09 12:30:22.848053200 +0200
+++ mjpegtools-1.9.0-patch/y4mutils/y4mspatialfilter.c	2009-09-09 12:42:21.205860714 +0200
@@ -23,3 +23,7 @@
 
+#ifdef HAVE___PROGNAME
 extern  char    *__progname;
+#else
+char    *__progname="y4mspatialfilter";
+#endif
 
