--- elisa-1.0.1/setup.py-orig	2009-05-28 11:53:01.233018000 -0500
+++ elisa-1.0.1/setup.py	2009-05-28 11:53:25.327682000 -0500
@@ -53,6 +53,16 @@ if platform.system() == 'Linux':
                         ['data/com.fluendo.elisa.service'])
                       ])
 
+elif platform.system() == 'SunOS':
+    data_files.extend([('share/pixmaps', ['data/moovida.png']),
+                       ('share/man/man1', ['docs/moovida.1']),
+                       ('share/applications',
+                        ['data/moovida.desktop']),
+                       ('share/icons', ['data/moovida.png']),
+                       ('share/dbus-1/services',
+                        ['data/com.fluendo.elisa.service'])
+                      ])
+
 elif platform.system() == 'Windows':
     module_include = ['tarfile', 'md5']
     package_include = ['distutils']
