--- elisa-1.0.5/elisa/core/utils/misc.py-orig	2009-07-21 04:59:22.299642000 -0500
+++ elisa-1.0.5/elisa/core/utils/misc.py	2009-07-21 04:59:59.222810000 -0500
@@ -211,7 +211,7 @@ def get_os_name():
     else:
         return platform.system().lower()
 
-def linux_pidof(program_name):
+def unix_pidof(program_name):
     """
     Get the Linux process id of the given program name. Because
     multiple processes can exist, we return a list of the pids.
@@ -219,9 +219,17 @@ def linux_pidof(program_name):
     @rtype: C{list} of C{int}
     @returns: the list of running pids of given program name
     """
-    output = subprocess.Popen(["pidof", program_name],
-                              stderr=subprocess.STDOUT,
-                              stdout=subprocess.PIPE).communicate()[0]
+    if platform.system() == 'SunOS':
+        output = subprocess.Popen(["/bin/sh", "-c", "/usr/bin/pgrep -u `whoami` %s" % program_name],
+                                  stderr=subprocess.STDOUT,
+                                  stdout=subprocess.PIPE,
+                                  close_fds=True).communicate()[0]
+    else:
+        output = subprocess.Popen(["pidof", program_name],
+                                  stderr=subprocess.STDOUT,
+                                  stdout=subprocess.PIPE,
+                                  close_fds=True).communicate()[0]
+
     return [int(pid) for pid in output.split()]
 
 def get_user_desktop_name():
@@ -257,7 +265,7 @@ def get_user_desktop_name():
                          'xfwm4': 'xfce',
                          }
                 for prog, name in progs.iteritems():
-                    if linux_pidof(prog) != []:
+                    if unix_pidof(prog) != []:
                         desktop_name = name
                         break
             if desktop_name is None:
