--- mplayer/configure.orig	Sat Nov  6 14:34:12 2010
+++ mplayer/configure	Sat Nov  6 14:37:22 2010
@@ -6708,7 +6708,7 @@
   _real=no
   res_comment="dynamic loader support needed"
   if test "$_dl" = yes || test "$_win32dll" = yes &&
-     (linux || freebsd || netbsd || openbsd || dragonfly || darwin || win32 || os2) ; then
+     (linux || freebsd || netbsd || openbsd || dragonfly || darwin || win32 || os2 || sunos) ; then
     _real=yes
   fi
 fi
