--- mplayer/configure.orig	Sun Nov 28 09:16:30 2010
+++ mplayer/configure	Fri Dec 24 14:31:59 2010
@@ -3055,6 +3055,14 @@
 fi
 echores "$_builtin_expect"
 
+echocheck "rpath"
+if test "$_rpath" = yes ; then
+  for I in $(echo $extra_ldflags | sed 's/-L//g') ; do
+    tmp="$tmp $(echo $I | sed 's/.*/ -L& -Wl,-R&/')"
+  done
+extra_ldflags=$tmp
+fi
+echores "$_rpath"
 
 echocheck "kstat"
 _kstat=no
@@ -3576,15 +3584,6 @@
 test "$_w32threads" = yes && def_threads='#define HAVE_THREADS 1'
 echores "$_w32threads"
 
-echocheck "rpath"
-if test "$_rpath" = yes ; then
-  for I in $(echo $extra_ldflags | sed 's/-L//g') ; do
-    tmp="$tmp $(echo $I | sed 's/.*/ -L& -Wl,-R&/')"
-  done
-extra_ldflags=$tmp
-fi
-echores "$_rpath"
-
 echocheck "iconv"
 if test "$_iconv" = auto ; then
   cat > $TMPC << EOF
