--- MPlayer-1.0rc3/configure	Sat May 29 22:16:38 2010
+++ MPlayer-1.0rc3-patch/configure	Sun May 30 16:37:36 2010
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/ksh
 #
 # Original version (C) 2000 Pontscho/fresh!mindworkz
 #                      pontscho@makacs.poliod.hu
@@ -2844,6 +2844,15 @@
 fi
 echores "$_builtin_expect"
 
+echocheck "rpath"
+netbsd &&_rpath=yes
+if test "$_rpath" = yes ; then
+  for I in `echo $extra_ldflags | sed 's/-L//g'` ; do
+    tmp="$tmp ` echo $I | sed 's/.*/ -L& -Wl,-R&/'`"
+  done
+extra_ldflags=$tmp
+fi
+echores "$_rpath"
 
 echocheck "kstat"
 cat > $TMPC << EOF
@@ -3323,16 +3332,6 @@
 test "$_w32threads" = yes && def_threads='#define HAVE_THREADS 1'
 echores "$_w32threads"
 
-echocheck "rpath"
-netbsd &&_rpath=yes
-if test "$_rpath" = yes ; then
-  for I in `echo $extra_ldflags | sed 's/-L//g'` ; do
-    tmp="$tmp ` echo $I | sed 's/.*/ -L& -Wl,-R&/'`"
-  done
-extra_ldflags=$tmp
-fi
-echores "$_rpath"
-
 echocheck "iconv"
 if test "$_iconv" = auto ; then
   cat > $TMPC << EOF
