--- mplayer/configure.orig	Sat Nov  6 14:32:20 2010
+++ mplayer/configure	Sat Nov  6 14:34:12 2010
@@ -3053,6 +3053,15 @@
 fi
 echores "$_builtin_expect"
 
+echocheck "rpath"
+netbsd &&_rpath=yes
+if test "$_rpath" = yes ; then
+  for I in $(echo $extra_ldflags | sed 's/-L//g') ; do
+    tmp="$tmp $(echo $I | sed 's/.*/ -L& -Wl,-R&/')"
+  done
+extra_ldflags=$tmp
+fi
+echores "$_rpath"
 
 echocheck "kstat"
 _kstat=no
@@ -3583,16 +3592,6 @@
 test "$_w32threads" = yes && def_threads='#define HAVE_THREADS 1'
 echores "$_w32threads"
 
-echocheck "rpath"
-netbsd &&_rpath=yes
-if test "$_rpath" = yes ; then
-  for I in $(echo $extra_ldflags | sed 's/-L//g') ; do
-    tmp="$tmp $(echo $I | sed 's/.*/ -L& -Wl,-R&/')"
-  done
-extra_ldflags=$tmp
-fi
-echores "$_rpath"
-
 echocheck "iconv"
 if test "$_iconv" = auto ; then
   cat > $TMPC << EOF
