--- MPlayer-1.0rc4/stream/stream_cddb.c	Sun Jan 30 13:01:16 2011
+++ MPlayer-1.0rc4-patch/stream/stream_cddb.c	Sun Jan 30 13:00:11 2011
@@ -56,7 +56,7 @@
 
 #if defined(__linux__)
 #include <linux/cdrom.h>
-#elif defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__DragonFly__)
+#elif defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__DragonFly__) || (defined(__sun) && defined(__SVR4))
 #include <sys/cdio.h>
 #elif defined(__MINGW32__) || defined(__CYGWIN__)
 #include <ddk/ntddcdrm.h>
@@ -207,7 +207,7 @@
         return drive;
     }
 
-#if defined(__linux__) || defined(__bsdi__)
+#if defined(__linux__) || defined(__bsdi__) || (defined (__sun) && defined (__SVR4))
     {
         struct cdrom_tochdr tochdr;
         ioctl(drive, CDROMREADTOCHDR, &tochdr);
