commit af8d71251a25c6b542b5fcb661427fec35b8e7cf
Author: Halton Huo <halton.huo@sun.com>
Date:   Tue Aug 4 18:44:23 2009 +0800

    Make version check compatible with solaris shell. Closes #590719.

diff --git a/configure.in b/configure.in
index 660c7fc..cffaddc 100644
--- a/configure.in
+++ b/configure.in
@@ -544,8 +544,10 @@ mutter-$VERSION:
 "
 
 MUTTER_MINOR_VERSION=mutter_minor_version
-if test $(( $(echo $MUTTER_MINOR_VERSION) %2)) == "1"; then
-   stable_version=$(( ($MUTTER_MINOR_VERSION / 2) * 2))
+if_unstable=`expr $MUTTER_MINOR_VERSION  % 2`
+if test $if_unstable == "1"; then
+   tmp=`expr $MUTTER_MINOR_VERSION / 2`
+   stable_version=`expr $tmp \* 2`
    echo "This is the UNSTABLE branch of mutter"
    echo -n "Use 2.$stable_version.x for stable "
    echo    "(gnome-2-$stable_version branch in Subversion)"
