--- mutter-3.4.1/src/Makefile.am-orig	2012-05-10 01:32:15.648477332 -0500
+++ mutter-3.4.1/src/Makefile.am	2012-05-10 01:32:21.131430979 -0500
@@ -229,7 +229,6 @@ Meta-$(api_version).gir: libmutter.la
 	mutter-enum-types.h			\
 	$(libmutterinclude_base_headers)	\
 	$(filter %.c,$(libmutter_la_SOURCES))
-@META_GIR@_SCANNERFLAGS = --warn-all --warn-error
 
 endif
 
--- mutter-3.4.1/src/core/delete.c-orig	2012-05-10 01:27:55.980329288 -0500
+++ mutter-3.4.1/src/core/delete.c	2012-05-10 01:28:59.971831192 -0500
@@ -22,7 +22,9 @@
  * 02111-1307, USA.
  */
 
-#define _XOPEN_SOURCE /* for kill() */
+#if 0
+#define _XOPEN_SOURCE  /* for kill() */
+#endif
 
 #include <config.h>
 #include <meta/util.h>
