--- mutter-2.27.1/src/core/util.c-orig	2009-08-03 17:24:13.457662000 -0500
+++ mutter-2.27.1/src/core/util.c	2009-08-03 17:24:24.315286000 -0500
@@ -23,7 +23,6 @@
  */
 
 #define _GNU_SOURCE
-#define _POSIX_C_SOURCE 200112L /* for fdopen() */
 
 #include <config.h>
 #include "util.h"
--- mutter-2.27.1/src/ui/preview-widget.c-orig	2009-08-03 17:24:45.205215000 -0500
+++ mutter-2.27.1/src/ui/preview-widget.c	2009-08-03 17:24:52.034927000 -0500
@@ -22,7 +22,6 @@
  */
 
 #define _GNU_SOURCE
-#define _XOPEN_SOURCE 600 /* for the maths routines over floats */
 
 #include <math.h>
 #include <gtk/gtk.h>
