--- mutter-2.27.1/src/compositor/plugins/Makefile.am-orig	2009-08-03 17:27:24.235223000 -0500
+++ mutter-2.27.1/src/compositor/plugins/Makefile.am	2009-08-03 17:27:31.774619000 -0500
@@ -4,7 +4,7 @@ pkglibdir=@MUTTER_PLUGIN_DIR@
 INCLUDES=@MUTTER_CFLAGS@ -I $(top_srcdir)/src/include -DMUTTER_LIBEXECDIR=\"$(libexecdir)\" -DHOST_ALIAS=\"@HOST_ALIAS@\" -DMUTTER_LOCALEDIR=\"$(prefix)/@DATADIRNAME@/locale\" -DMUTTER_PKGDATADIR=\"$(pkgdatadir)\" -DMUTTER_DATADIR=\"$(datadir)\" -DG_LOG_DOMAIN=\"mutter\" -DSN_API_NOT_YET_FROZEN=1 -DMUTTER_MAJOR_VERSION=$(MUTTER_MAJOR_VERSION) -DMUTTER_MINOR_VERSION=$(MUTTER_MINOR_VERSION) -DMUTTER_MICRO_VERSION=$(MUTTER_MICRO_VERSION) -DMUTTER_PLUGIN_API_VERSION=$(MUTTER_PLUGIN_API_VERSION) -DMUTTER_PLUGIN_DIR=\"@MUTTER_PLUGIN_DIR@\"
 
 default_la_CFLAGS   = -fPIC
-default_la_SOURCES  = default.c
+default_la_SOURCES  = default.c ../mutter-plugin.c
 default_la_LDFLAGS  = -module -avoid-version -no-undefined
 default_la_LIBADD   = @CLUTTER_LIBS@
 
