--- mutter-3.4.1/src/compositor/plugins/Makefile.am-orig	2012-05-10 01:24:18.730674004 -0500
+++ mutter-3.4.1/src/compositor/plugins/Makefile.am	2012-05-10 01:25:58.120297549 -0500
@@ -5,7 +5,7 @@ INCLUDES=@MUTTER_CFLAGS@ -I $(top_srcdir
 
 default_la_CFLAGS   = -fPIC
 default_la_SOURCES  = default.c
-default_la_LDFLAGS  = -module -avoid-version -no-undefined
+default_la_LDFLAGS  = -module -avoid-version -no-undefined -Wl,-M,./mapfile-mutter
 default_la_LIBADD   = @CLUTTER_LIBS@
 
 pkglib_LTLIBRARIES = default.la
--- /dev/null	2012-05-10 01:33:15.000000000 -0500
+++ mutter-3.4.1/src/compositor/plugins/mapfile-mutter	2012-05-10 01:24:13.493919038 -0500
@@ -0,0 +1,46 @@
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+# Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
+# Use is subject to license terms.
+#
+# ident "@(#)mapfile-libsqlite3 1.4     09/06/05 SMI"
+#
+# Defines the public interface to SQLite3
+#
+{
+       global:
+                meta_window_get_icon_geometry = FUNCTION extern;
+                meta_window_actor_get_workspace = FUNCTION extern;
+                meta_window_actor_get_meta_window = FUNCTION extern;
+                meta_plugin_unmaximize_completed = FUNCTION extern;
+                meta_plugin_debug_mode = FUNCTION extern;
+                meta_get_stage_for_screen = FUNCTION extern;
+                meta_plugin_switch_workspace_completed = FUNCTION extern;
+                meta_plugin_map_completed = FUNCTION extern;
+                meta_screen_get_size = FUNCTION extern;
+                meta_plugin_destroy_completed = FUNCTION extern;
+                meta_plugin_get_type = FUNCTION extern;
+                meta_plugin_maximize_completed = FUNCTION extern;
+                meta_plugin_minimize_completed = FUNCTION extern;
+                meta_window_actor_get_type = FUNCTION extern;
+                meta_window_get_window_type = FUNCTION extern;
+                meta_get_window_actors = FUNCTION extern;
+                meta_plugin_get_screen = FUNCTION extern;
+};
