--- mutter/src/compositor/mutter/compoistor-mutter.c-orig	2009-04-28 17:02:58.536836000 -0500
+++ mutter/src/compositor/mutter/compositor-mutter.c	2009-04-28 17:03:05.450805000 -0500
@@ -1,7 +1,7 @@
 /* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
 
 #define _GNU_SOURCE
-#define _XOPEN_SOURCE 500 /* for usleep() */
+#define _XOPEN_SOURCE 600 /* for usleep() */
 
 #include <config.h>
 
--- mutter/src/compositor/compositor-xrender.c-orig	2009-04-28 17:18:29.495107000 -0500
+++ mutter/src/compositor/compositor-xrender.c	2009-04-28 17:18:35.930930000 -0500
@@ -22,7 +22,7 @@
  */
 
 #define _GNU_SOURCE
-#define _XOPEN_SOURCE 500 /* for usleep() */
+#define _XOPEN_SOURCE 600 /* for usleep() */
 
 #include <config.h>
 
