--- netpbm-10.35.77/converter/other/Makefile	2010-11-17 16:26:16.913797412 +0100
+++ netpbm-10.35.77-patch/converter/other/Makefile	2010-11-17 16:26:04.138947911 +0100
@@ -27,7 +27,7 @@
   endif
 endif
 
-ifeq ($(shell libpng-config --version),)
+ifeq ($(shell libpng$(PNGVER)-config --version),)
   ifneq ($(PNGLIB),NONE)
     HAVE_PNGLIB = Y
     ifneq ($(PNGHDR_DIR)x,x)
@@ -39,7 +39,7 @@
   endif
 else
   HAVE_PNGLIB = Y
-  INCLUDES += $(shell libpng-config --cflags)
+  INCLUDES += $(shell libpng$(PNGVER)-config --cflags)
 endif
 
 ifneq ($(JPEGLIB),NONE)
