--- ocaml-3.11.2-orig/otherlibs/systhreads/Makefile	2008-07-15 15:31:32.000000000 +0000
+++ ocaml-3.11.2/otherlibs/systhreads/Makefile	2010-05-23 03:09:50.833958107 +0000
@@ -32,7 +32,7 @@
 allopt: libthreadsnat.a threads.cmxa
 
 libthreads.a: $(BYTECODE_C_OBJS)
-	$(MKLIB) -o threads $(BYTECODE_C_OBJS)
+	$(MKLIB) -ldopt "$(BYTECCLIBS)" -o threads $(BYTECODE_C_OBJS)
 
 posix_b.o: posix.c
 	$(BYTECC) -O -I../../byterun $(BYTECCCOMPOPTS) $(SHAREDCCCOMPOPTS) \
--- ocaml-3.11.2-orig/otherlibs/threads/Makefile	2008-10-08 13:09:39.000000000 +0000
+++ ocaml-3.11.2/otherlibs/threads/Makefile	2010-05-23 03:09:50.834326250 +0000
@@ -54,7 +54,7 @@
 allopt:
 
 libvmthreads.a: $(C_OBJS)
-	$(MKLIB) -o threads -oc vmthreads $(C_OBJS)
+	$(MKLIB) -ldopt "$(BYTECCLIBS)" -o threads -oc vmthreads $(C_OBJS)
 
 threads.cma: $(CAML_OBJS)
 	$(MKLIB) -ocamlc '$(CAMLC)' -o threads -oc vmthreads $(CAML_OBJS)
--- ocaml-3.11.2-orig/otherlibs/unix/Makefile	2007-11-06 15:16:56.000000000 +0000
+++ ocaml-3.11.2/otherlibs/unix/Makefile	2010-05-23 03:19:32.345485105 +0000
@@ -19,6 +19,8 @@
 
 EXTRACAMLFLAGS=-nolabels
 
+LDOPTS=-ldopt "$(BYTECCLIBS)"
+
 COBJS=accept.o access.o addrofstr.o alarm.o bind.o chdir.o chmod.o \
   chown.o chroot.o close.o closedir.o connect.o cst2constr.o cstringv.o \
   dup.o dup2.o envir.o errmsg.o execv.o execve.o execvp.o exit.o \
