--- opencity-0.0.6.2stable/configure.ac	2009-10-22 21:00:46.000000000 +0200
+++ opencity-0.0.6.2stable-patch/configure.ac	2010-05-17 15:53:27.127410650 +0200
@@ -185,7 +185,7 @@
 dnl ===========================================================================
 dnl Set the compiler and the linker options
 dnl ===========================================================================
-CXXFLAGS="-Wall -Wmissing-braces -Wparentheses -pedantic-errors $CXXFLAGS"
+CXXFLAGS="-Wall -Wmissing-braces -Wparentheses $CXXFLAGS"
 
 CXXFLAGS="-I$with_gl_prefix/include $enable_gprof $enable_debug $CXXFLAGS"
 
