--- pcmanfm-0.9.10/src/desktop.c	2011-10-24 22:21:16.000000000 +0200
+++ pcmanfm-0.9.10/src/desktop.c	2012-02-05 13:37:15.541150086 +0100
@@ -2112,12 +2112,6 @@
     save_item_pos(desktop);
 }
 
-/* round() is only available in C99. Don't use it now for portability. */
-inline double _round(double x)
-{
-    return (x > 0.0) ? floor(x + 0.5) : ceil(x - 0.5);
-}
-
 static void on_snap_to_grid(GtkAction* act, gpointer user_data)
 {
     FmDesktop* desktop = FM_DESKTOP(user_data);
@@ -2141,8 +2135,8 @@
         item = (FmDesktopItem*)l->data;
         if(!item->fixed_pos)
             continue;
-        new_x = x + _round((double)(item->x - x) / desktop->cell_w) * desktop->cell_w;
-        new_y = y + _round((double)(item->y - y) / desktop->cell_h) * desktop->cell_h;
+        new_x = x + round((double)(item->x - x) / desktop->cell_w) * desktop->cell_w;
+        new_y = y + round((double)(item->y - y) / desktop->cell_h) * desktop->cell_h;
         move_item(desktop, item, new_x, new_y, FALSE);
     }
     g_list_free(items);
