--- pcmanfm-0.9.8/configure.ac-orig	2011-06-08 11:41:49.182109819 -0500
+++ pcmanfm-0.9.8/configure.ac	2011-06-08 11:41:56.659145247 -0500
@@ -3,7 +3,7 @@
 
 AC_PREREQ([2.63])
 AC_INIT([pcmanfm], [0.9.10], [http://pcmanfm.sf.net/])
-AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+AM_INIT_AUTOMAKE()
 
 AC_CONFIG_HEADERS([config.h])
 
--- pcmanfm-0.9.8/src/Makefile.am-orig	2011-06-08 11:42:12.449585769 -0500
+++ pcmanfm-0.9.8/src/Makefile.am	2011-06-08 11:42:19.682480100 -0500
@@ -32,8 +32,6 @@ pcmanfm_CFLAGS = \
 	$(FM_CFLAGS) \
 	$(MENU_CACHE_CFLAGS) \
 	$(G_CAST_CHECKS) \
-	-Wall \
-	-Werror-implicit-function-declaration \
 	$(NULL)
 
 pcmanfm_LDADD = \
