Index: trunk/src/notifier/notifier-impl.c
===================================================================
--- trunk/src/notifier/notifier-impl.c	(revision 464)
+++ trunk/src/notifier/notifier-impl.c	(working copy)
@@ -34,7 +34,7 @@
 #define TIMEOUT     6000
 
 static int width, height, max_slots;
-static int slots[MAX_SLOTS] = {[0 ... (MAX_SLOTS - 1)] = -1 };
+static int slots[MAX_SLOTS];
 static int notifier_initialized = 0;
 static int popup_timeout = TIMEOUT;
 static GdkPixbuf *icon_pixbuf;
