--- postfix-2.5.6/src/util/sys_defs.h.orig	Tue Jan 15 01:51:44 2008
+++ postfix-2.5.6/src/util/sys_defs.h	Mon Jan  4 10:43:18 2010
@@ -388,7 +388,6 @@
 #define DEF_DB_TYPE	"dbm"
 #define ALIAS_DB_MAP	"dbm:/etc/mail/aliases"
 #define HAS_NIS
-#define HAS_NISPLUS
 #define USE_SYS_SOCKIO_H		/* Solaris 2.5, changed sys/ioctl.h */
 #define GETTIMEOFDAY(t)	gettimeofday(t)
 #define ROOT_PATH	"/bin:/usr/bin:/sbin:/usr/sbin:/usr/ucb"
