--- pure-ftpd-1.0.31/configure	Sun Apr 17 17:08:36 2011
+++ pure-ftpd-1.0.31-patch/configure	Sun Apr 24 18:27:50 2011
@@ -11330,2 +11330,2 @@
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking for ber_init in -llber" >&5
-$as_echo_n "checking for ber_init in -llber... " >&6; }
+    { $as_echo "$as_me:${as_lineno-$LINENO}: checking for ber_init in -llber-2.4" >&5
+$as_echo_n "checking for ber_init in -llber-2.4... " >&6; }
@@ -11336,1 +11336,1 @@
-LIBS="-llber  $LIBS"
+LIBS="-llber-2.4  $LIBS"
@@ -11371,1 +11371,1 @@
-  LIBS="-llber $LIBS"
+  LIBS="-llber-2.4 $LIBS"
@@ -11377,2 +11377,2 @@
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking for ldap_init in -lldap" >&5
-$as_echo_n "checking for ldap_init in -lldap... " >&6; }
+    { $as_echo "$as_me:${as_lineno-$LINENO}: checking for ldap_init in -lldap-2.4" >&5
+$as_echo_n "checking for ldap_init in -lldap-2.4... " >&6; }
@@ -11383,1 +11383,1 @@
-LIBS="-lldap  $LIBS"
+LIBS="-lldap-2.4  $LIBS"
@@ -11418,1 +11418,1 @@
-  LIBS="-lldap $LIBS"
+  LIBS="-lldap-2.4 $LIBS"
@@ -11421,4 +11421,4 @@
-  { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Initial check for -lldap failed! May need -lssl -lcrypto" >&5
-$as_echo "$as_me: WARNING: Initial check for -lldap failed! May need -lssl -lcrypto" >&2;}
-      { $as_echo "$as_me:${as_lineno-$LINENO}: checking for ldap_bind in -lldap" >&5
-$as_echo_n "checking for ldap_bind in -lldap... " >&6; }
+  { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Initial check for -lldap-2.4 failed! May need -lssl -lcrypto" >&5
+$as_echo "$as_me: WARNING: Initial check for -lldap-2.4 failed! May need -lssl -lcrypto" >&2;}
+      { $as_echo "$as_me:${as_lineno-$LINENO}: checking for ldap_bind in -lldap-2.4" >&5
+$as_echo_n "checking for ldap_bind in -lldap-2.4... " >&6; }
@@ -11429,1 +11429,1 @@
-LIBS="-lldap -lssl -lcrypto $LIBS"
+LIBS="-lldap-2.4 -lssl -lcrypto $LIBS"
