--- pure-ftpd-1.0.30/configure	2010-12-30 11:42:52.000000000 +0100
+++ pure-ftpd-1.0.30-patch/configure	2011-04-04 23:37:03.247665122 +0200
@@ -23522,2 +23522,2 @@
-{ echo "$as_me:$LINENO: checking for ber_init in -llber" >&5
-echo $ECHO_N "checking for ber_init in -llber... $ECHO_C" >&6; }
+{ echo "$as_me:$LINENO: checking for ber_init in -llber-2.4" >&5
+echo $ECHO_N "checking for ber_init in -llber-2.4... $ECHO_C" >&6; }
@@ -23528 +23528 @@
-LIBS="-llber  $LIBS"
+LIBS="-llber-2.4  $LIBS"
@@ -23588 +23588 @@
-  LIBS="-llber $LIBS"
+  LIBS="-llber-2.4 $LIBS"
@@ -23591,2 +23591,2 @@
-  { { echo "$as_me:$LINENO: error: liblber is needed for LDAP support" >&5
-echo "$as_me: error: liblber is needed for LDAP support" >&2;}
+  { { echo "$as_me:$LINENO: error: liblber-2.4 is needed for LDAP support" >&5
+echo "$as_me: error: liblber-2.4 is needed for LDAP support" >&2;}
@@ -23597,2 +23597,2 @@
-{ echo "$as_me:$LINENO: checking for ldap_init in -lldap" >&5
-echo $ECHO_N "checking for ldap_init in -lldap... $ECHO_C" >&6; }
+{ echo "$as_me:$LINENO: checking for ldap_init in -lldap-2.4" >&5
+echo $ECHO_N "checking for ldap_init in -lldap-2.4... $ECHO_C" >&6; }
@@ -23603 +23603 @@
-LIBS="-lldap  $LIBS"
+LIBS="-lldap-2.4  $LIBS"
@@ -23663 +23663 @@
-  LIBS="-lldap $LIBS"
+  LIBS="-lldap-2.4 $LIBS"
@@ -23666,4 +23666,4 @@
-  { echo "$as_me:$LINENO: WARNING: Initial check for -lldap failed! May need -lssl -lcrypto" >&5
-echo "$as_me: WARNING: Initial check for -lldap failed! May need -lssl -lcrypto" >&2;}
-      { echo "$as_me:$LINENO: checking for ldap_bind in -lldap" >&5
-echo $ECHO_N "checking for ldap_bind in -lldap... $ECHO_C" >&6; }
+  { echo "$as_me:$LINENO: WARNING: Initial check for -lldap-2.4 failed! May need -lssl -lcrypto" >&5
+echo "$as_me: WARNING: Initial check for -lldap-2.4 failed! May need -lssl -lcrypto" >&2;}
+      { echo "$as_me:$LINENO: checking for ldap_bind in -lldap-2.4" >&5
+echo $ECHO_N "checking for ldap_bind in -lldap-2.4... $ECHO_C" >&6; }
@@ -23674 +23674 @@
-LIBS="-lldap -lssl -lcrypto $LIBS"
+LIBS="-lldap-2.4 -lssl -lcrypto $LIBS"
