--- pygame-1.9.1release/src/scrap.c-orig	2010-02-03 07:54:55.311873723 -0600
+++ pygame-1.9.1release/src/scrap.c	2010-02-03 07:55:14.764025824 -0600
@@ -52,7 +52,7 @@ static PyObject* _scrap_lost_scrap (PyOb
 static PyObject* _scrap_set_mode (PyObject* self, PyObject* args);
 
 /* Determine what type of clipboard we are using */
-#if defined(__unix__) && defined(SDL_VIDEO_DRIVER_X11)
+#if defined(__sun) && defined(SDL_VIDEO_DRIVER_X11)
 /*!defined(__QNXNTO__) &&*/
     #define X11_SCRAP
     #include <time.h> /* Needed for clipboard timeouts. */
