--- qbittorrent-2.9.11/src/misc.cpp	Sun Aug  5 21:39:29 2012
+++ qbittorrent-2.9.11/src/misc.cpp	Sun Aug  5 21:42:07 2012
@@ -224,9 +224,14 @@
 
 #ifndef Q_WS_WIN
   unsigned long long available;
-  struct statfs stats;
   const QString statfs_path = dir_path.path()+"/.";
+#ifdef Q_OS_SOLARIS
+  struct statvfs stats;
+  const int ret = statvfs (qPrintable(statfs_path), &stats) ;
+#else
+  struct statfs stats;
   const int ret = statfs (qPrintable(statfs_path), &stats) ;
+#endif
   if(ret == 0) {
     available = ((unsigned long long)stats.f_bavail) *
         ((unsigned long long)stats.f_bsize) ;
