--- recordmysolaris-0.3/src/rms_getzpixmap.c	Sun Apr  5 16:05:44 2009
+++ recordmysolaris-0.3-patch/src/rms_getzpixmap.c	Tue Apr 27 23:09:53 2010
@@ -31,7 +31,7 @@
 
 #include <X11/Xlib.h>
 #include <X11/Xlibint.h>
-#include <X11/extensions/shmstr.h>
+#include <X11/extensions/shmproto.h>
 #include <X11/extensions/XShm.h>
 
 
--- recordmysolaris-0.3/src/rms_update_image.c	Sun Apr  5 16:05:44 2009
+++ recordmysolaris-0.3-patch/src/rms_update_image.c	Tue Apr 27 23:12:46 2010
@@ -32,7 +32,6 @@
 #include "rms_types.h"
 
 #include <X11/Xlibint.h>
-#include <X11/extensions/shmstr.h>
 #include <X11/extensions/XShm.h>
 
 
