--- samba-4.0.0alpha9/source4/lib/tls/config.m4.old	2009-12-01 17:15:54.336860000 +0800
+++ samba-4.0.0alpha9/source4/lib/tls/config.m4	2009-12-01 18:38:08.955586000 +0800
@@ -41,5 +41,6 @@
 	AC_DEFINE(ENABLE_GNUTLS,1,[Whether we have gnutls support (SSL)])
 	AC_CHECK_HEADERS(gcrypt.h)
 	AC_CHECK_LIB_EXT(gcrypt, GCRYPT_LIBS, gcry_control)
+	AC_CHECK_LIB_EXT(gpg-error, GCRYPT_LIBS, gpg_err_code_from_errno) 
 	SMB_EXT_LIB(GCRYPT, $GCRYPT_LIBS)
 fi
