--- samba-4.0.0alpha7/source4/build/m4/check_cc.m4.old	2009-11-18 15:40:17.419004000 +0800
+++ samba-4.0.0alpha7/source4/build/m4/check_cc.m4	2009-11-18 16:19:30.966758000 +0800
@@ -16,6 +16,24 @@
 	CFLAGS="${CFLAGS} -g"
 fi
 
+dnl ###########################################################################
+dnl _AC_LANG_COMPILER_SUNCC
+dnl Check whether the compiler for the current language is really Sun compiler.
+dnl ###########################################################################
+m4_define([AC_LANG_COMPILER_SUNCC],
+[AC_CACHE_CHECK([whether we are really using the Sun _AC_LANG compiler],
+                [ac_cv_[]_AC_LANG_ABBREV[]_compiler_suncc],
+[_AC_COMPILE_IFELSE([AC_LANG_PROGRAM([], [[#ifndef __SUNPRO_C
+       choke me
+#endif
+]])],
+                   [ac_compiler_suncc=yes],
+                   [ac_compiler_suncc=no])
+ac_cv_[]_AC_LANG_ABBREV[]_compiler_suncc=$ac_compiler_suncc
+])])
+
+AC_LANG_COMPILER_SUNCC
+
 ############################################
 # check if the compiler handles c99 struct initialization
 LIBREPLACE_C99_STRUCT_INIT(samba_cv_c99_struct_initialization=yes,
@@ -129,7 +147,7 @@
 	    AC_SUBST(CFLAG_NO_CAST_QUAL)
 	    AX_CFLAGS_GCC_OPTION(-Wno-unused-macros, CFLAG_NO_UNUSED_MACROS)
 	    AC_SUBST(CFLAG_NO_UNUSED_MACROS)
-	else
+        elif test x"$ac_cv_c_compiler_suncc" != x"yes"; then
 	    AX_CFLAGS_IRIX_OPTION(-fullwarn, DEVELOPER_CFLAGS)
 	fi
 
