--- samba-4.0.0alpha7/source4/param/samba-hostconfig.pc.in.orig	2009-06-01 09:36:22.568682217 +0800
+++ samba-4.0.0alpha7/source4/param/samba-hostconfig.pc.in	2009-06-01 09:42:46.721722463 +0800
@@ -7,4 +7,4 @@
 Description: Host-wide Samba configuration
 Version: 0.0.1
 Libs: -L${libdir} -lsamba-hostconfig 
-Cflags: -I${includedir}  -DHAVE_IMMEDIATE_STRUCTURES=1
+Cflags: -I${includedir}
--- samba-4.0.0alpha7/source4/auth/gensec/gensec.pc.in.orig	2009-06-01 09:36:51.497845184 +0800
+++ samba-4.0.0alpha7/source4/auth/gensec/gensec.pc.in	2009-06-01 09:42:46.722059867 +0800
@@ -8,4 +8,4 @@
 Description: Generic Security Library
 Version: 0.0.1
 Libs: -L${libdir} -lgensec 
-Cflags: -I${includedir}  -DHAVE_IMMEDIATE_STRUCTURES=1
+Cflags: -I${includedir}
--- samba-4.0.0alpha7/source4/lib/registry/registry.pc.in.orig	2009-06-01 09:38:30.603147182 +0800
+++ samba-4.0.0alpha7/source4/lib/registry/registry.pc.in	2009-06-01 09:42:46.723353028 +0800
@@ -9,4 +9,4 @@
 Requires.private: ldb
 Version: 0.0.1
 Libs: -L${libdir} -lregistry 
-Cflags: -I${includedir}  -DHAVE_IMMEDIATE_STRUCTURES=1
+Cflags: -I${includedir}
--- samba-4.0.0alpha7/source4/heimdal/lib/roken/vis.c.orig	2009-06-01 09:41:07.934194361 +0800
+++ samba-4.0.0alpha7/source4/heimdal/lib/roken/vis.c	2009-06-01 09:42:46.726750847 +0800
@@ -226,8 +226,8 @@
 	}
 	if (isextra || ((c & 0177) == ' ') || (flag & VIS_OCTAL)) {
 		*dst++ = '\\';
-		*dst++ = (u_char)(((u_int32_t)(u_char)c >> 6) & 03) + '0';
-		*dst++ = (u_char)(((u_int32_t)(u_char)c >> 3) & 07) + '0';
+		*dst++ = (u_char)(((uint32_t)(u_char)c >> 6) & 03) + '0';
+		*dst++ = (u_char)(((uint32_t)(u_char)c >> 3) & 07) + '0';
 		*dst++ =			     (c	      & 07) + '0';
 	} else {
 		if ((flag & VIS_NOSLASH) == 0) *dst++ = '\\';
--- samba-4.0.0alpha7/source4/heimdal/lib/roken/vis.hin.orig	2009-02-26 10:21:08.000000000 +0800
+++ samba-4.0.0alpha7/source4/heimdal/lib/roken/vis.hin	2009-06-01 09:45:40.577189112 +0800
@@ -84,7 +84,6 @@
 #include <sys/cdefs.h>
 #endif
 
-__BEGIN_DECLS
 char * ROKEN_LIB_FUNCTION
 	rk_vis(char *, int, int, int);
 char * ROKEN_LIB_FUNCTION
@@ -103,7 +102,6 @@
 	rk_strunvisx(char *, const char *, int);
 int ROKEN_LIB_FUNCTION
 	rk_unvis(char *, int, int *, int);
-__END_DECLS
 
 #undef vis
 #define vis(a,b,c,d) rk_vis(a,b,c,d)
--- samba-4.0.0alpha7/source4/librpc/dcerpc.pc.in.orig	2009-06-01 09:38:02.938601496 +0800
+++ samba-4.0.0alpha7/source4/librpc/dcerpc.pc.in	2009-06-01 09:42:46.723014367 +0800
@@ -8,4 +8,4 @@
 Requires: ndr
 Version: 0.0.1
 Libs: -L${libdir} -ldcerpc 
-Cflags: -I${includedir} -DHAVE_IMMEDIATE_STRUCTURES=1
+Cflags: -I${includedir}
--- samba-4.0.0alpha7/source4/librpc/dcerpc_samr.pc.in.orig	2009-06-01 09:37:19.625753106 +0800
+++ samba-4.0.0alpha7/source4/librpc/dcerpc_samr.pc.in	2009-06-01 09:42:46.722387431 +0800
@@ -8,4 +8,4 @@
 Requires.private: dcerpc ndr
 Version: 0.0.1
 Libs: -L${libdir} -ldcerpc_samr 
-Cflags: -I${includedir}  -DHAVE_IMMEDIATE_STRUCTURES=1
+Cflags: -I${includedir}
--- samba-4.0.0alpha7/source4/librpc/dcerpc_atsvc.pc.in.orig	2009-06-01 09:37:40.442833262 +0800
+++ samba-4.0.0alpha7/source4/librpc/dcerpc_atsvc.pc.in	2009-06-01 09:42:46.722704353 +0800
@@ -8,4 +8,4 @@
 Requires.private: dcerpc ndr
 Version: 0.0.1
 Libs: -L${libdir} -ldcerpc_atsvc
-Cflags: -I${includedir}  -DHAVE_IMMEDIATE_STRUCTURES=1
+Cflags: -I${includedir}
--- samba-4.0.0alpha7/lib/torture/torture.pc.in.orig	2009-06-01 09:35:59.113157789 +0800
+++ samba-4.0.0alpha7/lib/torture/torture.pc.in	2009-06-01 09:42:46.721384906 +0800
@@ -9,4 +9,4 @@
 Requires: talloc
 Version: 0.0.1
 Libs: -L${libdir} -ltorture 
-Cflags: -I${includedir}  -DHAVE_IMMEDIATE_STRUCTURES=1
+Cflags: -I${includedir}
--- samba-4.0.0alpha7/lib/util/debug.h.orig	2009-06-01 09:39:35.821053200 +0800
+++ samba-4.0.0alpha7/lib/util/debug.h	2009-06-01 09:42:46.724076857 +0800
@@ -52,7 +52,7 @@
 	if (DEBUGLVL(level)) { \
 		void* _debug_ctx=NULL; \
 		if (header) { \
-			dbghdr(level, __location__, __FUNCTION__); \
+			dbghdr(level, __location__, __func__); \
 		} \
 		dbgtext body; \
 		talloc_free(_debug_ctx); \
--- samba-4.0.0alpha7/nsswitch/winbind_nss_solaris.h.orig	2009-06-01 09:40:21.741375969 +0800
+++ samba-4.0.0alpha7/nsswitch/winbind_nss_solaris.h	2009-06-01 09:42:46.725244405 +0800
@@ -37,17 +37,6 @@
 
 #include <nss_common.h>
 
-#ifndef __cplusplus
-#define class #error DONT_USE_CPLUSPLUS_RESERVED_NAMES
-#define private #error DONT_USE_CPLUSPLUS_RESERVED_NAMES
-#define public #error DONT_USE_CPLUSPLUS_RESERVED_NAMES
-#define protected #error DONT_USE_CPLUSPLUS_RESERVED_NAMES
-#define template #error DONT_USE_CPLUSPLUS_RESERVED_NAMES
-#define this #error DONT_USE_CPLUSPLUS_RESERVED_NAMES
-#define new #error DONT_USE_CPLUSPLUS_RESERVED_NAMES
-#define delete #error DONT_USE_CPLUSPLUS_RESERVED_NAMES
-#define friend #error DONT_USE_CPLUSPLUS_RESERVED_NAMES
-#endif
 
 #include <nss_dbdefs.h>
 #include <nsswitch.h>
--- samba-4.0.0alpha7/librpc/ndr/libndr.h.orig	2009-02-26 10:21:08.000000000 +0800
+++ samba-4.0.0alpha7/librpc/ndr/libndr.h	2009-06-01 09:42:46.724892466 +0800
@@ -419,6 +419,7 @@
 			 struct ndr_token_list **list, 
 			 const void *key, 
 			 uint32_t value);
+typedef int (*comparison_fn_t)(const void *, const void *);
 enum ndr_err_code ndr_token_retrieve_cmp_fn(struct ndr_token_list **list, const void *key, uint32_t *v, comparison_fn_t _cmp_fn, bool _remove_tok);
 enum ndr_err_code ndr_token_retrieve(struct ndr_token_list **list, const void *key, uint32_t *v);
 uint32_t ndr_token_peek(struct ndr_token_list **list, const void *key);
--- samba-4.0.0alpha7/librpc/ndr.pc.in.orig	2009-06-01 09:39:07.323825171 +0800
+++ samba-4.0.0alpha7/librpc/ndr.pc.in	2009-06-01 09:42:46.723702231 +0800
@@ -8,4 +8,4 @@
 Requires: samba-hostconfig talloc
 Version: 0.0.1
 Libs: -L${libdir} -lndr 
-Cflags: -I${includedir}  -DHAVE_IMMEDIATE_STRUCTURES=1 -D_GNU_SOURCE=1
+Cflags: -I${includedir}
