--- samba-4.0.0alpha7/source4/lib/ldb/ldb_map/ldb_map.h.old	2009-05-12 18:24:42.752178818 +0800
+++ samba-4.0.0alpha7/source4/lib/ldb/ldb_map/ldb_map.h	2009-05-12 18:31:36.394468236 +0800
@@ -61,7 +61,7 @@
 	enum ldb_map_attr_type { 
 		MAP_IGNORE, /* Ignore this local attribute. Doesn't exist remotely.  */
 		MAP_KEEP,   /* Keep as is. Same name locally and remotely. */
-		MAP_RENAME, /* Simply rename the attribute. Name changes, data is the same */
+		LDB_MAP_RENAME, /* Simply rename the attribute. Name changes, data is the same */
 		MAP_CONVERT, /* Rename + convert data */
 		MAP_GENERATE /* Use generate function for generating new name/data. 
 						Used for generating attributes based on 
--- samba-4.0.0alpha7/source4/lib/ldb/ldb_map/ldb_map_outbound.c.old	2009-05-12 18:33:03.538404925 +0800
+++ samba-4.0.0alpha7/source4/lib/ldb/ldb_map/ldb_map_outbound.c	2009-05-12 18:33:31.917850575 +0800
@@ -130,7 +130,7 @@
 			name = attrs[i];
 			goto named;
 
-		case MAP_RENAME:
+		case LDB_MAP_RENAME:
 		case MAP_CONVERT:
 			name = map->u.rename.remote_name;
 			goto named;
@@ -288,7 +288,7 @@
 	case MAP_KEEP:
 		remote_name = attr_name;
 		break;
-	case MAP_RENAME:
+	case LDB_MAP_RENAME:
 		remote_name = map->u.rename.remote_name;
 		break;
 	case MAP_GENERATE:
@@ -309,7 +309,7 @@
 		}
 		/* fall through */
 	case MAP_KEEP:
-	case MAP_RENAME:
+	case LDB_MAP_RENAME:
 		old = ldb_msg_find_element(remote, remote_name);
 		if (old) {
 			el = ldb_msg_el_map_remote(module, local, map, attr_name, old);
@@ -813,7 +813,7 @@
 		return 0;
 	}
 
-	if (map->type == MAP_RENAME) {
+	if (map->type == LDB_MAP_RENAME) {
 		/* Nothing more to do here, the attribute has been renamed */
 		return 0;
 	}
--- samba-4.0.0alpha7/source4/lib/ldb/ldb_map/ldb_map.c.old	2009-05-12 18:33:43.362567087 +0800
+++ samba-4.0.0alpha7/source4/lib/ldb/ldb_map/ldb_map.c	2009-05-12 18:34:05.203945756 +0800
@@ -322,7 +322,7 @@
 			}
 			break;
 
-		case MAP_RENAME:
+		case LDB_MAP_RENAME:
 		case MAP_CONVERT:
 			if (ldb_attr_cmp(map->u.rename.remote_name, name) == 0) {
 				return map;
@@ -373,7 +373,7 @@
 	case MAP_KEEP:
 		return talloc_strdup(mem_ctx, attr);
 
-	case MAP_RENAME:
+	case LDB_MAP_RENAME:
 	case MAP_CONVERT:
 		return talloc_strdup(mem_ctx, map->u.rename.remote_name);
 
@@ -514,7 +514,7 @@
 			}
 			/* fall through */
 		case MAP_KEEP:
-		case MAP_RENAME:
+		case LDB_MAP_RENAME:
 			name = map_attr_map_local(newdn, map, ldb_dn_get_component_name(dn, i));
 			if (name == NULL) goto failed;
 
@@ -589,7 +589,7 @@
 			}
 			/* fall through */
 		case MAP_KEEP:
-		case MAP_RENAME:
+		case LDB_MAP_RENAME:
 			name = map_attr_map_remote(newdn, map, ldb_dn_get_component_name(dn, i));
 			if (name == NULL) goto failed;
 
--- samba-4.0.0alpha7/source4/lib/ldb/ldb_map/ldb_map_inbound.c.old	2009-05-12 18:32:15.810252100 +0800
+++ samba-4.0.0alpha7/source4/lib/ldb/ldb_map/ldb_map_inbound.c	2009-05-12 18:32:21.795690250 +0800
@@ -91,7 +91,7 @@
 		}
 		/* fall through */
 	case MAP_KEEP:
-	case MAP_RENAME:
+	case LDB_MAP_RENAME:
 		el = ldb_msg_el_map_local(module, remote, map, old);
 		break;
 
