--- scorched/src/server/tankai/TankAICurrentTarget.cpp	Sat May  8 12:57:53 2010
+++ scorched-patch/src/server/tankai/TankAICurrentTarget.cpp	Sat May  8 12:57:40 2010
@@ -179,20 +179,20 @@
 
 		if (damageTaken_.find(currentTank) != damageTaken_.end())
 		{
-			damageTakenSorted.insert(std::pair<float, Tank *>(
+			damageTakenSorted.insert(std::pair<const float, Tank *>(
 				damageTaken_[currentTank], currentTank));
 		}
 		if (damageGiven_.find(currentTank) != damageGiven_.end())
 		{
-			damageGivenSorted.insert(std::pair<float, Tank *>(
+			damageGivenSorted.insert(std::pair<const float, Tank *>(
 				damageGiven_[currentTank], currentTank));
 		}
 		if (shotAt_.find(currentTank) != shotAt_.end())
 		{
-			shotAtSorted.insert(std::pair<float, Tank *>(
+			shotAtSorted.insert(std::pair<const float, Tank *>(
 				shotAt_[currentTank], currentTank));
 		}
-		distanceSorted.insert(std::pair<float, Tank *>(
+		distanceSorted.insert(std::pair<const float, Tank *>(
 			((currentTank->getPosition().getTankPosition() - 
 			thisTank->getPosition().getTankPosition()).Magnitude()).asFloat(), currentTank));
 	}
@@ -233,7 +233,7 @@
 			damagetakenScore * damagetaken_ +
 			damagedoneScore * damagedone_ +
 			distanceScore * distance_;
-		weightedTanks.insert(std::pair<float, Tank *>(weight, currentTank));
+		weightedTanks.insert(std::pair<const float, Tank *>(weight, currentTank));
 	}
 
 	// Form the final list
--- scorched/src/server/tankai/TankAIWeaponSets.cpp	Sat May  8 12:58:13 2010
+++ scorched-patch/src/server/tankai/TankAIWeaponSets.cpp	Sat May  8 12:58:07 2010
@@ -126,7 +126,7 @@
 			if (weapon.weaponValid(tankAccessories, lastRound))
 			{
 				potentialWeapons.insert(
-					std::pair<unsigned int, WeaponSetEntry *>
+					std::pair<const unsigned int, WeaponSetEntry *>
 						(weapon.prioritybuy, &weapon));
 			}
 		}
--- scorched/src/server/server/ServerStateEnoughPlayers.cpp	Sat May  8 12:58:32 2010
+++ scorched-patch/src/server/server/ServerStateEnoughPlayers.cpp	Sat May  8 12:58:28 2010
@@ -171,7 +171,7 @@
 		{
 			unsigned int startTime = (unsigned int)
 				current->getScore().getStartTime();
-			ais_.insert(std::pair<unsigned int, unsigned int>
+			ais_.insert(std::pair<const unsigned int, unsigned int>
 				(startTime, current->getPlayerId()));
 		}
 	}
@@ -182,7 +182,7 @@
 		noPlayers > requiredPlayers && aiItor != ais_.rend(); 
 		aiItor++, noPlayers--)
 	{
-		std::pair<unsigned int, unsigned int> item = *aiItor;
+		std::pair<const unsigned int, unsigned int> item = *aiItor;
 		ServerMessageHandler::instance()->destroyPlayer(
 			item.second, "Auto-kick");
 	}
@@ -205,7 +205,7 @@
 		if (current->getDestinationId() == 0 &&
 			!current->getTankAI()->removedPlayer())
 		{
-			ais_.insert(std::pair<std::string, unsigned int>
+			ais_.insert(std::pair<const std::string, unsigned int>
 				(current->getTankAI()->getName(), 
 				current->getPlayerId()));
 		}
--- scorched/src/client/console/ConsoleRules.cpp	Sat May  8 12:58:46 2010
+++ scorched-patch/src/client/console/ConsoleRules.cpp	Sat May  8 12:58:42 2010
@@ -38,7 +38,7 @@
 
 	std::string addName = rule->getName();
 	_strlwr((char *) addName.c_str());
-	rules_.insert(std::pair<std::string, ConsoleRule *>(addName, rule));
+	rules_.insert(std::pair<const std::string, ConsoleRule *>(addName, rule));
 }
 
 void ConsoleRules::removeRule(ConsoleRule *rule) 
@@ -189,7 +189,7 @@
 	{
 		ConsoleRule *rule = itor->second;
 		matchedRules.insert(
-			std::pair<int, ConsoleRule *>(
+			std::pair<const int, ConsoleRule *>(
 				(int) rule->getParams().size(), rule));
 	}
 
--- scorched/src/common/landscapemap/HeightMap.cpp	Sat May  8 12:59:03 2010
+++ scorched-patch/src/common/landscapemap/HeightMap.cpp	Sat May  8 12:58:56 2010
@@ -36,7 +36,7 @@
 	delete [] heightData_;
 }
 
-void HeightMap::create(const int width, const int height)
+void HeightMap::create(int width, int height)
 {
 	width_ = width; 
 	height_ = height;
