--- soundtouch/source/SoundTouch/FIRFilter.cpp-orig	2009-02-24 13:11:31.358424000 -0600
+++ soundtouch/source/SoundTouch/FIRFilter.cpp	2009-02-24 13:11:52.837897000 -0600
@@ -181,7 +181,7 @@ void FIRFilter::setCoefficients(const SA
     assert(length == newLength);
 
     resultDivFactor = uResultDivFactor;
-    resultDivider = (SAMPLETYPE)::pow(2, resultDivFactor);
+    resultDivider = (SAMPLETYPE)::pow(2, (double)resultDivFactor);
 
     delete[] filterCoeffs;
     filterCoeffs = new SAMPLETYPE[length];
