--- soundtouch/include/STTypes.h-orig	2009-02-24 13:15:12.106034000 -0600
+++ soundtouch/include/STTypes.h	2009-02-24 13:15:45.875838000 -0600
@@ -94,7 +94,9 @@ namespace soundtouch
         /// routines compiled for whatever reason, you may disable these optimizations 
         /// to make the library compile.
 
+#ifdef __GNUC__
         #define ALLOW_X86_OPTIMIZATIONS     1
+#endif
 
     #endif
 
--- soundtouch/source/SoundTouch/cpu_detect_x86_gcc.cpp-orig	2009-02-24 13:28:33.344231000 -0600
+++ soundtouch/source/SoundTouch/cpu_detect_x86_gcc.cpp	2009-02-24 13:29:44.333520000 -0600
@@ -45,9 +45,11 @@
 #include "cpu_detect.h"
 #include "STTypes.h"
 
+#ifdef ALLOW_X86_OPTIMIZATIONS
 #ifndef __GNUC__
 #error wrong platform - this source code file is for the GNU C compiler.
 #endif
+#endif
 
 using namespace std;
 
