diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -27,5 +27,5 @@
 	rm -f "$(DESTDIR)$(libdir)/sshnodelay.so"
 
 sshnodelay.so:
-	$(CC) -Wall -W -s --shared -fPIC $(sshnodelay_libs) sshnodelay.c -o sshnodelay.so
+	$(CC) $(CFLAGS) -s -shared -fPIC $(sshnodelay_libs) sshnodelay.c -o sshnodelay.so
 endif
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -4,7 +4,9 @@
 
 AC_PROG_CC
 AM_PROG_CC_C_O
-CFLAGS="$CFLAGS -Wall -W"
+if test "$GCC" = "yes"; then
+	CFLAGS="$CFLAGS -Wall -W"
+fi
 LIBS=
 AC_SEARCH_LIBS(dlsym, [dl])
 sshnodelay_libs=$LIBS
