--- a/generate_config	2004-07-23 15:58:48.000000000 +1000
+++ b/generate_config	2009-11-24 11:02:02.000000000 +1100
@@ -1,10 +1,16 @@
-#!/bin/sh -e
+#!/bin/bash -e
 
 #
 # Figure out the system's mailname
 #
 
-syshostname=`hostname --fqdn`
+if test `uname` == 'SunOS'
+then
+	syshostname=`uname -n`
+else
+	syshostname=`hostname --fqdn`
+fi
+
 if test -f /etc/mailname
 then
 	mailname="`head -1 /etc/mailname`"
