--- sugar-0.87.3/bin/sugar-session-orig	2010-02-02 13:00:28.456268622 -0600
+++ sugar-0.87.3/bin/sugar-session	2010-02-02 13:01:41.633535000 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # Copyright (C) 2006, Red Hat, Inc.
 # Copyright (C) 2009, One Laptop Per Child Association Inc
 #
--- sugar-0.87.3/bin/sugar-launch-orig	2010-02-02 13:00:51.019984553 -0600
+++ sugar-0.87.3/bin/sugar-launch	2010-02-02 13:01:35.536884078 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 # Copyright (C) 2007, Red Hat, Inc.
 #
--- sugar-0.87.3/bin/sugar-activity-orig	2010-02-02 13:00:24.596152944 -0600
+++ sugar-0.87.3/bin/sugar-activity	2010-02-02 13:01:04.303843633 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 
 # Copyright (C) 2006-2008, Red Hat, Inc.
 #
--- sugar-0.87.3/bin/sugar-ui-check-orig	2010-02-02 13:00:43.817912287 -0600
+++ sugar-0.87.3/bin/sugar-ui-check	2010-02-02 13:01:48.145870025 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # Copyright (C) 2008, Red Hat, Inc.
 #
 # This program is free software; you can redistribute it and/or modify
--- sugar-0.87.3/bin/sugar-control-panel-orig	2010-02-02 13:00:19.330422454 -0600
+++ sugar-0.87.3/bin/sugar-control-panel	2010-02-02 13:01:13.024490449 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 # Copyright (C) 2008, Red Hat, Inc.
 #
 # This program is free software; you can redistribute it and/or modify
--- sugar-0.87.3/bin/sugar-install-bundle-orig	2010-02-02 13:00:39.348714698 -0600
+++ sugar-0.87.3/bin/sugar-install-bundle	2010-02-02 13:01:28.864832912 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 import sys
 
 from sugar.bundle.activitybundle import ActivityBundle
--- sugar-0.87.8/data/em.py-orig	2010-03-10 12:41:12.197483500 -0600
+++ sugar-0.87.8/data/em.py	2010-03-10 12:41:17.631671204 -0600
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2.6
 #
 # $Id: //projects/empy/em.py#146 $ $Date: 2003/10/27 $
 
--- sugar-0.87.8/bin/sugar.in-orig	2010-03-10 12:40:57.072700009 -0600
+++ sugar-0.87.8/bin/sugar.in	2010-03-10 12:41:04.691467318 -0600
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 
 if [ "$(id -u)" -eq 0 -o "$(id -ru)" -eq 0 ] ; then
 	echo Refusing to run as root.
