--- telepathy-gabble-0.10.3/src/jingle-factory.c	Wed Oct  6 14:43:30 2010
+++ telepathy-gabble-0.10.3-patch/src/jingle-factory.c	Sat Feb 19 15:14:19 2011
@@ -254,7 +254,23 @@
   GabbleJingleFactoryPrivate *priv = fac->priv;
   LmMessageSubType sub_type;
   LmMessageNode *query_node, *node;
+  const gchar *from = wocky_node_get_attribute (wocky_stanza_get_top_node (message), "from");
 
+  if (from != NULL)
+    {
+      TpBaseConnection *base_conn = TP_BASE_CONNECTION (priv->conn);
+      TpHandleRepoIface *contact_repo = tp_base_connection_get_handles (
+          base_conn, TP_HANDLE_TYPE_CONTACT);
+      TpHandle sender = tp_handle_lookup (contact_repo, from, NULL, NULL);
+
+      if (sender != base_conn->self_handle)
+        {
+          DEBUG ("ignoring jingleinfo from '%s', not ourself nor the server",
+              from);
+          return LM_HANDLER_RESULT_ALLOW_MORE_HANDLERS;
+        }
+    }
+
   query_node = lm_message_node_get_child_with_namespace (
       wocky_stanza_get_top_node (message), "query", NS_GOOGLE_JINGLE_INFO);
 
