--- twolame-0.3.12/configure.ac	2009-07-04 12:45:44.451324467 +0200
+++ twolame-0.3.12-patch/configure.ac	2009-07-04 12:46:24.689967885 +0200
@@ -94,8 +94,8 @@
 	) ]
 )
 			
-AC_SUBST( SNDFILE_CFLAGS )
-AC_SUBST( SNDFILE_LIBS )
+AC_SUBST(SNDFILE_CFLAGS)
+AC_SUBST(SNDFILE_LIBS)
 
 
 
@@ -141,7 +141,7 @@
 else 
 	AC_MSG_WARN([Not building twolame frontend because libsndfile is missing.])
 fi
-AC_SUBST( TWOLAME_BIN )
+AC_SUBST(TWOLAME_BIN)
 
 
 dnl ############## Final Output
