--- viking-1.2.1/src/Makefile.in	2011-06-16 00:25:33.000000000 +0200
+++ viking-1.2.1-patch/src/Makefile.in	2011-07-31 19:57:07.637206964 +0200
@@ -433,7 +433,7 @@
 viking_SOURCES = main.c
 LDADD = libviking.a $(PACKAGE_LIBS) @EXPAT_LIBS@ @LIBCURL@ \
 	icons/libicons.a $(am__append_8)
-AM_CFLAGS = -Wall -g -D_GNU_SOURCE \
+AM_CFLAGS = -g -D_GNU_SOURCE \
 	$(PACKAGE_CFLAGS) \
 	@EXPAT_CFLAGS@ @LIBCURL_CPPFLAGS@
 
--- viking-1.2.1/src/icons/Makefile.in	2011-06-16 00:25:33.000000000 +0200
+++ viking-1.2.1-patch/src/icons/Makefile.in	2011-07-31 19:56:59.408917744 +0200
@@ -404,7 +404,7 @@
 CLEANFILES = $(BUILT_SOURCES)
 noinst_LIBRARIES = libicons.a
 libicons_a_SOURCES = icons.c icons.h
-AM_CFLAGS = -Wall -g -D_GNU_SOURCE \
+AM_CFLAGS = -g -D_GNU_SOURCE \
 	$(PACKAGE_CFLAGS)
 
 iconsdir = $(datadir)/icons/hicolor/48x48/apps
