--- vlc-1.0.5/modules/gui/skins2/utils/var_tree.hpp.orig	2009-12-20 18:43:40.000000000 +0100
+++ vlc-1.0.5/modules/gui/skins2/utils/var_tree.hpp	2010-03-15 14:03:05.174409000 +0100
@@ -25,6 +25,15 @@
 #ifndef VAR_TREE_HPP
 #define VAR_TREE_HPP
 
+// TODO : patch me
+// maybe a bug in vlc and/or the libc++ of gcc433
+// that is just to undef the symbol _GLIBCXX_CONCEPT_CHECKS, because the Solaris gcc433
+// is compiled with --enable-concept-checks
+
+#include <bits/c++config.h>
+#undef _GLIBCXX_CONCEPT_CHECKS
+
+
 #include <list>
 
 #include "variable.hpp"
