From GIT-Snapshot 20111203
vlc-1.1-f536e59/xcb

diff -ur vlc-1.1.12/modules/video_output/xcb/xvideo.c vlc-1.1-f536e59/xcb/xvideo.c
--- vlc-1.1.12/modules/video_output/xcb/xvideo.c	2011-12-04 00:07:16.982053258 +0100
+++ vlc-1.1.12/modules/video_output/xcb/xvideo.c	2011-12-03 23:52:43.000000000 +0100
@@ -134,9 +134,6 @@
 static vlc_fourcc_t ParseFormat (vout_display_t *vd,
                                  const xcb_xv_image_format_info_t *restrict f)
 {
-    if (f->byte_order != ORDER && f->bpp != 8)
-        return 0; /* Argh! */
-
     switch (f->type)
     {
       case XCB_XV_IMAGE_FORMAT_INFO_TYPE_RGB:
@@ -156,6 +153,8 @@
                     return VLC_CODEC_RGB24;
                 break;
               case 16:
+                if (f->byte_order != ORDER)
+                    return 0; /* Mixed endian! */
                 if (f->depth == 16)
                     return VLC_CODEC_RGB16;
                 if (f->depth == 15)
@@ -399,6 +398,7 @@
                 continue; /* OMAP framebuffer sucks at YUV 4:2:0 */
             }
 
+            free (p_sys->att);
             xfmt = FindFormat (vd, chroma, &fmt, a->base_id, r, &p_sys->att);
             if (xfmt != NULL)
             {
@@ -415,6 +415,8 @@
                 }
                 break;
             }
+            else
+                p_sys->att = NULL;
         }
         free (r);
         if (xfmt == NULL) /* No acceptable image formats */
@@ -509,7 +511,7 @@
         msg_Err (vd, "no available XVideo adaptor");
         goto error;
     }
-    else
+    /* Compute video (window) placement within the parent window */
     {
         xcb_map_window (conn, p_sys->window);
 
@@ -535,6 +537,15 @@
     xcb_create_gc (conn, p_sys->gc, p_sys->window, 0, NULL);
     msg_Dbg (vd, "using X11 graphic context 0x%08"PRIx32, p_sys->gc);
 
+    /* Disable color keying if applicable */
+    {
+        xcb_intern_atom_reply_t *r =
+            xcb_intern_atom_reply (conn,
+                xcb_intern_atom (conn, 1, 21, "XV_AUTOPAINT_COLORKEY"), NULL);
+        if (r != NULL && r->atom != 0)
+            xcb_xv_set_port_attribute(conn, p_sys->port, r->atom, 1);
+    }
+
     /* Create cursor */
     p_sys->cursor = CreateBlankCursor (conn, screen);
 
