*** vlc-1.1.12/modules/gui/skins2/x11/x11_factory.cpp.orig	Wed Oct 19 15:45:28 2011
--- vlc-1.1.12/modules/gui/skins2/x11/x11_factory.cpp	Wed Oct 19 15:52:52 2011
***************
*** 211,217 ****
      struct
      {
          struct dirent ent;
!         char buf[NAME_MAX + 1];
      } buf;
      struct dirent *file;
      DIR *dir;
--- 211,217 ----
      struct
      {
          struct dirent ent;
!         char buf[8192 + 1]; 
      } buf; 
      struct dirent *file;
      DIR *dir;
