*** vlc-1.1.12/modules/audio_output/oss.c.orig	Tue Jul 12 19:11:24 2011
--- vlc-1.1.12/modules/audio_output/oss.c	Wed Oct 19 16:06:08 2011
***************
*** 32,37 ****
--- 32,40 ----
  #endif
  
  #include <fcntl.h>                                       /* open(), O_WRONLY */
+ #ifdef sun
+ #include <sys/file.h>
+ #endif
  #include <sys/ioctl.h>                                            /* ioctl() */
  #include <unistd.h>                                      /* write(), close() */
  
