diff --git a/modules/audio_output/pulse.c b/modules/audio_output/pulse.c
index 8f48bf2..8b3fde8 100644
--- vlc-1.1.12/modules/audio_output/pulse.c
+++ vlc-1.1.12/modules/audio_output/pulse.c
@@ -837,6 +837,7 @@ static int Open(vlc_object_t *obj)
         formatv[formatc]->encoding = encoding;
         pa_format_info_set_rate(formatv[formatc], ss.rate);
         pa_format_info_set_channels(formatv[formatc], ss.channels);
+        pa_format_info_set_channel_map(formatv[formatc], &map);
         formatc++;
     }

@@ -846,6 +847,7 @@ static int Open(vlc_object_t *obj)
