diff -burNp waf-1.5.13/waf-light waf-1.5.13-dud/waf-light
--- waf-1.5.13/waf-light	2010-02-16 22:42:45.000000000 +0100
+++ waf-1.5.13-dud/waf-light	2010-03-02 23:37:42.640873759 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 # encoding: ISO8859-1
 # Thomas Nagy, 2005-2010
 
@@ -127,7 +127,7 @@ def find_lib():
 		if w: return w
 		err("waf-light requires wafadmin -> export WAFDIR=/folder")
 
-	dir = "/lib/%s-%s-%s/" % (WAF, VERSION, REVISION)
+	dir = "/lib/waf/"
 	for i in [INSTALL,'/usr','/usr/local','/opt']:
 		w = test(i+dir)
 		if w: return w
--- a/wscript	2010-05-05 03:57:41.990709182 +0800
+++ b/wscript	2010-05-05 03:59:10.109421594 +0800
@@ -340,7 +340,7 @@
 		if val != True and val != "y": sys.exit(1)
 		create_waf()
 
-	dir = os.path.join('lib', 'waf-%s-%s' % (VERSION, REVISION), 'wafadmin')
+	dir = os.path.join('lib', 'waf', 'wafadmin')
 
 	wafadmin = bld(features = 'py')
 	wafadmin.find_sources_in_dirs('wafadmin', exts=['.py'])
