diff -r d2a07bdc6744 WebCore/dom/NodeRenderStyle.h
--- a/WebCore/dom/NodeRenderStyle.h	Tue May 18 19:09:56 2010 +0800
+++ b/WebCore/dom/NodeRenderStyle.h	Tue May 18 19:43:13 2010 +0800
@@ -33,7 +33,7 @@
 
 inline RenderStyle* Node::renderStyle() const
 {
-    return m_renderer ? m_renderer->style() : nonRendererRenderStyle();
+    return m_renderer ? (RenderStyle*)m_renderer->style() : nonRendererRenderStyle();
 }
 
 }
diff -r d2a07bdc6744 WebCore/loader/DocumentLoader.h
--- a/WebCore/loader/DocumentLoader.h	Tue May 18 19:09:56 2010 +0800
+++ b/WebCore/loader/DocumentLoader.h	Tue May 18 19:43:13 2010 +0800
@@ -163,7 +163,7 @@
         String clientRedirectDestinationForHistory() const { return urlForHistory(); }
         void setClientRedirectSourceForHistory(const String& clientedirectSourceForHistory) { m_clientRedirectSourceForHistory = clientedirectSourceForHistory; }
         
-        String serverRedirectSourceForHistory() const { return urlForHistory() == url() ? String() : urlForHistory(); } // null if no server redirect occurred.
+        String serverRedirectSourceForHistory() const { return urlForHistory() == url() ? String() : (String)urlForHistory(); } // null if no server redirect occurred.
         String serverRedirectDestinationForHistory() const { return url(); }
 
         bool didCreateGlobalHistoryEntry() const { return m_didCreateGlobalHistoryEntry; }
diff -r d2a07bdc6744 WebCore/loader/FrameLoader.cpp
--- a/WebCore/loader/FrameLoader.cpp	Tue May 18 19:09:56 2010 +0800
+++ b/WebCore/loader/FrameLoader.cpp	Tue May 18 19:43:13 2010 +0800
@@ -1988,7 +1988,7 @@
     m_loadType = FrameLoadTypeStandard;
     RefPtr<DocumentLoader> loader = m_client->createDocumentLoader(request, substituteData);
     if (lockHistory && m_documentLoader)
-        loader->setClientRedirectSourceForHistory(m_documentLoader->didCreateGlobalHistoryEntry() ? m_documentLoader->urlForHistory() : m_documentLoader->clientRedirectSourceForHistory());
+        loader->setClientRedirectSourceForHistory(m_documentLoader->didCreateGlobalHistoryEntry() ? (String)m_documentLoader->urlForHistory() : m_documentLoader->clientRedirectSourceForHistory());
     load(loader.get());
 }
 
@@ -2012,7 +2012,7 @@
 {
     RefPtr<DocumentLoader> loader = m_client->createDocumentLoader(request, SubstituteData());
     if (lockHistory && m_documentLoader)
-        loader->setClientRedirectSourceForHistory(m_documentLoader->didCreateGlobalHistoryEntry() ? m_documentLoader->urlForHistory() : m_documentLoader->clientRedirectSourceForHistory());
+        loader->setClientRedirectSourceForHistory(m_documentLoader->didCreateGlobalHistoryEntry() ? (String)m_documentLoader->urlForHistory() : m_documentLoader->clientRedirectSourceForHistory());
 
     loader->setTriggeringAction(action);
     if (m_documentLoader)
