--- webkit-1.2.7/WebCore/platform/gtk/PopupMenuGtk.cpp	Fri Sep 10 15:20:33 2010
+++ webkit-1.2.7-patch/WebCore/platform/gtk/PopupMenuGtk.cpp	Sat Mar 19 23:27:28 2011
@@ -55,7 +55,7 @@
         m_popup = GTK_MENU(gtk_menu_new());
         g_signal_connect(m_popup.get(), "unmap", G_CALLBACK(menuUnmapped), this);
     } else
-        gtk_container_foreach(GTK_CONTAINER(m_popup.get()), reinterpret_cast<GtkCallback>(menuRemoveItem), this);
+        gtk_container_foreach(GTK_CONTAINER(m_popup.get()), (GtkCallback)(menuRemoveItem), this);
 
     int x, y;
     gdk_window_get_origin(gtk_widget_get_window(GTK_WIDGET(view->hostWindow()->platformPageClient())), &x, &y);
@@ -107,7 +107,7 @@
             m_menuPosition.setY(m_menuPosition.y() - rect.height() / 2);
 
     g_list_free(children);
-    gtk_menu_popup(m_popup.get(), 0, 0, reinterpret_cast<GtkMenuPositionFunc>(menuPositionFunction), this, 0, gtk_get_current_event_time());
+    gtk_menu_popup(m_popup.get(), 0, 0, (GtkMenuPositionFunc)(menuPositionFunction), this, 0, gtk_get_current_event_time());
 }
 
 void PopupMenu::hide()
--- webkit-1.2.7/WebCore/platform/network/soup/ResourceHandleSoup.cpp	Fri Sep 10 15:20:33 2010
+++ webkit-1.2.7-patch/WebCore/platform/network/soup/ResourceHandleSoup.cpp	Sat Mar 19 23:26:06 2011
@@ -542,7 +542,7 @@
                                                                         g_mapped_file_get_length(fileMapping),
                                                                         fileMapping,
 #if GLIB_CHECK_VERSION(2, 21, 3)
-                                                                        reinterpret_cast<GDestroyNotify>(g_mapped_file_unref));
+                                                                        (GDestroyNotify)(g_mapped_file_unref));
 #else
                                                                         reinterpret_cast<GDestroyNotify>(g_mapped_file_free));
 #endif
