diff -r d2a07bdc6744 JavaScriptCore/jsc.cpp
--- a/JavaScriptCore/jsc.cpp	Tue May 18 19:09:56 2010 +0800
+++ b/JavaScriptCore/jsc.cpp	Tue May 18 19:38:43 2010 +0800
@@ -304,6 +304,10 @@
     // Without this, Visual Studio will complain that this method does not return a value.
     return jsUndefined();
 #endif
+
+#if COMPILER(SUNCC)
+    return jsUndefined();
+#endif
 }
 
 // Use SEH for Release builds only to get rid of the crash report dialog
diff -r d2a07bdc6744 JavaScriptCore/wtf/FastMalloc.cpp
--- a/JavaScriptCore/wtf/FastMalloc.cpp	Tue May 18 19:09:56 2010 +0800
+++ b/JavaScriptCore/wtf/FastMalloc.cpp	Tue May 18 19:38:43 2010 +0800
@@ -1495,7 +1495,7 @@
 void* TCMalloc_PageHeap::runScavengerThread(void* context)
 {
   static_cast<TCMalloc_PageHeap*>(context)->scavengerThread();
-#if COMPILER(MSVC)
+#if COMPILER(MSVC) || COMPILER(SUNCC)
   // Without this, Visual Studio will complain that this method does not return a value.
   return 0;
 #endif
diff -r 7f0899abd82d JavaScriptCore/wtf/FastMalloc.cpp
--- a/JavaScriptCore/wtf/FastMalloc.cpp	Fri May 21 01:19:20 2010 +0800
+++ b/JavaScriptCore/wtf/FastMalloc.cpp	Fri May 21 01:20:53 2010 +0800
@@ -3506,8 +3506,8 @@
   }
   if (!ret) {
 #ifdef WTF_CHANGES
-    if (crashOnFailure) // This branch should be optimized out by the compiler.
-        CRASH();
+    //if (crashOnFailure) // This branch should be optimized out by the compiler.
+     //   CRASH();
 #else
     errno = ENOMEM;
 #endif
