--- webkit-1.2.7/GNUmakefile.am	Fri Sep 10 15:20:23 2010
+++ webkit-1.2.7-patch/GNUmakefile.am	Sat Mar 19 22:47:21 2011
@@ -95,7 +95,6 @@
 # Don't include them for now to reduce the noise when compiling
 # $(GLIB_CFLAGS) $(CAIRO_CFLAGS) $(PANGO_CFLAGS)  $(FREETYPE_CFLAGS)
 corekit_cflags += \
-	-fno-strict-aliasing \
 	$(COVERAGE_CFLAGS) \
 	$(ENCHANT_CFLAGS) \
 	$(GAIL_CFLAGS) \
@@ -126,17 +125,10 @@
 endif
 
 # Default compiler flags
-global_cflags += \
-	-Wall -W -Wcast-align -Wchar-subscripts -Wreturn-type \
-	-Wformat -Wformat-security -Wno-format-y2k -Wundef \
-	-Wmissing-format-attribute -Wpointer-arith -Wwrite-strings \
-	-Wno-unused-parameter -Wno-parentheses \
-	-fno-exceptions
+global_cflags += -mt
 
 global_cxxflags += \
-	$(global_cflags) \
-	$(SYMBOL_VISIBILITY_INLINES) \
-	-fno-rtti
+	$(global_cflags)
 
 # -no-undefined required for building DLLs on Windows
 # It breaks the build on other platforms, so we use it conditionally
@@ -188,8 +180,7 @@
 	$(libJavaScriptCore_la_CFLAGS)
 
 libJavaScriptCore_la_CFLAGS = \
-	-fstrict-aliasing \
-	-O3 \
+	-xO4 \
 	$(global_cflags) \
 	$(GLIB_CFLAGS) \
 	$(UNICODE_CFLAGS)
@@ -283,7 +274,7 @@
 
 if !ENABLE_DEBUG
 global_cppflags += -DNDEBUG
-global_cflags += $(SYMBOL_VISIBILITY)
+global_cflags += -xldscope==hidden
 else
 #webcoregtk_cppflags += \
 #	-DG_DISABLE_DEPRECATED \
@@ -544,7 +535,6 @@
 noinst_PROGRAMS += $(TEST_PROGS)
 
 webkit_tests_cflags = \
-	-fno-strict-aliasing \
 	-I$(srcdir)/JavaScriptCore/ForwardingHeaders \
 	-I$(srcdir)/WebKit/gtk \
 	-I$(top_builddir)/WebKit/gtk \
@@ -559,9 +549,7 @@
 	$(GLIB_LIBS) \
 	$(LIBSOUP_LIBS)
 
-webkit_tests_ldflags = \
-	-no-install \
-	-no-fast-install
+webkit_tests_ldflags =
 
 TEST_PROGS += Programs/unittests/testhttpbackend \
 	Programs/unittests/testloading \
--- webkit-1.2.7/GNUmakefile.in	Mon Feb  7 18:43:44 2011
+++ webkit-1.2.7-patch/GNUmakefile.in	Sat Mar 19 22:48:10 2011
@@ -99,7 +99,7 @@
 @ENABLE_FAST_MALLOC_FALSE@	-DUSE_SYSTEM_MALLOC
 
 @ENABLE_DEBUG_FALSE@am__append_5 = -DNDEBUG
-@ENABLE_DEBUG_FALSE@am__append_6 = $(SYMBOL_VISIBILITY)
+@ENABLE_DEBUG_FALSE@am__append_6 = -xldscope=hidden
 #webcoregtk_cppflags += \
 #	-DG_DISABLE_DEPRECATED \
 #	-DGDK_PIXBUF_DISABLE_DEPRECATED \
@@ -6360,7 +6360,7 @@
 INTROSPECTION_LIBS = @INTROSPECTION_LIBS@
 JPEG_LIBS = @JPEG_LIBS@
 LD = @LD@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS@ -Wl,-z -Wl,muldefs
 LIBOBJS = @LIBOBJS@
 LIBS = @LIBS@
 LIBSOUP_2_29_90_CFLAGS = @LIBSOUP_2_29_90_CFLAGS@
@@ -6409,8 +6409,8 @@
 SQLITE3_CFLAGS = @SQLITE3_CFLAGS@
 SQLITE3_LIBS = @SQLITE3_LIBS@
 STRIP = @STRIP@
-SYMBOL_VISIBILITY = @SYMBOL_VISIBILITY@
-SYMBOL_VISIBILITY_INLINES = @SYMBOL_VISIBILITY_INLINES@
+SYMBOL_VISIBILITY =
+SYMBOL_VISIBILITY_INLINES =
 UNICODE_CFLAGS = @UNICODE_CFLAGS@
 UNICODE_LIBS = @UNICODE_LIBS@
 VERSION = @VERSION@
@@ -8878,13 +8878,9 @@
 	$(am__append_5) $(am__append_7) $(am__append_44)
 
 # Default compiler flags
-global_cflags := -Wall -W -Wcast-align -Wchar-subscripts -Wreturn-type \
-	-Wformat -Wformat-security -Wno-format-y2k -Wundef \
-	-Wmissing-format-attribute -Wpointer-arith -Wwrite-strings \
-	-Wno-unused-parameter -Wno-parentheses -fno-exceptions \
+global_cflags := \
 	$(am__append_6)
-global_cxxflags := $(global_cflags) $(SYMBOL_VISIBILITY_INLINES) \
-	-fno-rtti
+global_cxxflags := $(global_cflags)
 
 # CFLAGS/CXXFLAGS used by WebCore and WebKit
 #
@@ -8891,7 +8887,7 @@
 # gtk+.pc already include glib, cairo, freetype and pango CFLAGS
 # Don't include them for now to reduce the noise when compiling
 # $(GLIB_CFLAGS) $(CAIRO_CFLAGS) $(PANGO_CFLAGS)  $(FREETYPE_CFLAGS)
-corekit_cflags := -fno-strict-aliasing $(COVERAGE_CFLAGS) \
+corekit_cflags := $(COVERAGE_CFLAGS) \
 	$(ENCHANT_CFLAGS) $(GAIL_CFLAGS) $(GEOCLUE_CFLAGS) \
 	$(GLIB_CFLAGS) $(GSTREAMER_CFLAGS) $(GTK_CFLAGS) \
 	$(HILDON_CFLAGS) $(LIBSOUP_CFLAGS) $(LIBXML_CFLAGS) \
@@ -8994,8 +8990,6 @@
 	$(libJavaScriptCore_la_CFLAGS)
 
 libJavaScriptCore_la_CFLAGS = \
-	-fstrict-aliasing \
-	-O3 \
 	$(global_cflags) \
 	$(GLIB_CFLAGS) \
 	$(UNICODE_CFLAGS)
@@ -9096,8 +9090,6 @@
 	$(javascriptcore_cppflags)
 
 Programs_minidom_CFLAGS = \
-	-ansi \
-	-fno-strict-aliasing \
 	$(global_cflags) \
 	$(GLOBALDEPS_CFLAGS)
 
@@ -9104,11 +9096,10 @@
 Programs_minidom_LDADD = \
 	 libJavaScriptCore.la \
 	 -lm \
-	 -lstdc++
+	 -lCstd \
+	 -lCrun
 
-Programs_minidom_LDFLAGS = \
-	-no-install \
-	-no-fast-install
+Programs_minidom_LDFLAGS =
 
 
 # jsc
@@ -9120,7 +9111,6 @@
 	$(javascriptcore_cppflags)
 
 Programs_jsc_CXXFLAGS = \
-	-fno-strict-aliasing \
 	$(global_cxxflags) \
 	$(global_cflags) \
 	$(GLOBALDEPS_CFLAGS) \
@@ -9254,8 +9244,6 @@
 	WebKitTools/GtkLauncher/main.c
 
 Programs_GtkLauncher_CFLAGS = \
-	-ansi \
-	-fno-strict-aliasing \
 	$(global_cflags) \
 	$(GTK_CFLAGS) \
 	$(LIBSOUP_CFLAGS)
@@ -9265,9 +9253,7 @@
 	$(GTK_LIBS) \
 	$(GLIB_LIBS)
 
-Programs_GtkLauncher_LDFLAGS = \
-	-no-fast-install \
-	-no-install
+Programs_GtkLauncher_LDFLAGS =
 
 
 # DumpRenderTree
@@ -9310,7 +9296,6 @@
 	$(Programs_DumpRenderTree_CFLAGS)
 
 Programs_DumpRenderTree_CFLAGS = \
-	-fno-strict-aliasing \
 	$(dumprendertree_cppflags) \
 	$(global_cflags) \
 	$(GLOBALDEPS_CFLAGS) \
@@ -9328,9 +9313,7 @@
 	$(LIBSOUP_LIBS) \
 	$(FREETYPE_LIBS)
 
-Programs_DumpRenderTree_LDFLAGS = \
-	-no-fast-install \
-	-no-install
+Programs_DumpRenderTree_LDFLAGS =
 
 @TARGET_X11_TRUE@TestNetscapePlugin_libtestnetscapeplugin_la_CPPFLAGS = \
 @TARGET_X11_TRUE@	-I$(srcdir)/WebKitTools/DumpRenderTree \
@@ -9364,7 +9347,6 @@
 MSGMERGE := $(shell which msgmerge)
 MSGMERGE_UPDATE := $(MSGMERGE) --update
 webkit_tests_cflags = \
-	-fno-strict-aliasing \
 	-I$(srcdir)/JavaScriptCore/ForwardingHeaders \
 	-I$(srcdir)/WebKit/gtk \
 	-I$(top_builddir)/WebKit/gtk \
--- webkit-1.2.7/JavaScriptCore/GNUmakefile.am	Fri Sep 10 15:20:23 2010
+++ webkit-1.2.7-patch/JavaScriptCore/GNUmakefile.am	Sat Mar 19 22:46:12 2011
@@ -554,8 +554,6 @@
 	$(javascriptcore_cppflags)
 
 Programs_minidom_CFLAGS = \
-	-ansi \
-	-fno-strict-aliasing \
 	$(global_cflags) \
 	$(GLOBALDEPS_CFLAGS)
 
@@ -562,11 +560,9 @@
 Programs_minidom_LDADD = \
 	 libJavaScriptCore.la \
 	 -lm \
-	 -lstdc++
+	 -lCstd
 
-Programs_minidom_LDFLAGS = \
-	-no-install \
-	-no-fast-install
+Programs_minidom_LDFLAGS =
 
 # jsc
 Programs_jsc_SOURCES = \
@@ -577,7 +573,6 @@
 	$(javascriptcore_cppflags)
 
 Programs_jsc_CXXFLAGS = \
-	-fno-strict-aliasing \
 	$(global_cxxflags) \
 	$(global_cflags) \
 	$(GLOBALDEPS_CFLAGS) \
--- webkit-1.2.7/configure.ac	Mon Feb  7 18:38:11 2011
+++ webkit-1.2.7-patch/configure.ac	Sat Mar 19 22:46:12 2011
@@ -666,11 +666,11 @@
 
 # Add the appropriate 'O' level for optimized builds
 if test "$enable_optimizations" = "yes"; then
-   CXXFLAGS="$CXXFLAGS -O2"
-   CFLAGS="$CFLAGS -O2"
+   CXXFLAGS="$CXXFLAGS -xO4"
+   CFLAGS="$CFLAGS -xO4"
 else
-   CXXFLAGS="$CXXFLAGS -O0"
-   CFLAGS="$CFLAGS -O0"
+   CXXFLAGS="$CXXFLAGS -xO0"
+   CFLAGS="$CFLAGS -xO0"
 fi
 
 PKG_CHECK_MODULES([LIBSOUP],
