--- webkit-1.2.7/WebCore/rendering/RenderListMarker.cpp	Fri Sep 10 15:20:33 2010
+++ webkit-1.2.7-patch/WebCore/rendering/RenderListMarker.cpp	Sat Mar 19 23:10:38 2011
@@ -116,7 +116,7 @@
     return String(&letters[lettersSize - length], length);
 }
 
-static String toAlphabetic(int number, const UChar* alphabet, int alphabetSize)
+static inline String toAlphabetic(int number, const UChar* alphabet, int alphabetSize)
 {
     if (number < 1)
         return String::number(number);
@@ -124,7 +124,7 @@
     return toAlphabeticOrNumeric(number, alphabet, alphabetSize, AlphabeticSequence);
 }
 
-static String toNumeric(int number, const UChar* numerals, int numeralsSize)
+static inline String toNumeric(int number, const UChar* numerals, int numeralsSize)
 {
     return toAlphabeticOrNumeric(number, numerals, numeralsSize, NumericSequence);
 }
