--- webkit-1.2.7/WebCore/platform/gtk/LocalizedStringsGtk.cpp	Fri Sep 10 15:20:33 2010
+++ webkit-1.2.7-patch/WebCore/platform/gtk/LocalizedStringsGtk.cpp	Sat Mar 19 23:08:30 2011
@@ -457,7 +457,7 @@
 
 String localizedMediaTimeDescription(float time)
 {
-    if (!isfinite(time))
+    if (isnan(time) || isnan(0*time))
         return String::fromUTF8(_("indefinite time"));
 
     int seconds = (int)fabsf(time);
