--- webkit-1.2.7/JavaScriptCore/wtf/MathExtras.h	Fri Sep 10 15:20:23 2010
+++ webkit-1.2.7-patch/JavaScriptCore/wtf/MathExtras.h	Sat Mar 19 23:28:55 2011
@@ -32,6 +32,7 @@
 
 #if OS(SOLARIS)
 #include <ieeefp.h>
+#include <math.h>
 #endif
 
 #if OS(OPENBSD)
@@ -188,7 +189,7 @@
 inline float rad2grad(float r) { return r * 200.0f / piFloat; }
 inline float grad2rad(float g) { return g * piFloat / 200.0f; }
 
-#if !COMPILER(MSVC) && !COMPILER(RVCT) && !OS(ANDROID) && !COMPILER(WINSCW)
+#if !COMPILER(MSVC) && !COMPILER(RVCT) && !OS(ANDROID) && !COMPILER(WINSCW) && !COMPILER(SUNCC)
 using std::isfinite;
 using std::isinf;
 using std::isnan;
