diff -r d2a07bdc6744 JavaScriptCore/runtime/SmallStrings.cpp
--- a/JavaScriptCore/runtime/SmallStrings.cpp	Tue May 18 19:09:56 2010 +0800
+++ b/JavaScriptCore/runtime/SmallStrings.cpp	Tue May 18 19:51:11 2010 +0800
@@ -61,7 +61,7 @@
 
 SmallStrings::SmallStrings()
 {
-    COMPILE_ASSERT(numCharactersToStore == arrayLength(m_singleCharacterStrings), IsNumCharactersConstInSyncWithClassUsage);
+    //COMPILE_ASSERT(numCharactersToStore == arrayLength(m_singleCharacterStrings), IsNumCharactersConstInSyncWithClassUsage);
     clear();
 }
 
diff -r d2a07bdc6744 JavaScriptCore/wtf/Vector.h
--- a/JavaScriptCore/wtf/Vector.h	Tue May 18 19:09:56 2010 +0800
+++ b/JavaScriptCore/wtf/Vector.h	Tue May 18 19:51:11 2010 +0800
@@ -948,7 +948,7 @@
         // one is needed. 
         new (end()) T(static_cast<T>(*ptr));
 #else
-        new (end()) T(*ptr);
+	new (end()) T(*(T*)ptr);
 #endif
         ++m_size;
     }
