diff -r 29063d7baea0 GNUmakefile.am
--- a/GNUmakefile.am	Tue Dec 29 16:38:16 2009 +0800
+++ b/GNUmakefile.am	Tue Dec 29 16:39:12 2009 +0800
@@ -94,7 +94,6 @@
 # Don't include them for now to reduce the noise when compiling
 # $(GLIB_CFLAGS) $(CAIRO_CFLAGS) $(PANGO_CFLAGS)  $(FREETYPE_CFLAGS)
 corekit_cflags += \
-	-fno-strict-aliasing \
 	$(COVERAGE_CFLAGS) \
 	$(ENCHANT_CFLAGS) \
 	$(GAIL_CFLAGS) \
@@ -125,17 +124,9 @@
 endif
 
 # Default compiler flags
-global_cflags += \
-	-Wall -W -Wcast-align -Wchar-subscripts -Wreturn-type \
-	-Wformat -Wformat-security -Wno-format-y2k -Wundef \
-	-Wmissing-format-attribute -Wpointer-arith -Wwrite-strings \
-	-Wno-unused-parameter -Wno-parentheses \
-	-fno-exceptions
+global_cflags :=
 
-global_cxxflags += \
-	$(global_cflags) \
-	$(SYMBOL_VISIBILITY_INLINES) \
-	-fno-rtti
+global_cxxflags :=
 
 # -no-undefined required for building DLLs on Windows
 # It breaks the build on other platforms, so we use it conditionally
@@ -186,8 +177,7 @@
 	$(libJavaScriptCore_la_CFLAGS)
 
 libJavaScriptCore_la_CFLAGS = \
-	-fstrict-aliasing \
-	-O3 \
+	-xO3 \
 	$(global_cflags) \
 	$(GLIB_CFLAGS) \
 	$(UNICODE_CFLAGS)
@@ -306,7 +296,6 @@
 
 if !ENABLE_DEBUG
 global_cppflags += -DNDEBUG
-global_cflags += $(SYMBOL_VISIBILITY)
 
 else
 webcoregtk_cppflags += \
@@ -561,7 +550,6 @@
 noinst_PROGRAMS += $(TEST_PROGS)
 
 webkit_tests_cflags = \
-	-fno-strict-aliasing \
 	-I$(srcdir)/JavaScriptCore/ForwardingHeaders \
 	-I$(srcdir)/WebKit/gtk \
 	-I$(top_builddir)/WebKit/gtk \
diff -r 29063d7baea0 JavaScriptCore/GNUmakefile.am
--- a/JavaScriptCore/GNUmakefile.am	Tue Dec 29 16:38:16 2009 +0800
+++ b/JavaScriptCore/GNUmakefile.am	Tue Dec 29 16:39:12 2009 +0800
@@ -552,15 +552,13 @@
 	$(javascriptcore_cppflags)
 
 Programs_minidom_CFLAGS = \
-	-ansi \
-	-fno-strict-aliasing \
 	$(global_cflags) \
 	$(GLOBALDEPS_CFLAGS)
 
 Programs_minidom_LDADD = \
 	 libJavaScriptCore.la \
 	 -lm \
-	 -lstdc++
+	 -lstdcxx4
 
 Programs_minidom_LDFLAGS = \
 	-no-install \
@@ -575,7 +573,6 @@
 	$(javascriptcore_cppflags)
 
 Programs_jsc_CXXFLAGS = \
-	-fno-strict-aliasing \
 	$(global_cxxflags) \
 	$(global_cflags) \
 	$(GLOBALDEPS_CFLAGS) \
diff -r 29063d7baea0 JavaScriptCore/interpreter/RegisterFile.cpp
--- a/JavaScriptCore/interpreter/RegisterFile.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/JavaScriptCore/interpreter/RegisterFile.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -34,7 +34,7 @@
 RegisterFile::~RegisterFile()
 {
 #if HAVE(MMAP)
-    munmap(m_buffer, ((m_max - m_start) + m_maxGlobals) * sizeof(Register));
+    munmap((char *)m_buffer, ((m_max - m_start) + m_maxGlobals) * sizeof(Register));
 #elif HAVE(VIRTUALALLOC)
 #if PLATFORM(WINCE)
     VirtualFree(m_buffer, DWORD(m_commitEnd) - DWORD(m_buffer), MEM_DECOMMIT);
diff -r 29063d7baea0 JavaScriptCore/jsc.cpp
--- a/JavaScriptCore/jsc.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/JavaScriptCore/jsc.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -294,6 +294,7 @@
 {
     cleanupGlobalData(&exec->globalData());
     exit(EXIT_SUCCESS);
+    return jsNull();
 }
 
 // Use SEH for Release builds only to get rid of the crash report dialog
diff -r 29063d7baea0 JavaScriptCore/pcre/pcre_exec.cpp
--- a/JavaScriptCore/pcre/pcre_exec.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/JavaScriptCore/pcre/pcre_exec.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -1821,8 +1821,8 @@
         case 24: goto RRETURN_24;
         case 26: goto RRETURN_26;
         case 27: goto RRETURN_27;
-        case 28: goto RRETURN_28;
-        case 29: goto RRETURN_29;
+	  //        case 28: goto RRETURN_28;
+	  //        case 29: goto RRETURN_29;
         case 30: goto RRETURN_30;
         case 31: goto RRETURN_31;
         case 38: goto RRETURN_38;
diff -r 29063d7baea0 JavaScriptCore/wtf/TCSystemAlloc.cpp
--- a/JavaScriptCore/wtf/TCSystemAlloc.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/JavaScriptCore/wtf/TCSystemAlloc.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -193,10 +193,10 @@
 
   // Return the unused memory to the system
   if (adjust > 0) {
-    munmap(reinterpret_cast<void*>(ptr), adjust);
+    munmap(reinterpret_cast<char*>(ptr), adjust);
   }
   if (adjust < extra) {
-    munmap(reinterpret_cast<void*>(ptr + adjust + size), extra - adjust);
+    munmap(reinterpret_cast<char*>(ptr + adjust + size), extra - adjust);
   }
 
   ptr += adjust;
@@ -442,7 +442,7 @@
 
 void TCMalloc_SystemRelease(void* start, size_t length)
 {
-  void* newAddress = mmap(start, length, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_FIXED, -1, 0);
+  void* newAddress = mmap((char *)start, length, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_FIXED, -1, 0);
   // If the mmap failed then that's ok, we just won't return the memory to the system.
   ASSERT_UNUSED(newAddress, newAddress == start || newAddress == reinterpret_cast<void*>(MAP_FAILED));
 }
diff -r 29063d7baea0 JavaScriptCore/wtf/Vector.h
--- a/JavaScriptCore/wtf/Vector.h	Tue Dec 29 16:38:16 2009 +0800
+++ b/JavaScriptCore/wtf/Vector.h	Tue Dec 29 16:39:12 2009 +0800
@@ -38,7 +38,7 @@
     using std::max;
 
     // WTF_ALIGN_OF / WTF_ALIGNED
-    #if COMPILER(GCC) || COMPILER(MINGW) || COMPILER(RVCT) || COMPILER(WINSCW)
+    #if COMPILER(GCC) || COMPILER(MINGW) || COMPILER(RVCT) || COMPILER(WINSCW)|| COMPILER(SUNCC) 
         #define WTF_ALIGN_OF(type) __alignof__(type)
         #define WTF_ALIGNED(variable_type, variable, n) variable_type variable __attribute__((__aligned__(n)))
     #elif COMPILER(MSVC)
@@ -127,7 +127,7 @@
         {
             while (src != srcEnd) {
                 new (dst) T(*src);
-                src->~T();
+                ((T*)src)->~T();
                 ++dst;
                 ++src;
             }
diff -r 29063d7baea0 WebCore/bindings/js/JSJavaScriptCallFrameCustom.cpp
--- a/WebCore/bindings/js/JSJavaScriptCallFrameCustom.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/bindings/js/JSJavaScriptCallFrameCustom.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -48,7 +48,7 @@
 
 JSValue JSJavaScriptCallFrame::thisObject(ExecState*) const
 {
-    return impl()->thisObject() ? impl()->thisObject() : jsNull();
+    return (JSValue)impl()->thisObject() ? (JSValue)impl()->thisObject() : jsNull();
 }
 
 JSValue JSJavaScriptCallFrame::type(ExecState* exec) const
diff -r 29063d7baea0 WebCore/bindings/js/JSPluginElementFunctions.cpp
--- a/WebCore/bindings/js/JSPluginElementFunctions.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/bindings/js/JSPluginElementFunctions.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -62,7 +62,7 @@
     JSHTMLElement* thisObj = static_cast<JSHTMLElement*>(asObject(slot.slotBase()));
     HTMLElement* element = static_cast<HTMLElement*>(thisObj->impl());
     RuntimeObjectImp* runtimeObject = getRuntimeObject(exec, element);
-    return runtimeObject ? runtimeObject : jsUndefined();
+    return (JSValue)runtimeObject ? (JSValue)runtimeObject : jsUndefined();
 }
 
 JSValue runtimeObjectPropertyGetter(ExecState* exec, const Identifier& propertyName, const PropertySlot& slot)
diff -r 29063d7baea0 WebCore/css/make-css-file-arrays.pl
--- a/WebCore/css/make-css-file-arrays.pl	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/css/make-css-file-arrays.pl	Tue Dec 29 16:39:12 2009 +0800
@@ -28,7 +28,7 @@
 GetOptions('preprocessor=s' => \$preprocessor);
 
 if (!$preprocessor) {
-    $preprocessor = "/usr/bin/gcc -E -P -x c++";
+    $preprocessor = "/usr/sfw/bin/gcc -E -P -x c++";
 }
 
 my $header = $ARGV[0];
diff -r 29063d7baea0 WebCore/dom/NodeRenderStyle.h
--- a/WebCore/dom/NodeRenderStyle.h	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/dom/NodeRenderStyle.h	Tue Dec 29 16:39:12 2009 +0800
@@ -33,7 +33,7 @@
 
 inline RenderStyle* Node::renderStyle() const
 {
-    return m_renderer ? m_renderer->style() : nonRendererRenderStyle();
+    return m_renderer ? (RenderStyle*)m_renderer->style() : nonRendererRenderStyle();
 }
 
 }
diff -r 29063d7baea0 WebCore/dom/XMLTokenizerLibxml2.cpp
--- a/WebCore/dom/XMLTokenizerLibxml2.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/dom/XMLTokenizerLibxml2.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -891,7 +891,7 @@
     if (m_parserStopped)
         return;
 
-#if COMPILER(MSVC)
+#if COMPILER(MSVC) || 1
     char m[1024];
     vsnprintf(m, sizeof(m) - 1, message, args);
 #else
diff -r 29063d7baea0 WebCore/editing/CompositeEditCommand.cpp
--- a/WebCore/editing/CompositeEditCommand.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/editing/CompositeEditCommand.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -918,7 +918,7 @@
 
     // FIXME: This is an inefficient way to preserve style on nodes in the paragraph to move. It
     // shouldn't matter though, since moved paragraphs will usually be quite small.
-    RefPtr<DocumentFragment> fragment = startOfParagraphToMove != endOfParagraphToMove ? createFragmentFromMarkup(document(), createMarkup(range.get(), 0, DoNotAnnotateForInterchange, true), "") : 0;
+    RefPtr<DocumentFragment> fragment = startOfParagraphToMove != endOfParagraphToMove ? createFragmentFromMarkup(document(), createMarkup(range.get(), 0, DoNotAnnotateForInterchange, true), "") : (WTF::PassRefPtr<WebCore::DocumentFragment>)0;
     
     // A non-empty paragraph's style is moved when we copy and move it.  We don't move 
     // anything if we're given an empty paragraph, but an empty paragraph can have style
diff -r 29063d7baea0 WebCore/editing/markup.cpp
--- a/WebCore/editing/markup.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/editing/markup.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -967,7 +967,7 @@
     Node* body = enclosingNodeWithTag(Position(commonAncestor, 0), bodyTag);
     // FIXME: Do this for all fully selected blocks, not just the body.
     Node* fullySelectedRoot = body && *VisibleSelection::selectionFromContentsOfNode(body).toNormalizedRange() == *updatedRange ? body : 0;
-    RefPtr<CSSMutableStyleDeclaration> fullySelectedRootStyle = fullySelectedRoot ? styleFromMatchedRulesAndInlineDecl(fullySelectedRoot) : 0;
+    RefPtr<CSSMutableStyleDeclaration> fullySelectedRootStyle = fullySelectedRoot ? styleFromMatchedRulesAndInlineDecl(fullySelectedRoot) : (WTF::PassRefPtr<WebCore::CSSMutableStyleDeclaration>)0;
     if (annotate && fullySelectedRoot) {
         if (shouldIncludeWrapperForFullySelectedRoot(fullySelectedRoot, fullySelectedRootStyle.get()))
             specialCommonAncestor = fullySelectedRoot;
diff -r 29063d7baea0 WebCore/html/HTMLParser.cpp
--- a/WebCore/html/HTMLParser.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/html/HTMLParser.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -137,7 +137,7 @@
     , m_reportErrors(reportErrors)
     , m_handlingResidualStyleAcrossBlocks(false)
     , m_inStrayTableContent(0)
-    , m_parserQuirks(m_document->page() ? m_document->page()->chrome()->client()->createHTMLParserQuirks() : 0)
+    , m_parserQuirks(m_document->page() ? m_document->page()->chrome()->client()->createHTMLParserQuirks() : (WTF::PassOwnPtr<WebCore::HTMLParserQuirks>)0)
 {
 }
 
@@ -155,7 +155,7 @@
     , m_reportErrors(false)
     , m_handlingResidualStyleAcrossBlocks(false)
     , m_inStrayTableContent(0)
-    , m_parserQuirks(m_document->page() ? m_document->page()->chrome()->client()->createHTMLParserQuirks() : 0)
+    , m_parserQuirks(m_document->page() ? m_document->page()->chrome()->client()->createHTMLParserQuirks() : (WTF::PassOwnPtr<WebCore::HTMLParserQuirks>)0)
 {
     if (frag)
         frag->ref();
diff -r 29063d7baea0 WebCore/loader/DocumentLoader.cpp
--- a/WebCore/loader/DocumentLoader.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/loader/DocumentLoader.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -500,7 +500,7 @@
 
 PassRefPtr<Archive> DocumentLoader::popArchiveForSubframe(const String& frameName)
 {
-    return m_archiveResourceCollection ? m_archiveResourceCollection->popSubframeArchive(frameName) : 0;
+    return m_archiveResourceCollection ? m_archiveResourceCollection->popSubframeArchive(frameName) : (WTF::PassRefPtr<WebCore::Archive>)0;
 }
 
 void DocumentLoader::clearArchiveResources()
diff -r 29063d7baea0 WebCore/loader/DocumentLoader.h
--- a/WebCore/loader/DocumentLoader.h	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/loader/DocumentLoader.h	Tue Dec 29 16:39:12 2009 +0800
@@ -163,7 +163,7 @@
         String clientRedirectDestinationForHistory() const { return urlForHistory(); }
         void setClientRedirectSourceForHistory(const String& clientedirectSourceForHistory) { m_clientRedirectSourceForHistory = clientedirectSourceForHistory; }
         
-        String serverRedirectSourceForHistory() const { return urlForHistory() == url() ? String() : urlForHistory(); } // null if no server redirect occurred.
+        String serverRedirectSourceForHistory() const { return urlForHistory() == url() ? (KURL)String() : urlForHistory(); } // null if no server redirect occurred.
         String serverRedirectDestinationForHistory() const { return url(); }
 
         bool didCreateGlobalHistoryEntry() const { return m_didCreateGlobalHistoryEntry; }
diff -r 29063d7baea0 WebCore/loader/FrameLoader.cpp
--- a/WebCore/loader/FrameLoader.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/loader/FrameLoader.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -1932,7 +1932,7 @@
     m_loadType = FrameLoadTypeStandard;
     RefPtr<DocumentLoader> loader = m_client->createDocumentLoader(request, substituteData);
     if (lockHistory && m_documentLoader)
-        loader->setClientRedirectSourceForHistory(m_documentLoader->didCreateGlobalHistoryEntry() ? m_documentLoader->urlForHistory() : m_documentLoader->clientRedirectSourceForHistory());
+        loader->setClientRedirectSourceForHistory(m_documentLoader->didCreateGlobalHistoryEntry() ? m_documentLoader->urlForHistory() : (KURL)m_documentLoader->clientRedirectSourceForHistory());
     load(loader.get());
 }
 
@@ -1956,7 +1956,7 @@
 {
     RefPtr<DocumentLoader> loader = m_client->createDocumentLoader(request, SubstituteData());
     if (lockHistory && m_documentLoader)
-        loader->setClientRedirectSourceForHistory(m_documentLoader->didCreateGlobalHistoryEntry() ? m_documentLoader->urlForHistory() : m_documentLoader->clientRedirectSourceForHistory());
+        loader->setClientRedirectSourceForHistory(m_documentLoader->didCreateGlobalHistoryEntry() ? m_documentLoader->urlForHistory() : (KURL)m_documentLoader->clientRedirectSourceForHistory());
 
     loader->setTriggeringAction(action);
     if (m_documentLoader)
diff -r 29063d7baea0 WebCore/loader/archive/ArchiveFactory.cpp
--- a/WebCore/loader/archive/ArchiveFactory.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/loader/archive/ArchiveFactory.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -76,7 +76,7 @@
 PassRefPtr<Archive> ArchiveFactory::create(SharedBuffer* data, const String& mimeType)
 {
     RawDataCreationFunction* function = archiveMIMETypes().get(mimeType);
-    return function ? function(data) : 0;
+    return function ? function(data) : (WTF::PassRefPtr<WebCore::Archive>)0;
 }
 
 void ArchiveFactory::registerKnownArchiveMIMETypes()
diff -r 29063d7baea0 WebCore/loader/archive/ArchiveResource.cpp
--- a/WebCore/loader/archive/ArchiveResource.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/loader/archive/ArchiveResource.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -35,17 +35,17 @@
 
 PassRefPtr<ArchiveResource> ArchiveResource::create(PassRefPtr<SharedBuffer> data, const KURL& url, const ResourceResponse& response)
 {
-    return data ? adoptRef(new ArchiveResource(data, url, response)) : 0;
+    return data ? adoptRef(new ArchiveResource(data, url, response)) : (WTF::PassRefPtr<WebCore::ArchiveResource>)0;
 }
 
 PassRefPtr<ArchiveResource> ArchiveResource::create(PassRefPtr<SharedBuffer> data, const KURL& url, const String& mimeType, const String& textEncoding, const String& frameName)
 {
-    return data ? adoptRef(new ArchiveResource(data, url, mimeType, textEncoding, frameName)) : 0;
+    return data ? adoptRef(new ArchiveResource(data, url, mimeType, textEncoding, frameName)) : (WTF::PassRefPtr<WebCore::ArchiveResource>)0;
 }
 
 PassRefPtr<ArchiveResource> ArchiveResource::create(PassRefPtr<SharedBuffer> data, const KURL& url, const String& mimeType, const String& textEncoding, const String& frameName, const ResourceResponse& resourceResponse)
 {
-    return data ? adoptRef(new ArchiveResource(data, url, mimeType, textEncoding, frameName, resourceResponse)) : 0;
+    return data ? adoptRef(new ArchiveResource(data, url, mimeType, textEncoding, frameName, resourceResponse)) : (WTF::PassRefPtr<WebCore::ArchiveResource>)0;
 }
 
 ArchiveResource::ArchiveResource(PassRefPtr<SharedBuffer> data, const KURL& url, const ResourceResponse& response)
diff -r 29063d7baea0 WebCore/page/animation/AnimationBase.cpp
--- a/WebCore/page/animation/AnimationBase.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/page/animation/AnimationBase.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -146,7 +146,7 @@
         for (unsigned i = 0; i < size; i++) {
             RefPtr<TransformOperation> fromOp = (i < fromSize) ? from.operations()[i].get() : 0;
             RefPtr<TransformOperation> toOp = (i < toSize) ? to.operations()[i].get() : 0;
-            RefPtr<TransformOperation> blendedOp = toOp ? toOp->blend(fromOp.get(), progress) : (fromOp ? fromOp->blend(0, progress, true) : 0);
+            RefPtr<TransformOperation> blendedOp = toOp ? toOp->blend(fromOp.get(), progress) : (fromOp ? fromOp->blend(0, progress, true) : (WTF::PassRefPtr<WebCore::TransformOperation>)0);
             if (blendedOp)
                 result.operations().append(blendedOp);
             else {
diff -r 29063d7baea0 WebCore/platform/network/ResourceHandleInternal.h
--- a/WebCore/platform/network/ResourceHandleInternal.h	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/platform/network/ResourceHandleInternal.h	Tue Dec 29 16:39:12 2009 +0800
@@ -47,7 +47,7 @@
 
 #if USE(SOUP)
 #include <libsoup/soup.h>
-class Frame;
+class WebCore::Frame;
 #endif
 
 #if PLATFORM(QT)
diff -r 29063d7baea0 WebCore/platform/text/TextCodecICU.cpp
--- a/WebCore/platform/text/TextCodecICU.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/platform/text/TextCodecICU.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -355,6 +355,7 @@
     }
 }
 
+U_CDECL_BEGIN
 // Invalid character handler when writing escaped entities for unrepresentable
 // characters. See the declaration of TextCodec::encode for more.
 static void urlEscapedEntityCallback(const void* context, UConverterFromUnicodeArgs* fromUArgs, const UChar* codeUnits, int32_t length,
@@ -370,6 +371,7 @@
         UCNV_FROM_U_CALLBACK_ESCAPE(context, fromUArgs, codeUnits, length, codePoint, reason, err);
 }
 
+
 // Substitutes special GBK characters, escaping all other unassigned entities.
 static void gbkCallbackEscape(const void* context, UConverterFromUnicodeArgs* fromUArgs, const UChar* codeUnits, int32_t length,
                               UChar32 codePoint, UConverterCallbackReason reason, UErrorCode* err) 
@@ -414,6 +416,8 @@
     UCNV_FROM_U_CALLBACK_SUBSTITUTE(context, fromUArgs, codeUnits, length, codePoint, reason, err);
 }
 
+U_CDECL_END
+
 CString TextCodecICU::encode(const UChar* characters, size_t length, UnencodableHandling handling)
 {
     if (!length)
diff -r 29063d7baea0 WebCore/plugins/npapi.cpp
--- a/WebCore/plugins/npapi.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/plugins/npapi.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -171,7 +171,9 @@
     pluginViewForInstance(instance)->popPopupsEnabledState();
 }
 
+U_CDECL_BEGIN
 void NPN_PluginThreadAsyncCall(NPP instance, void (*func) (void *), void *userData)
 {
     PluginMainThreadScheduler::scheduler().scheduleCall(instance, func, userData);
 }
+U_CDECL_END
diff -r 29063d7baea0 WebCore/rendering/RenderLayer.cpp
--- a/WebCore/rendering/RenderLayer.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/rendering/RenderLayer.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -3332,7 +3332,7 @@
 void RenderLayer::updateScrollCornerStyle()
 {
     RenderObject* actualRenderer = renderer()->node() ? renderer()->node()->shadowAncestorNode()->renderer() : renderer();
-    RefPtr<RenderStyle> corner = renderer()->hasOverflowClip() ? actualRenderer->getUncachedPseudoStyle(SCROLLBAR_CORNER, actualRenderer->style()) : 0;
+    RefPtr<RenderStyle> corner = renderer()->hasOverflowClip() ? actualRenderer->getUncachedPseudoStyle(SCROLLBAR_CORNER, actualRenderer->style()) : (WTF::PassRefPtr<WebCore::RenderStyle>)0;
     if (corner) {
         if (!m_scrollCorner) {
             m_scrollCorner = new (renderer()->renderArena()) RenderScrollbarPart(renderer()->document());
@@ -3348,7 +3348,7 @@
 void RenderLayer::updateResizerStyle()
 {
     RenderObject* actualRenderer = renderer()->node() ? renderer()->node()->shadowAncestorNode()->renderer() : renderer();
-    RefPtr<RenderStyle> resizer = renderer()->hasOverflowClip() ? actualRenderer->getUncachedPseudoStyle(RESIZER, actualRenderer->style()) : 0;
+    RefPtr<RenderStyle> resizer = renderer()->hasOverflowClip() ? actualRenderer->getUncachedPseudoStyle(RESIZER, actualRenderer->style()) : (WTF::PassRefPtr<WebCore::RenderStyle>)0;
     if (resizer) {
         if (!m_resizer) {
             m_resizer = new (renderer()->renderArena()) RenderScrollbarPart(renderer()->document());
diff -r 29063d7baea0 WebCore/rendering/RenderScrollbar.cpp
--- a/WebCore/rendering/RenderScrollbar.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/rendering/RenderScrollbar.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -199,7 +199,7 @@
     if (partType == NoPart)
         return;
 
-    RefPtr<RenderStyle> partStyle = !destroy ? getScrollbarPseudoStyle(partType,  pseudoForScrollbarPart(partType)) : 0;
+    RefPtr<RenderStyle> partStyle = !destroy ? getScrollbarPseudoStyle(partType,  pseudoForScrollbarPart(partType)) : (WTF::PassRefPtr<WebCore::RenderStyle>)0;
     
     bool needRenderer = !destroy && partStyle && partStyle->display() != NONE && partStyle->visibility() == VISIBLE;
     
diff -r 29063d7baea0 WebCore/rendering/RenderTextFragment.cpp
--- a/WebCore/rendering/RenderTextFragment.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/rendering/RenderTextFragment.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -28,7 +28,7 @@
 namespace WebCore {
 
 RenderTextFragment::RenderTextFragment(Node* node, StringImpl* str, int startOffset, int length)
-    : RenderText(node, str ? str->substring(startOffset, length) : 0)
+    : RenderText(node, str ? str->substring(startOffset, length) : (WTF::PassRefPtr<WebCore::StringImpl>)0)
     , m_start(startOffset)
     , m_end(length)
     , m_firstLetter(0)
diff -r 29063d7baea0 WebCore/svg/SVGComponentTransferFunctionElement.cpp
--- a/WebCore/svg/SVGComponentTransferFunctionElement.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGComponentTransferFunctionElement.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -30,7 +30,7 @@
 
 namespace WebCore {
 
-char SVGComponentTransferFunctionElementIdentifier[] = "SVGComponentTransferFunctionElement";
+const char SVGComponentTransferFunctionElementIdentifier[] = "SVGComponentTransferFunctionElement";
 
 SVGComponentTransferFunctionElement::SVGComponentTransferFunctionElement(const QualifiedName& tagName, Document* doc)
     : SVGElement(tagName, doc)
diff -r 29063d7baea0 WebCore/svg/SVGElement.cpp
--- a/WebCore/svg/SVGElement.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGElement.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -209,7 +209,7 @@
             event->setTarget(currentTarget);
             currentTarget->dispatchGenericEvent(event.release());
         }
-        currentTarget = (parent && parent->isSVGElement()) ? static_pointer_cast<SVGElement>(parent) : 0;
+        currentTarget = (parent && parent->isSVGElement()) ? static_pointer_cast<SVGElement>(parent) : (WTF::RefPtr<WebCore::SVGElement>)0;
     }
 }
 
diff -r 29063d7baea0 WebCore/svg/SVGExternalResourcesRequired.cpp
--- a/WebCore/svg/SVGExternalResourcesRequired.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGExternalResourcesRequired.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -30,7 +30,7 @@
 
 namespace WebCore {
 
-char SVGExternalResourcesRequiredIdentifier[] = "SVGExternalResourcesRequired";
+const char SVGExternalResourcesRequiredIdentifier[] = "SVGExternalResourcesRequired";
 
 SVGExternalResourcesRequired::SVGExternalResourcesRequired()
 {
diff -r 29063d7baea0 WebCore/svg/SVGExternalResourcesRequired.h
--- a/WebCore/svg/SVGExternalResourcesRequired.h	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGExternalResourcesRequired.h	Tue Dec 29 16:39:12 2009 +0800
@@ -27,7 +27,7 @@
 
 namespace WebCore {
 
-    extern char SVGExternalResourcesRequiredIdentifier[];
+    extern const char SVGExternalResourcesRequiredIdentifier[];
     class MappedAttribute;
 
     // Notes on a SVG 1.1 spec discrepancy:
diff -r 29063d7baea0 WebCore/svg/SVGFEDiffuseLightingElement.cpp
--- a/WebCore/svg/SVGFEDiffuseLightingElement.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGFEDiffuseLightingElement.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -35,8 +35,8 @@
 
 namespace WebCore {
 
-char SVGKernelUnitLengthXIdentifier[] = "SVGKernelUnitLengthX";
-char SVGKernelUnitLengthYIdentifier[] = "SVGKernelUnitLengthY";
+const char SVGKernelUnitLengthXIdentifier[] = "SVGKernelUnitLengthX";
+const char SVGKernelUnitLengthYIdentifier[] = "SVGKernelUnitLengthY";
 
 SVGFEDiffuseLightingElement::SVGFEDiffuseLightingElement(const QualifiedName& tagName, Document* doc)
     : SVGFilterPrimitiveStandardAttributes(tagName, doc)
diff -r 29063d7baea0 WebCore/svg/SVGFEGaussianBlurElement.cpp
--- a/WebCore/svg/SVGFEGaussianBlurElement.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGFEGaussianBlurElement.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -30,8 +30,8 @@
 
 namespace WebCore {
 
-char SVGStdDeviationXAttrIdentifier[] = "SVGStdDeviationXAttr";
-char SVGStdDeviationYAttrIdentifier[] = "SVGStdDeviationYAttr";
+const char SVGStdDeviationXAttrIdentifier[] = "SVGStdDeviationXAttr";
+const char SVGStdDeviationYAttrIdentifier[] = "SVGStdDeviationYAttr";
 
 SVGFEGaussianBlurElement::SVGFEGaussianBlurElement(const QualifiedName& tagName, Document* doc)
     : SVGFilterPrimitiveStandardAttributes(tagName, doc)
diff -r 29063d7baea0 WebCore/svg/SVGFELightElement.cpp
--- a/WebCore/svg/SVGFELightElement.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGFELightElement.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -29,7 +29,7 @@
 
 namespace WebCore {
 
-char SVGFELightElementIdentifier[] = "SVGFELightElement";
+const char SVGFELightElementIdentifier[] = "SVGFELightElement";
 
 SVGFELightElement::SVGFELightElement(const QualifiedName& tagName, Document* doc)
     : SVGElement(tagName, doc)
diff -r 29063d7baea0 WebCore/svg/SVGFETurbulenceElement.cpp
--- a/WebCore/svg/SVGFETurbulenceElement.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGFETurbulenceElement.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -29,8 +29,8 @@
 
 namespace WebCore {
 
-char SVGBaseFrequencyXIdentifier[] = "SVGBaseFrequencyX";
-char SVGBaseFrequencyYIdentifier[] = "SVGBaseFrequencyY";
+const char SVGBaseFrequencyXIdentifier[] = "SVGBaseFrequencyX";
+const char SVGBaseFrequencyYIdentifier[] = "SVGBaseFrequencyY";
 
 SVGFETurbulenceElement::SVGFETurbulenceElement(const QualifiedName& tagName, Document* doc)
     : SVGFilterPrimitiveStandardAttributes(tagName, doc)
diff -r 29063d7baea0 WebCore/svg/SVGFilterElement.cpp
--- a/WebCore/svg/SVGFilterElement.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGFilterElement.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -39,8 +39,8 @@
 
 namespace WebCore {
 
-char SVGFilterResXIdentifier[] = "SVGFilterResX";
-char SVGFilterResYIdentifier[] = "SVGFilterResY";
+const char SVGFilterResXIdentifier[] = "SVGFilterResX";
+const char SVGFilterResYIdentifier[] = "SVGFilterResY";
 
 SVGFilterElement::SVGFilterElement(const QualifiedName& tagName, Document* doc)
     : SVGStyledElement(tagName, doc)
diff -r 29063d7baea0 WebCore/svg/SVGFilterPrimitiveStandardAttributes.cpp
--- a/WebCore/svg/SVGFilterPrimitiveStandardAttributes.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGFilterPrimitiveStandardAttributes.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -33,7 +33,7 @@
 
 namespace WebCore {
 
-char SVGFilterPrimitiveStandardAttributesIdentifierIdentifier[] = "SVGFilterPrimitiveStandardAttributesIdentifier";
+const char SVGFilterPrimitiveStandardAttributesIdentifierIdentifier[] = "SVGFilterPrimitiveStandardAttributesIdentifier";
 
 SVGFilterPrimitiveStandardAttributes::SVGFilterPrimitiveStandardAttributes(const QualifiedName& tagName, Document* doc)
     : SVGStyledElement(tagName, doc)
diff -r 29063d7baea0 WebCore/svg/SVGFitToViewBox.cpp
--- a/WebCore/svg/SVGFitToViewBox.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGFitToViewBox.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -35,7 +35,7 @@
 
 namespace WebCore {
 
-char SVGFitToViewBoxIdentifier[] = "SVGFitToViewBox";
+const char SVGFitToViewBoxIdentifier[] = "SVGFitToViewBox";
 
 SVGFitToViewBox::SVGFitToViewBox()
 {
diff -r 29063d7baea0 WebCore/svg/SVGFitToViewBox.h
--- a/WebCore/svg/SVGFitToViewBox.h	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGFitToViewBox.h	Tue Dec 29 16:39:12 2009 +0800
@@ -27,7 +27,7 @@
 
 namespace WebCore {
 
-    extern char SVGFitToViewBoxIdentifier[];
+    extern const char SVGFitToViewBoxIdentifier[];
 
     class TransformationMatrix;
 
diff -r 29063d7baea0 WebCore/svg/SVGGradientElement.cpp
--- a/WebCore/svg/SVGGradientElement.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGGradientElement.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -37,7 +37,7 @@
 
 namespace WebCore {
 
-char SVGGradientElementIdentifier[] = "SVGGradientElement";
+const char SVGGradientElementIdentifier[] = "SVGGradientElement";
 
 SVGGradientElement::SVGGradientElement(const QualifiedName& tagName, Document* doc)
     : SVGStyledElement(tagName, doc)
diff -r 29063d7baea0 WebCore/svg/SVGGradientElement.h
--- a/WebCore/svg/SVGGradientElement.h	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGGradientElement.h	Tue Dec 29 16:39:12 2009 +0800
@@ -30,7 +30,7 @@
 
 namespace WebCore {
 
-    extern char SVGGradientElementIdentifier[];
+    extern const char SVGGradientElementIdentifier[];
 
     class SVGGradientElement : public SVGStyledElement,
                                public SVGURIReference,
diff -r 29063d7baea0 WebCore/svg/SVGMarkerElement.cpp
--- a/WebCore/svg/SVGMarkerElement.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGMarkerElement.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -35,8 +35,8 @@
 
 namespace WebCore {
 
-char SVGOrientTypeAttrIdentifier[] = "SVGOrientTypeAttr";
-char SVGOrientAngleAttrIdentifier[] = "SVGOrientAngleAttr";
+const char SVGOrientTypeAttrIdentifier[] = "SVGOrientTypeAttr";
+const char SVGOrientAngleAttrIdentifier[] = "SVGOrientAngleAttr";
 
 SVGMarkerElement::SVGMarkerElement(const QualifiedName& tagName, Document* doc)
     : SVGStyledElement(tagName, doc)
diff -r 29063d7baea0 WebCore/svg/SVGMarkerElement.h
--- a/WebCore/svg/SVGMarkerElement.h	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGMarkerElement.h	Tue Dec 29 16:39:12 2009 +0800
@@ -35,8 +35,8 @@
     class Document;
     class SVGAngle;
 
-    extern char SVGOrientTypeAttrIdentifier[];
-    extern char SVGOrientAngleAttrIdentifier[];
+    extern const char SVGOrientTypeAttrIdentifier[];
+    extern const char SVGOrientAngleAttrIdentifier[];
 
     class SVGMarkerElement : public SVGStyledElement,
                              public SVGLangSpace,
diff -r 29063d7baea0 WebCore/svg/SVGStyledElement.cpp
--- a/WebCore/svg/SVGStyledElement.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGStyledElement.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -44,7 +44,7 @@
 
 using namespace SVGNames;
 
-char SVGStyledElementIdentifier[] = "SVGStyledElement";
+const char SVGStyledElementIdentifier[] = "SVGStyledElement";
 static HashSet<const SVGStyledElement*>* gElementsWithInstanceUpdatesBlocked = 0;
 
 SVGStyledElement::SVGStyledElement(const QualifiedName& tagName, Document* doc)
diff -r 29063d7baea0 WebCore/svg/SVGStyledElement.h
--- a/WebCore/svg/SVGStyledElement.h	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGStyledElement.h	Tue Dec 29 16:39:12 2009 +0800
@@ -28,7 +28,7 @@
 
 namespace WebCore {
 
-    extern char SVGStyledElementIdentifier[];
+    extern const char SVGStyledElementIdentifier[];
     class SVGResource;
 
     class SVGStyledElement : public SVGElement,
diff -r 29063d7baea0 WebCore/svg/SVGStyledTransformableElement.cpp
--- a/WebCore/svg/SVGStyledTransformableElement.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGStyledTransformableElement.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -33,7 +33,7 @@
 
 namespace WebCore {
 
-char SVGStyledTransformableElementIdentifier[] = "SVGStyledTransformableElement";
+const char SVGStyledTransformableElementIdentifier[] = "SVGStyledTransformableElement";
 
 SVGStyledTransformableElement::SVGStyledTransformableElement(const QualifiedName& tagName, Document* doc)
     : SVGStyledLocatableElement(tagName, doc)
diff -r 29063d7baea0 WebCore/svg/SVGStyledTransformableElement.h
--- a/WebCore/svg/SVGStyledTransformableElement.h	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGStyledTransformableElement.h	Tue Dec 29 16:39:12 2009 +0800
@@ -28,7 +28,7 @@
 
 namespace WebCore {
 
-    extern char SVGStyledTransformableElementIdentifier[];
+    extern const char SVGStyledTransformableElementIdentifier[];
 
     class TransformationMatrix;
 
diff -r 29063d7baea0 WebCore/svg/SVGTextContentElement.cpp
--- a/WebCore/svg/SVGTextContentElement.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGTextContentElement.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -43,7 +43,7 @@
 
 namespace WebCore {
 
-char SVGTextContentElementIdentifier[] = "SVGTextContentElement";
+const char SVGTextContentElementIdentifier[] = "SVGTextContentElement";
 
 SVGTextContentElement::SVGTextContentElement(const QualifiedName& tagName, Document* doc)
     : SVGStyledElement(tagName, doc)
diff -r 29063d7baea0 WebCore/svg/SVGTextContentElement.h
--- a/WebCore/svg/SVGTextContentElement.h	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGTextContentElement.h	Tue Dec 29 16:39:12 2009 +0800
@@ -29,7 +29,7 @@
 
 namespace WebCore {
 
-    extern char SVGTextContentElementIdentifier[];
+    extern const char SVGTextContentElementIdentifier[];
 
     class SVGLength;
 
diff -r 29063d7baea0 WebCore/svg/SVGTextPositioningElement.cpp
--- a/WebCore/svg/SVGTextPositioningElement.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGTextPositioningElement.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -30,7 +30,7 @@
 
 namespace WebCore {
 
-char SVGTextPositioningElementIdentifier[] = "SVGTextPositioningElement";
+const char SVGTextPositioningElementIdentifier[] = "SVGTextPositioningElement";
 
 SVGTextPositioningElement::SVGTextPositioningElement(const QualifiedName& tagName, Document* doc)
     : SVGTextContentElement(tagName, doc)
diff -r 29063d7baea0 WebCore/svg/SVGTextPositioningElement.h
--- a/WebCore/svg/SVGTextPositioningElement.h	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGTextPositioningElement.h	Tue Dec 29 16:39:12 2009 +0800
@@ -28,7 +28,7 @@
 
 namespace WebCore {
 
-    extern char SVGTextPositioningElementIdentifier[];
+    extern const char SVGTextPositioningElementIdentifier[];
 
     class SVGTextPositioningElement : public SVGTextContentElement {
     public:
diff -r 29063d7baea0 WebCore/svg/SVGURIReference.cpp
--- a/WebCore/svg/SVGURIReference.cpp	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGURIReference.cpp	Tue Dec 29 16:39:12 2009 +0800
@@ -27,7 +27,7 @@
 
 namespace WebCore {
 
-char SVGURIReferenceIdentifier[] = "SVGURIReference";
+const char SVGURIReferenceIdentifier[] = "SVGURIReference";
 
 SVGURIReference::SVGURIReference()
 {
diff -r 29063d7baea0 WebCore/svg/SVGURIReference.h
--- a/WebCore/svg/SVGURIReference.h	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebCore/svg/SVGURIReference.h	Tue Dec 29 16:39:12 2009 +0800
@@ -27,7 +27,7 @@
 
 namespace WebCore {
 
-    extern char SVGURIReferenceIdentifier[];
+    extern const char SVGURIReferenceIdentifier[];
     class MappedAttribute;
 
     class SVGURIReference {
diff -r 29063d7baea0 WebKitTools/GNUmakefile.am
--- a/WebKitTools/GNUmakefile.am	Tue Dec 29 16:38:16 2009 +0800
+++ b/WebKitTools/GNUmakefile.am	Tue Dec 29 16:39:12 2009 +0800
@@ -13,8 +13,6 @@
 	WebKitTools/GtkLauncher/main.c
 
 Programs_GtkLauncher_CFLAGS = \
-	-ansi \
-	-fno-strict-aliasing \
 	$(global_cflags) \
 	$(GTK_CFLAGS) \
 	$(LIBSOUP_CFLAGS)
@@ -72,7 +70,6 @@
 	$(Programs_DumpRenderTree_CFLAGS)
 
 Programs_DumpRenderTree_CFLAGS = \
-	-fno-strict-aliasing \
 	$(dumprendertree_cppflags) \
 	$(global_cflags) \
 	$(GLOBALDEPS_CFLAGS) \
diff -r 29063d7baea0 configure.ac
--- a/configure.ac	Tue Dec 29 16:38:16 2009 +0800
+++ b/configure.ac	Tue Dec 29 16:39:12 2009 +0800
@@ -66,6 +66,9 @@
      *-*-darwin*)
        os_darwin=yes
        ;;
+     *solaris*)
+       os_sunos=yes
+       ;;
 esac
 
 case "$host_os" in
@@ -551,27 +554,34 @@
                              [Enable JIT compilation default=yes]),
               [],[enable_jit="yes"])
 if test "$enable_jit" = "yes"; then
-    case "$host_cpu" in
-        i*86|x86_64)
-            AC_DEFINE([ENABLE_JIT], [1], [Define to enable JIT])
-            AC_DEFINE([ENABLE_YARR], [1], [Define to enable YARR])
-            AC_DEFINE([ENABLE_YARR_JIT], [1], [Define to enable YARR JIT])
-            AC_DEFINE([ENABLE_JIT_OPTIMIZE_CALL], [1], [Define to enable optimizing calls])
-            AC_DEFINE([ENABLE_JIT_OPTIMIZE_PROPERTY_ACCESS], [1], [Define to enable optimized property access])
-            AC_DEFINE([ENABLE_JIT_OPTIMIZE_ARITHMETIC], [1], [Define to enable optimized arithmetic])
+    case "$host" in
+        *solaris*)
+            enable_jit="no (OS '$target_os' not supported)"
+        ;;
+        *)
             case "$host_cpu" in
-                i*86)
-                    AC_DEFINE([WTF_USE_JIT_STUB_ARGUMENT_VA_LIST], [1], [Use stub va_list])
+                i*86|x86_64)
+                    AC_DEFINE([ENABLE_JIT], [1], [Define to enable JIT])
+                    AC_DEFINE([ENABLE_YARR], [1], [Define to enable YARR])
+                    AC_DEFINE([ENABLE_YARR_JIT], [1], [Define to enable YARR JIT])
+                    AC_DEFINE([ENABLE_JIT_OPTIMIZE_CALL], [1], [Define to enable optimizing calls])
+                    AC_DEFINE([ENABLE_JIT_OPTIMIZE_PROPERTY_ACCESS], [1], [Define to enable optimized property access])
+                    AC_DEFINE([ENABLE_JIT_OPTIMIZE_ARITHMETIC], [1], [Define to enable optimized arithmetic])
+                    case "$host_cpu" in
+                        i*86)
+                            AC_DEFINE([WTF_USE_JIT_STUB_ARGUMENT_VA_LIST], [1], [Use stub va_list])
+                        ;;
+                        x86_64)
+                            AC_DEFINE([WTF_USE_JIT_STUB_ARGUMENT_REGISTER], [1], [Use stub register])
+                            AC_DEFINE([WTF_USE_ALTERNATE_JSIMMEDIATE], [1], [Use alternate JSImmediate])
+                        ;;
+                    esac
                 ;;
-                x86_64)
-                    AC_DEFINE([WTF_USE_JIT_STUB_ARGUMENT_REGISTER], [1], [Use stub register])
-                    AC_DEFINE([WTF_USE_ALTERNATE_JSIMMEDIATE], [1], [Use alternate JSImmediate])
+                *)
+                    enable_jit="no (CPU '$host_cpu' not supported)"
                 ;;
             esac
-        ;;
-        *)
-            enable_jit="no (CPU '$host_cpu' not supported)"
-        ;;
+    ;;
     esac
 fi
 AC_MSG_RESULT([$enable_jit])
@@ -631,11 +641,11 @@
 
 # Add the appropriate 'O' level for optimized builds
 if test "$enable_optimizations" = "yes"; then
-   CXXFLAGS="$CXXFLAGS -O2"
-   CFLAGS="$CFLAGS -O2"
+   CXXFLAGS="$CXXFLAGS -xO2"
+   CFLAGS="$CFLAGS -xO2"
 else
-   CXXFLAGS="$CXXFLAGS -O0"
-   CFLAGS="$CFLAGS -O0"
+   CXXFLAGS="$CXXFLAGS -xO0"
+   CFLAGS="$CFLAGS -xO0"
 fi
 
 PKG_CHECK_MODULES([LIBSOUP],
diff -r 102e5e9a65e8 autotools/webkit.m4
--- a/autotools/webkit.m4	Tue Dec 29 16:37:34 2009 +0800
+++ b/autotools/webkit.m4	Tue Dec 29 17:04:36 2009 +0800
@@ -153,6 +153,10 @@
 		UNICODE_CFLAGS=""
 		UNICODE_LIBS="-licuin -licuuc"
                 ;;
+            *solaris*)
+               UNICODE_CFLAGS="-I/usr/include"
+               UNICODE_LIBS="-licui18n -licuuc -licudata"
+                ;;
             *)
 		AC_PATH_PROG(icu_config, icu-config, no)
 		if test "$icu_config" = "no"; then
diff -r 102e5e9a65e8 WebCore/loader/icon/IconDatabase.cpp
--- a/WebCore/loader/icon/IconDatabase.cpp	Tue Dec 29 16:37:34 2009 +0800
+++ b/WebCore/loader/icon/IconDatabase.cpp	Tue Dec 29 17:15:55 2009 +0800
@@ -511,7 +511,7 @@
     if (!isOpen() || iconURLOriginal.isEmpty())
         return;
     
-    RefPtr<SharedBuffer> data = dataOriginal ? dataOriginal->copy() : 0;
+    RefPtr<SharedBuffer> data = dataOriginal ? dataOriginal->copy() : (WTF::PassRefPtr<WebCore::SharedBuffer>)0;
     String iconURL = iconURLOriginal.crossThreadString();
     
     Vector<String> pageURLs;
