diff -r 109daf33125e configure.ac
--- a/configure.ac	Wed Oct 13 00:43:41 2010 +0800
+++ b/configure.ac	Wed Oct 13 00:44:51 2010 +0800
@@ -131,7 +131,7 @@
 AC_SUBST([JPEG_LIBS])
 
 # Check for libpng the way Gtk+ does it
-for l in libpng libpng14 libpng12; do
+for l in libpng12; do
   AC_MSG_CHECKING(for $l)
   if $PKG_CONFIG --exists $l ; then
     AC_MSG_RESULT(yes)
@@ -157,7 +157,7 @@
       png_ok=no)
     AC_MSG_RESULT($png_ok)
     if test "$png_ok" = yes; then
-      PNG_LIBS='-lpng -lz'
+      PNG_LIBS='-lpng12 -lz'
     else
       AC_MSG_ERROR([PNG library (libpng) not found])
     fi
