--- webkit-1.2.7/JavaScriptCore/jsc.cpp	Fri Sep 10 15:20:23 2010
+++ webkit-1.2.7-patch/JavaScriptCore/jsc.cpp	Sat Mar 19 23:02:18 2011
@@ -304,6 +304,10 @@
     // Without this, Visual Studio will complain that this method does not return a value.
     return jsUndefined();
 #endif
+
+#if COMPILER(SUNCC)
+    return jsUndefined();
+#endif
 }
 
 // Use SEH for Release builds only to get rid of the crash report dialog
--- webkit-1.2.7/JavaScriptCore/wtf/FastMalloc.cpp	Fri Sep 10 15:20:23 2010
+++ webkit-1.2.7-patch/JavaScriptCore/wtf/FastMalloc.cpp	Sat Mar 19 23:02:18 2011
@@ -1499,7 +1499,7 @@
 void* TCMalloc_PageHeap::runScavengerThread(void* context)
 {
   static_cast<TCMalloc_PageHeap*>(context)->scavengerThread();
-#if COMPILER(MSVC)
+#if COMPILER(MSVC) || COMPILER(SUNCC)
   // Without this, Visual Studio will complain that this method does not return a value.
   return 0;
 #endif
@@ -3506,8 +3506,8 @@
   }
   if (!ret) {
 #ifdef WTF_CHANGES
-    if (crashOnFailure) // This branch should be optimized out by the compiler.
-        CRASH();
+    //if (crashOnFailure) // This branch should be optimized out by the compiler.
+     //   CRASH();
 #else
     errno = ENOMEM;
 #endif
