diff -r d4d2f47b7a3e WebCore/platform/gtk/GtkPluginWidget.cpp
--- a/WebCore/platform/gtk/GtkPluginWidget.cpp	Fri May 14 22:59:13 2010 +0800
+++ b/WebCore/platform/gtk/GtkPluginWidget.cpp	Tue May 18 00:03:26 2010 +0800
@@ -43,7 +43,7 @@
 void GtkPluginWidget::invalidateRect(const IntRect& _rect)
 {
     /* no need to */
-    if (GTK_WIDGET_NO_WINDOW(platformWidget()))
+    if (gtk_widget_get_has_window(platformWidget()))
         return;
 
     GdkWindow* window = platformWidget()->window;
@@ -71,11 +71,11 @@
         return;
 
     /* only paint widgets with NO_WINDOW this way */
-    if (!GTK_WIDGET_NO_WINDOW(platformWidget()))
+    if (!gtk_widget_get_has_window(platformWidget()))
         return;
 
     GtkWidget* widget = platformWidget();
-    ASSERT(GTK_WIDGET_NO_WINDOW(widget));
+    ASSERT(gtk_widget_get_has_window(widget));
 
     GdkEvent* event = gdk_event_new(GDK_EXPOSE);
     event->expose = *context->gdkExposeEvent();
diff -r d4d2f47b7a3e WebCore/platform/gtk/PlatformScreenGtk.cpp
--- a/WebCore/platform/gtk/PlatformScreenGtk.cpp	Fri May 14 22:59:13 2010 +0800
+++ b/WebCore/platform/gtk/PlatformScreenGtk.cpp	Tue May 18 00:03:26 2010 +0800
@@ -54,7 +54,7 @@
     if (!container)
         return 0;
 
-    if (!GTK_WIDGET_REALIZED(container)) {
+    if (!gtk_widget_get_realized(container)) {
         GtkWidget* toplevel = gtk_widget_get_toplevel(container);
 #if GTK_CHECK_VERSION(2, 18, 0)
         if (gtk_widget_is_toplevel(toplevel))
@@ -126,7 +126,7 @@
     if (!container)
         return FloatRect();
 
-    if (!GTK_WIDGET_REALIZED(container))
+    if (!gtk_widget_get_realized(container))
         return screenRect(widget);
 
     GdkDrawable* rootWindow = GDK_DRAWABLE(gtk_widget_get_root_window(container));
diff -r d4d2f47b7a3e WebCore/platform/gtk/ScrollbarGtk.cpp
--- a/WebCore/platform/gtk/ScrollbarGtk.cpp	Fri May 14 22:59:13 2010 +0800
+++ b/WebCore/platform/gtk/ScrollbarGtk.cpp	Tue May 18 00:03:26 2010 +0800
@@ -208,7 +208,7 @@
         return;
 
     GtkWidget* widget = platformWidget();
-    ASSERT(GTK_WIDGET_NO_WINDOW(widget));
+    ASSERT(gtk_widget_get_has_window(widget));
 
     GdkEvent* event = gdk_event_new(GDK_EXPOSE);
     event->expose = *context->gdkExposeEvent();
diff -r d4d2f47b7a3e WebKit/gtk/WebCoreSupport/ChromeClientGtk.cpp
--- a/WebKit/gtk/WebCoreSupport/ChromeClientGtk.cpp	Fri May 14 22:59:13 2010 +0800
+++ b/WebKit/gtk/WebCoreSupport/ChromeClientGtk.cpp	Tue May 18 00:03:26 2010 +0800
@@ -439,7 +439,7 @@
     // We need to queue a resize request only if the size changed,
     // otherwise we get into an infinite loop!
     GtkWidget* widget = GTK_WIDGET(m_webView);
-    if (GTK_WIDGET_REALIZED(widget)
+    if (gtk_widget_get_realized(widget)
         && (widget->requisition.height != size.height())
         || (widget->requisition.width != size.width()))
         gtk_widget_queue_resize_no_redraw(widget);
