diff -urNp weechat-0.3.4.orig/src/plugins/scripts/lua/weechat-lua-api.c weechat-0.3.4/src/plugins/scripts/lua/weechat-lua-api.c
--- weechat-0.3.4.orig/src/plugins/scripts/lua/weechat-lua-api.c	2011-02-17 00:00:18.000000000 +0100
+++ weechat-0.3.4/src/plugins/scripts/lua/weechat-lua-api.c	2011-02-17 00:00:58.000000000 +0100
@@ -1258,7 +1258,6 @@ weechat_lua_api_config_reload_cb (void *
     {
         lua_argv[0] = (script_callback->data) ? script_callback->data : empty_arg;
         lua_argv[1] = script_ptr2str (config_file);
-        lua_argv[2] = NULL;
         
         rc = (int *) weechat_lua_exec (script_callback->script,
                                        WEECHAT_SCRIPT_EXEC_INT,
