--- wesnoth-1.5.3/src/editor2/editor_common.hpp.fixfunc	2008-08-31 12:20:47.835904691 +0200
+++ wesnoth-1.5.3/src/editor2/editor_common.hpp	2008-08-31 12:27:06.932908252 +0200
@@ -28,7 +28,7 @@
 #define LOG_ED LOG_STREAM_INDENT(info, editor)
 #define WRN_ED LOG_STREAM_INDENT(warn, editor)
 #define ERR_ED LOG_STREAM_INDENT(err, editor)
-#define SCOPE_ED log_scope2(editor, __FUNCTION__)
+#define SCOPE_ED log_scope2(editor, __func__)
 
 class display;
 class gamemap;
