--- wesnoth-1.6a/src/util.hpp.fixround	2009-03-21 14:11:45.595300094 +0100
+++ wesnoth-1.6a/src/util.hpp	2009-03-21 14:11:48.736946843 +0100
@@ -25,6 +25,7 @@
 #include <cmath>
 #include <vector>
 #include <sstream>
+#include <math.h>
 
 template<typename T>
 inline bool is_odd(T num) {
--- wesnoth-1.6a/src/util.cpp.fixround	2009-03-21 14:10:24.528644800 +0100
+++ wesnoth-1.6a/src/util.cpp	2009-03-21 14:10:36.948535475 +0100
@@ -20,6 +20,7 @@
 #include "util.hpp"
 
 #include <cstdlib>
+#include <math.h>
 
 template<>
 int lexical_cast<int, const std::string&>(const std::string& a)
