--- wesnoth-1.5.3/src/astarnode.hpp.fixstd2	2008-08-30 17:09:22.884046751 +0200
+++ wesnoth-1.5.3/src/astarnode.hpp	2008-08-30 17:09:55.032035191 +0200
@@ -56,7 +56,7 @@
 		// and we divide by 300 * 10000 to avoid interfering with the defense subcost
 		// (see shortest_path_calculator::cost)
 		return distance_between(src, dst) -
-				(300.0 - sqrt( xdiff*xdiff + ydiff*ydiff)) / 3000000.0;
+				(300.0 - std::sqrt( xdiff*xdiff + ydiff*ydiff)) / 3000000.0;
 		
 		// TODO: move the heuristic function into the cost_calculator
 		// so we can use case-specific heuristic
