--- wildmidi-0.2.3.4/src/wildmidi.c	2010-12-26 11:30:09.484741055 +0100
+++ wildmidi-0.2.3.4-patch/src/wildmidi.c	2010-12-26 11:31:16.380552153 +0100
@@ -63,9 +63,17 @@
 #include "wildmidi_lib.h"
 
 #ifndef FNONBLOCK
+#ifdef __sun
+#define FNONBLOCK O_NONBLOCK
+#else
 #define FNONBLOCK _FNONBLOCK
+#endif
 #endif
 
+#ifndef MAP_FILE
+#define MAP_FILE 0
+#endif
+
 
 struct _midi_test {
 	unsigned char *data;
