--- xf86-input-synaptics-1.0.0.orig/tools/syndaemon.c	Sun Feb  1 17:36:56 2009
+++ xf86-input-synaptics-1.0.0/tools/syndaemon.c	Mon Mar  9 01:21:18 2009
@@ -106,11 +106,14 @@
     sigemptyset(&set);
     act.sa_handler = signal_handler;
     act.sa_mask = set;
+#ifndef SA_RESETHAND
 #ifdef SA_ONESHOT
-    act.sa_flags = SA_ONESHOT;
+#define SA_RESETHAND SA_ONESHOT
 #else
-    act.sa_flags = 0;
+#define SA_RESETHAND 0
 #endif
+#endif
+    act.sa_flags = SA_RESETHAND;
 
     for (i = 0; i < sizeof(signals) / sizeof(int); i++) {
 	if (sigaction(signals[i], &act, NULL) == -1) {
