--- xfce4-netload-plugin-1.0.0/configure.ac.orig	2011-06-11 12:55:21.597273900 -0400
+++ xfce4-netload-plugin-1.0.0/configure.ac	2011-06-11 12:56:27.875356400 -0400
@@ -54,7 +54,7 @@
 AC_TYPE_SIZE_T
 
 AC_CHECK_LIB(kstat, kstat_open, SOLLIBS="-lkstat -lsocket", SOLLIBS="")
-AC_CHECK_LIB(nsl, kstat_open, SOLLIBS="$SOLLIBS -linet_ntop", SOLLIBS="$SOLLIBS")
+AC_CHECK_LIB(nsl, inet_ntop, SOLLIBS="$SOLLIBS -lnsl", SOLLIBS="$SOLLIBS")
 AC_SUBST(SOLLIBS)
 
 dnl configure the panel plugin
