--- xmlto-0.0.23/Makefile.am.orig	2009-03-20 13:26:21.000000000 +0000
+++ xmlto-0.0.23/Makefile.am	2011-03-26 15:00:35.632297022 +0000
@@ -70,7 +70,7 @@
 $(man1_MANS): $(top_srcdir)/doc/xmlto.xml $(top_srcdir)/doc/xmlif.xml
 	for xml in xmlif.xml xmlto.xml; do \
 	  FORMAT_DIR=$(top_srcdir)/format \
-	   $(BASH) ./xmlto -o man/man1 man $(top_srcdir)/doc/$$xml ; \
+	   $(BASH) ./xmlto --skip-validation -o man/man1 man $(top_srcdir)/doc/$$xml ; \
 	done || ( RC=$$?; exit $$RC )
 
 TESTS_ENVIRONMENT = top_srcdir=$(top_srcdir) top_builddir=$(top_builddir)
--- xmlto-0.0.23/Makefile.in.orig	2009-09-21 14:08:16.000000000 +0100
+++ xmlto-0.0.23/Makefile.in	2011-03-26 15:00:18.059136714 +0000
@@ -850,7 +850,7 @@
 $(man1_MANS): $(top_srcdir)/doc/xmlto.xml $(top_srcdir)/doc/xmlif.xml
 	for xml in xmlif.xml xmlto.xml; do \
 	  FORMAT_DIR=$(top_srcdir)/format \
-	   $(BASH) ./xmlto -o man/man1 man $(top_srcdir)/doc/$$xml ; \
+	   $(BASH) ./xmlto --skip-validation -o man/man1 man $(top_srcdir)/doc/$$xml ; \
 	done || ( RC=$$?; exit $$RC )
 
 tag:
