--- xvidcore/build/generic/configure.in	2011-05-18 09:59:28.000000000 +0200
+++ xvidcore/build/generic/configure.in	2011-10-11 15:44:15.465740663 +0200
@@ -281,13 +281,26 @@
 ALTIVEC_CFLAGS=""
 PRE_SHARED_LIB=""
 case "$target_os" in
-     linux*|solaris*)
+     linux*)
 	AC_MSG_RESULT([ok])
 	STATIC_LIB="libxvidcore.\$(STATIC_EXTENSION)"
 	SHARED_LIB="libxvidcore.\$(SHARED_EXTENSION).\$(API_MAJOR).\$(API_MINOR)"
 	SPECIFIC_LDFLAGS="-Wl,-soname,libxvidcore.\$(SHARED_EXTENSION).\$(API_MAJOR) -shared -Wl,--version-script=libxvidcore.ld -lc -lm"
 	SPECIFIC_CFLAGS="-fPIC"
 	;;
+     solaris*)
+        AC_MSG_RESULT([ok])
+        STATIC_LIB="libxvidcore.\$(STATIC_EXTENSION)"
+        SHARED_LIB="libxvidcore.\$(SHARED_EXTENSION).\$(API_MAJOR).\$(API_MINOR)"
+        # because nasm don't produce pic code. And symbol 'main' is added and undefined.
+        #SPECIFIC_LDFLAGS="-Wl,-h,libxvidcore.\$(SHARED_EXTENSION).\$(API_MAJOR) -Wl,-G -Wl,-M,libxvidcore.ld -lc -lm"
+        SPECIFIC_LDFLAGS="-Wl,-h,libxvidcore.\$(SHARED_EXTENSION).\$(API_MAJOR) -Wl,-G -Wl,-M,libxvidcore.ld -lc -lm -shared -mimpure-text"
+        if test "`basename $CC`" = "gcc" ; then
+            SPECIFIC_CFLAGS="-fPIC"
+        else
+            SPECIFIC_CFLAGS="-KPIC -D__inline=inline"
+        fi
+        ;;
      *bsd*|irix*)
 	AC_MSG_RESULT([ok])
 	STATIC_LIB="libxvidcore.\$(STATIC_EXTENSION)"
