--- zeitgeist/zeitgeist-datahub-orig	2009-10-11 15:34:50.737423245 -0500
+++ zeitgeist/zeitgeist-datahub	2009-10-11 15:34:58.003764610 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2.6
 # -.- coding: utf-8 -.-
 
 # Zeitgeist
--- zeitgeist/zeitgeist-daemon-orig	2009-10-11 15:34:34.549214012 -0500
+++ zeitgeist/zeitgeist-daemon	2009-10-11 15:34:41.055903945 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2.6
 # -.- coding: utf-8 -.-
 
 # Zeitgeist
--- zeitgeist/extra/test-daemon.py-orig	2009-10-11 15:35:05.450431413 -0500
+++ zeitgeist/extra/test-daemon.py	2009-10-11 15:35:10.950998654 -0500
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python2.6
 # -.- coding: utf-8 -.-
 
 # Zeitgeist - Example / Test script
