--- qt-everywhere-opensource-src-4.7.0-beta2-orig/src/3rdparty/webkit/WebCore/bridge/runtime_array.h	2010-06-30 01:53:10.000000000 +0000
+++ qt-everywhere-opensource-src-4.7.0-beta2/src/3rdparty/webkit/WebCore/bridge/runtime_array.h	2010-07-25 02:20:31.493939000 +0000
@@ -33,6 +33,8 @@
     
 class RuntimeArray : public JSArray {
 public:
+    typedef Bindings::Array BindingsArray;
+  
     RuntimeArray(ExecState*, Bindings::Array*);
     virtual ~RuntimeArray();
 
@@ -50,7 +52,7 @@
     
     unsigned getLength() const { return getConcreteArray()->getLength(); }
     
-    Bindings::Array* getConcreteArray() const { return static_cast<Bindings::Array*>(subclassData()); }
+    Bindings::Array* getConcreteArray() const { return static_cast<BindingsArray*>(subclassData()); }
 
     static const ClassInfo s_info;
 
