--- src/3rdparty/webkit/JavaScriptCore/wtf/TCSystemAlloc.orig	2010-11-18 19:06:59.220416375 +0300
+++ src/3rdparty/webkit/JavaScriptCore/wtf/TCSystemAlloc.cpp	2010-11-18 19:11:33.541526702 +0300
@@ -194,10 +194,10 @@
 
   // Return the unused memory to the system
   if (adjust > 0) {
-    munmap(reinterpret_cast<void*>(ptr), adjust);
+    munmap(reinterpret_cast<char*>(ptr), adjust);
   }
   if (adjust < extra) {
-    munmap(reinterpret_cast<void*>(ptr + adjust + size), extra - adjust);
+    munmap(reinterpret_cast<char*>(ptr + adjust + size), extra - adjust);
   }
 
   ptr += adjust;
@@ -443,7 +443,7 @@
 
 void TCMalloc_SystemRelease(void* start, size_t length)
 {
-  void* newAddress = mmap(start, length, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_FIXED, -1, 0);
+  void* newAddress = mmap((char*)start, length, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS | MAP_FIXED, -1, 0);
   // If the mmap failed then that's ok, we just won't return the memory to the system.
   ASSERT_UNUSED(newAddress, newAddress == start || newAddress == reinterpret_cast<void*>(MAP_FAILED));
 }

