--- src/3rdparty/webkit/JavaScriptCore/wtf/MathExtras.h	Thu Jan  5 19:36:16 2012
+++ src/3rdparty/webkit/JavaScriptCore/wtf/MathExtras.h	Thu Jan  5 19:36:53 2012
@@ -190,7 +190,7 @@
 inline float rad2grad(float r) { return r * 200.0f / piFloat; }
 inline float grad2rad(float g) { return g * piFloat / 200.0f; }
 
-#if !COMPILER(MSVC) && !OS(ANDROID) && !OS(SYMBIAN)
+#if !COMPILER(MSVC) && !OS(ANDROID) && !OS(SYMBIAN) && !OS(SOLARIS)
 using std::isfinite;
 using std::isinf;
 using std::isnan;
