--- ./src/corelib/global/qglobal.h.orig	 2009-12-15 06:00:45.038516777 +0100
+++ ./src/corelib/global/qglobal.h	 2009-12-15 21:51:29.851260488 +0100
@@ -1700,7 +1700,7 @@
 #      define Q_FUNC_INFO __FUNCSIG__
 #  endif
 #else
-#   if defined(Q_OS_SOLARIS) || defined(Q_CC_XLC) || defined(Q_OS_SYMBIAN)
+#   if defined(Q_CC_XLC) || defined(Q_OS_SYMBIAN)
 #      define Q_FUNC_INFO __FILE__ "(line number unavailable)"
 #   else
         /* These two macros makes it possible to turn the builtin line expander into a
@@ -1711,7 +1711,7 @@
 #   endif
     /* The MIPSpro and RVCT compilers postpones macro expansion,
        and therefore macros must be in scope when being used. */
-#   if !defined(Q_CC_MIPS) && !defined(Q_CC_RVCT) && !defined(Q_CC_NOKIAX86)
+#   if !defined(Q_CC_MIPS) && !defined(Q_CC_RVCT) && !defined(Q_CC_NOKIAX86) && !defined(Q_CC_SUN)
 #       undef QT_STRINGIFY2
 #       undef QT_STRINGIFY
 #   endif
