--- ./src/gui/gui.pro.orig	2010-11-13 02:14:55.503445454 +0100
+++ ./src/gui/gui.pro	2010-11-13 02:40:51.755460095 +0100
@@ -107,7 +107,11 @@
                 mmx_compiler.commands += -Xarch_i386 -mmmx
                 mmx_compiler.commands += -Xarch_x86_64 -mmmx
             } else {
-                mmx_compiler.commands += -mmmx
+		solaris-cc* {
+			mmx_compiler.commands += -xarch=sse2
+		} else{
+                	mmx_compiler.commands += -mmmx
+		}
             }
 
             mmx_compiler.commands += $(CXXFLAGS) $(INCPATH) ${QMAKE_FILE_IN} -o ${QMAKE_FILE_OUT}
@@ -126,7 +130,11 @@
                 mmx3dnow_compiler.commands += -Xarch_i386 -m3dnow -Xarch_i386 -mmmx
                 mmx3dnow_compiler.commands += -Xarch_x86_64 -m3dnow -Xarch_x86_64 -mmmx
             } else {
-                mmx3dnow_compiler.commands += -m3dnow -mmmx
+		solaris-cc* {
+                    mmx3dnow_compiler.commands += -xarch=pentium_proa
+                } else {
+                    mmx3dnow_compiler.commands += -m3dnow -mmmx
+                }
             }
 
             mmx3dnow_compiler.commands += $(CXXFLAGS) $(INCPATH) ${QMAKE_FILE_IN} -o ${QMAKE_FILE_OUT}
@@ -144,7 +152,11 @@
                     sse3dnow_compiler.commands += -Xarch_i386 -m3dnow -Xarch_i386 -msse
                     sse3dnow_compiler.commands += -Xarch_x86_64 -m3dnow -Xarch_x86_64 -msse
                 } else {
-                    sse3dnow_compiler.commands += -m3dnow -msse
+		    solaris-cc* {
+                        sse3dnow_compiler.commands += -xarch=ssea
+                    } else {
+                        sse3dnow_compiler.commands += -m3dnow -msse
+                    }
                 }
 
                 sse3dnow_compiler.commands += $(CXXFLAGS) $(INCPATH) ${QMAKE_FILE_IN} -o ${QMAKE_FILE_OUT}
@@ -164,7 +176,11 @@
                 sse_compiler.commands += -Xarch_i386 -msse
                 sse_compiler.commands += -Xarch_x86_64 -msse
             } else {
-                sse_compiler.commands += -msse
+		solaris-cc* {
+                    sse_compiler.commands += -xarch=sse
+                } else {
+                    sse_compiler.commands += -msse
+                }
             }
 
             sse_compiler.commands += $(CXXFLAGS) $(INCPATH) ${QMAKE_FILE_IN} -o ${QMAKE_FILE_OUT}
@@ -183,7 +199,11 @@
                 sse2_compiler.commands += -Xarch_i386 -msse2
                 sse2_compiler.commands += -Xarch_x86_64 -msse2
             } else {
-                sse2_compiler.commands += -msse2
+		solaris-cc* {
+                    sse2_compiler.commands += -xarch=sse2
+                } else {
+                    sse2_compiler.commands += -msse2
+                }
             }
 
             sse2_compiler.commands += $(CXXFLAGS) $(INCPATH) ${QMAKE_FILE_IN} -o ${QMAKE_FILE_OUT}
@@ -202,7 +222,11 @@
                 ssse3_compiler.commands += -Xarch_i386 -mssse3
                 ssse3_compiler.commands += -Xarch_x86_64 -mssse3
             } else {
-                ssse3_compiler.commands += -mssse3
+		solaris-cc* {
+                    ssse3_compiler.commands += -xarch=ssse3
+                } else {
+                    ssse3_compiler.commands += -mssse3
+                }
             }
 
             ssse3_compiler.commands += $(CXXFLAGS) $(INCPATH) ${QMAKE_FILE_IN} -o ${QMAKE_FILE_OUT}
