--- ./src/corelib/global/qglobal.h.orig	2010-09-24 00:28:42.346692014 +0200
+++ ./src/corelib/global/qglobal.h	2010-09-24 00:29:18.683498213 +0200
@@ -664,7 +664,8 @@
     in which case _BOOL is not defined
         this is the default in 4.2 compatibility mode triggered by -compat=4 */
 #  if __SUNPRO_CC >= 0x500
-#    define QT_NO_TEMPLATE_TEMPLATE_PARAMETERS
+/* If it's 12.1 or later, template template parameters are ok. */
+// #    define QT_NO_TEMPLATE_TEMPLATE_PARAMETERS
    /* see http://developers.sun.com/sunstudio/support/Ccompare.html */
 #    if __SUNPRO_CC >= 0x590
 #      define Q_ALIGNOF(type)   __alignof__(type)
